// unitsnet-js/dist/absorbeddoseofionizingradiation.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbsorbedDoseOfIonizingRadiation = exports.AbsorbedDoseOfIonizingRadiationUnits = void 0;
/** AbsorbedDoseOfIonizingRadiationUnits enumeration */
var AbsorbedDoseOfIonizingRadiationUnits;
(function (AbsorbedDoseOfIonizingRadiationUnits) {
    /** The gray is the unit of ionizing radiation dose in the SI, defined as the absorption of one joule of radiation energy per kilogram of matter. */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Grays"] = 0] = "Grays";
    /** The rad is a unit of absorbed radiation dose, defined as 1 rad = 0.01 Gy = 0.01 J/kg. */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Rads"] = 1] = "Rads";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Femtograys"] = 2] = "Femtograys";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Picograys"] = 3] = "Picograys";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Nanograys"] = 4] = "Nanograys";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Micrograys"] = 5] = "Micrograys";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Milligrays"] = 6] = "Milligrays";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Centigrays"] = 7] = "Centigrays";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Kilograys"] = 8] = "Kilograys";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Megagrays"] = 9] = "Megagrays";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Gigagrays"] = 10] = "Gigagrays";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Teragrays"] = 11] = "Teragrays";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Petagrays"] = 12] = "Petagrays";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Millirads"] = 13] = "Millirads";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Kilorads"] = 14] = "Kilorads";
    /** */
    AbsorbedDoseOfIonizingRadiationUnits[AbsorbedDoseOfIonizingRadiationUnits["Megarads"] = 15] = "Megarads";
})(AbsorbedDoseOfIonizingRadiationUnits = exports.AbsorbedDoseOfIonizingRadiationUnits || (exports.AbsorbedDoseOfIonizingRadiationUnits = {}));
/** Absorbed dose is a dose quantity which is the measure of the energy deposited in matter by ionizing radiation per unit mass. */
class AbsorbedDoseOfIonizingRadiation {
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation.
     * @param value The value.
     * @param fromUnit The ‘AbsorbedDoseOfIonizingRadiation’ unit to create from.
     * The default unit is Grays
     */
    constructor(value, fromUnit = AbsorbedDoseOfIonizingRadiationUnits.Grays) {
        this.graysLazy = null;
        this.radsLazy = null;
        this.femtograysLazy = null;
        this.picograysLazy = null;
        this.nanograysLazy = null;
        this.micrograysLazy = null;
        this.milligraysLazy = null;
        this.centigraysLazy = null;
        this.kilograysLazy = null;
        this.megagraysLazy = null;
        this.gigagraysLazy = null;
        this.teragraysLazy = null;
        this.petagraysLazy = null;
        this.milliradsLazy = null;
        this.kiloradsLazy = null;
        this.megaradsLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of AbsorbedDoseOfIonizingRadiation is Grays.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** The gray is the unit of ionizing radiation dose in the SI, defined as the absorption of one joule of radiation energy per kilogram of matter. */
    get Grays() {
        if (this.graysLazy !== null) {
            return this.graysLazy;
        }
        return this.graysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Grays);
    }
    /** The rad is a unit of absorbed radiation dose, defined as 1 rad = 0.01 Gy = 0.01 J/kg. */
    get Rads() {
        if (this.radsLazy !== null) {
            return this.radsLazy;
        }
        return this.radsLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Rads);
    }
    /** */
    get Femtograys() {
        if (this.femtograysLazy !== null) {
            return this.femtograysLazy;
        }
        return this.femtograysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Femtograys);
    }
    /** */
    get Picograys() {
        if (this.picograysLazy !== null) {
            return this.picograysLazy;
        }
        return this.picograysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Picograys);
    }
    /** */
    get Nanograys() {
        if (this.nanograysLazy !== null) {
            return this.nanograysLazy;
        }
        return this.nanograysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Nanograys);
    }
    /** */
    get Micrograys() {
        if (this.micrograysLazy !== null) {
            return this.micrograysLazy;
        }
        return this.micrograysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Micrograys);
    }
    /** */
    get Milligrays() {
        if (this.milligraysLazy !== null) {
            return this.milligraysLazy;
        }
        return this.milligraysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Milligrays);
    }
    /** */
    get Centigrays() {
        if (this.centigraysLazy !== null) {
            return this.centigraysLazy;
        }
        return this.centigraysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Centigrays);
    }
    /** */
    get Kilograys() {
        if (this.kilograysLazy !== null) {
            return this.kilograysLazy;
        }
        return this.kilograysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Kilograys);
    }
    /** */
    get Megagrays() {
        if (this.megagraysLazy !== null) {
            return this.megagraysLazy;
        }
        return this.megagraysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Megagrays);
    }
    /** */
    get Gigagrays() {
        if (this.gigagraysLazy !== null) {
            return this.gigagraysLazy;
        }
        return this.gigagraysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Gigagrays);
    }
    /** */
    get Teragrays() {
        if (this.teragraysLazy !== null) {
            return this.teragraysLazy;
        }
        return this.teragraysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Teragrays);
    }
    /** */
    get Petagrays() {
        if (this.petagraysLazy !== null) {
            return this.petagraysLazy;
        }
        return this.petagraysLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Petagrays);
    }
    /** */
    get Millirads() {
        if (this.milliradsLazy !== null) {
            return this.milliradsLazy;
        }
        return this.milliradsLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Millirads);
    }
    /** */
    get Kilorads() {
        if (this.kiloradsLazy !== null) {
            return this.kiloradsLazy;
        }
        return this.kiloradsLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Kilorads);
    }
    /** */
    get Megarads() {
        if (this.megaradsLazy !== null) {
            return this.megaradsLazy;
        }
        return this.megaradsLazy = this.convertFromBase(AbsorbedDoseOfIonizingRadiationUnits.Megarads);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Grays
     * The gray is the unit of ionizing radiation dose in the SI, defined as the absorption of one joule of radiation energy per kilogram of matter.
     * @param value The unit as Grays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromGrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Grays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Rads
     * The rad is a unit of absorbed radiation dose, defined as 1 rad = 0.01 Gy = 0.01 J/kg.
     * @param value The unit as Rads to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromRads(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Rads);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Femtograys
     *
     * @param value The unit as Femtograys to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromFemtograys(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Femtograys);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Picograys
     *
     * @param value The unit as Picograys to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromPicograys(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Picograys);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Nanograys
     *
     * @param value The unit as Nanograys to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromNanograys(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Nanograys);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Micrograys
     *
     * @param value The unit as Micrograys to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromMicrograys(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Micrograys);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Milligrays
     *
     * @param value The unit as Milligrays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromMilligrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Milligrays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Centigrays
     *
     * @param value The unit as Centigrays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromCentigrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Centigrays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Kilograys
     *
     * @param value The unit as Kilograys to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromKilograys(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Kilograys);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Megagrays
     *
     * @param value The unit as Megagrays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromMegagrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Megagrays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Gigagrays
     *
     * @param value The unit as Gigagrays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromGigagrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Gigagrays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Teragrays
     *
     * @param value The unit as Teragrays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromTeragrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Teragrays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Petagrays
     *
     * @param value The unit as Petagrays to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromPetagrays(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Petagrays);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Millirads
     *
     * @param value The unit as Millirads to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromMillirads(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Millirads);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Kilorads
     *
     * @param value The unit as Kilorads to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromKilorads(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Kilorads);
    }
    /**
     * Create a new AbsorbedDoseOfIonizingRadiation instance from a Megarads
     *
     * @param value The unit as Megarads to create a new AbsorbedDoseOfIonizingRadiation from.
     * @returns The new AbsorbedDoseOfIonizingRadiation instance.
     */
    static FromMegarads(value) {
        return new AbsorbedDoseOfIonizingRadiation(value, AbsorbedDoseOfIonizingRadiationUnits.Megarads);
    }
    /**
     * Convert AbsorbedDoseOfIonizingRadiation to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case AbsorbedDoseOfIonizingRadiationUnits.Grays: return this.Grays;
            case AbsorbedDoseOfIonizingRadiationUnits.Rads: return this.Rads;
            case AbsorbedDoseOfIonizingRadiationUnits.Femtograys: return this.Femtograys;
            case AbsorbedDoseOfIonizingRadiationUnits.Picograys: return this.Picograys;
            case AbsorbedDoseOfIonizingRadiationUnits.Nanograys: return this.Nanograys;
            case AbsorbedDoseOfIonizingRadiationUnits.Micrograys: return this.Micrograys;
            case AbsorbedDoseOfIonizingRadiationUnits.Milligrays: return this.Milligrays;
            case AbsorbedDoseOfIonizingRadiationUnits.Centigrays: return this.Centigrays;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilograys: return this.Kilograys;
            case AbsorbedDoseOfIonizingRadiationUnits.Megagrays: return this.Megagrays;
            case AbsorbedDoseOfIonizingRadiationUnits.Gigagrays: return this.Gigagrays;
            case AbsorbedDoseOfIonizingRadiationUnits.Teragrays: return this.Teragrays;
            case AbsorbedDoseOfIonizingRadiationUnits.Petagrays: return this.Petagrays;
            case AbsorbedDoseOfIonizingRadiationUnits.Millirads: return this.Millirads;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilorads: return this.Kilorads;
            case AbsorbedDoseOfIonizingRadiationUnits.Megarads: return this.Megarads;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case AbsorbedDoseOfIonizingRadiationUnits.Grays:
                return this.value;
            case AbsorbedDoseOfIonizingRadiationUnits.Rads:
                return this.value * 100;
            case AbsorbedDoseOfIonizingRadiationUnits.Femtograys:
                return (this.value) / 1e-15;
            case AbsorbedDoseOfIonizingRadiationUnits.Picograys:
                return (this.value) / 1e-12;
            case AbsorbedDoseOfIonizingRadiationUnits.Nanograys:
                return (this.value) / 1e-9;
            case AbsorbedDoseOfIonizingRadiationUnits.Micrograys:
                return (this.value) / 0.000001;
            case AbsorbedDoseOfIonizingRadiationUnits.Milligrays:
                return (this.value) / 0.001;
            case AbsorbedDoseOfIonizingRadiationUnits.Centigrays:
                return (this.value) / 0.01;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilograys:
                return (this.value) / 1000;
            case AbsorbedDoseOfIonizingRadiationUnits.Megagrays:
                return (this.value) / 1000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Gigagrays:
                return (this.value) / 1000000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Teragrays:
                return (this.value) / 1000000000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Petagrays:
                return (this.value) / 1000000000000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Millirads:
                return (this.value * 100) / 0.001;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilorads:
                return (this.value * 100) / 1000;
            case AbsorbedDoseOfIonizingRadiationUnits.Megarads:
                return (this.value * 100) / 1000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case AbsorbedDoseOfIonizingRadiationUnits.Grays:
                return value;
            case AbsorbedDoseOfIonizingRadiationUnits.Rads:
                return value / 100;
            case AbsorbedDoseOfIonizingRadiationUnits.Femtograys:
                return (value) * 1e-15;
            case AbsorbedDoseOfIonizingRadiationUnits.Picograys:
                return (value) * 1e-12;
            case AbsorbedDoseOfIonizingRadiationUnits.Nanograys:
                return (value) * 1e-9;
            case AbsorbedDoseOfIonizingRadiationUnits.Micrograys:
                return (value) * 0.000001;
            case AbsorbedDoseOfIonizingRadiationUnits.Milligrays:
                return (value) * 0.001;
            case AbsorbedDoseOfIonizingRadiationUnits.Centigrays:
                return (value) * 0.01;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilograys:
                return (value) * 1000;
            case AbsorbedDoseOfIonizingRadiationUnits.Megagrays:
                return (value) * 1000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Gigagrays:
                return (value) * 1000000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Teragrays:
                return (value) * 1000000000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Petagrays:
                return (value) * 1000000000000000;
            case AbsorbedDoseOfIonizingRadiationUnits.Millirads:
                return (value / 100) * 0.001;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilorads:
                return (value / 100) * 1000;
            case AbsorbedDoseOfIonizingRadiationUnits.Megarads:
                return (value / 100) * 1000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the AbsorbedDoseOfIonizingRadiation to string.
     * Note! the default format for AbsorbedDoseOfIonizingRadiation is Grays.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the AbsorbedDoseOfIonizingRadiation.
     * @returns The string format of the AbsorbedDoseOfIonizingRadiation.
     */
    toString(unit = AbsorbedDoseOfIonizingRadiationUnits.Grays) {
        switch (unit) {
            case AbsorbedDoseOfIonizingRadiationUnits.Grays:
                return this.Grays + ` Gy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Rads:
                return this.Rads + ` rad`;
            case AbsorbedDoseOfIonizingRadiationUnits.Femtograys:
                return this.Femtograys + ` fGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Picograys:
                return this.Picograys + ` pGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Nanograys:
                return this.Nanograys + ` nGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Micrograys:
                return this.Micrograys + ` μGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Milligrays:
                return this.Milligrays + ` mGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Centigrays:
                return this.Centigrays + ` cGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilograys:
                return this.Kilograys + ` kGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Megagrays:
                return this.Megagrays + ` MGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Gigagrays:
                return this.Gigagrays + ` GGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Teragrays:
                return this.Teragrays + ` TGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Petagrays:
                return this.Petagrays + ` PGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Millirads:
                return this.Millirads + ` mrad`;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilorads:
                return this.Kilorads + ` krad`;
            case AbsorbedDoseOfIonizingRadiationUnits.Megarads:
                return this.Megarads + ` Mrad`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get AbsorbedDoseOfIonizingRadiation unit abbreviation.
     * Note! the default abbreviation for AbsorbedDoseOfIonizingRadiation is Grays.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the AbsorbedDoseOfIonizingRadiation.
     * @returns The abbreviation string of AbsorbedDoseOfIonizingRadiation.
     */
    getUnitAbbreviation(unitAbbreviation = AbsorbedDoseOfIonizingRadiationUnits.Grays) {
        switch (unitAbbreviation) {
            case AbsorbedDoseOfIonizingRadiationUnits.Grays:
                return `Gy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Rads:
                return `rad`;
            case AbsorbedDoseOfIonizingRadiationUnits.Femtograys:
                return `fGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Picograys:
                return `pGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Nanograys:
                return `nGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Micrograys:
                return `μGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Milligrays:
                return `mGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Centigrays:
                return `cGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilograys:
                return `kGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Megagrays:
                return `MGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Gigagrays:
                return `GGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Teragrays:
                return `TGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Petagrays:
                return `PGy`;
            case AbsorbedDoseOfIonizingRadiationUnits.Millirads:
                return `mrad`;
            case AbsorbedDoseOfIonizingRadiationUnits.Kilorads:
                return `krad`;
            case AbsorbedDoseOfIonizingRadiationUnits.Megarads:
                return `Mrad`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given AbsorbedDoseOfIonizingRadiation are equals to the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns True if the given AbsorbedDoseOfIonizingRadiation are equal to the current AbsorbedDoseOfIonizingRadiation.
     */
    equals(absorbedDoseOfIonizingRadiation) {
        return this.value === absorbedDoseOfIonizingRadiation.BaseValue;
    }
    /**
     * Compare the given AbsorbedDoseOfIonizingRadiation against the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns 0 if they are equal, -1 if the current AbsorbedDoseOfIonizingRadiation is less then other, 1 if the current AbsorbedDoseOfIonizingRadiation is greater then other.
     */
    compareTo(absorbedDoseOfIonizingRadiation) {
        if (this.value > absorbedDoseOfIonizingRadiation.BaseValue)
            return 1;
        if (this.value < absorbedDoseOfIonizingRadiation.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given AbsorbedDoseOfIonizingRadiation with the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns A new AbsorbedDoseOfIonizingRadiation instance with the results.
     */
    add(absorbedDoseOfIonizingRadiation) {
        return new AbsorbedDoseOfIonizingRadiation(this.value + absorbedDoseOfIonizingRadiation.BaseValue);
    }
    /**
     * Subtract the given AbsorbedDoseOfIonizingRadiation with the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns A new AbsorbedDoseOfIonizingRadiation instance with the results.
     */
    subtract(absorbedDoseOfIonizingRadiation) {
        return new AbsorbedDoseOfIonizingRadiation(this.value - absorbedDoseOfIonizingRadiation.BaseValue);
    }
    /**
     * Multiply the given AbsorbedDoseOfIonizingRadiation with the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns A new AbsorbedDoseOfIonizingRadiation instance with the results.
     */
    multiply(absorbedDoseOfIonizingRadiation) {
        return new AbsorbedDoseOfIonizingRadiation(this.value * absorbedDoseOfIonizingRadiation.BaseValue);
    }
    /**
     * Divide the given AbsorbedDoseOfIonizingRadiation with the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns A new AbsorbedDoseOfIonizingRadiation instance with the results.
     */
    divide(absorbedDoseOfIonizingRadiation) {
        return new AbsorbedDoseOfIonizingRadiation(this.value / absorbedDoseOfIonizingRadiation.BaseValue);
    }
    /**
     * Modulo the given AbsorbedDoseOfIonizingRadiation with the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns A new AbsorbedDoseOfIonizingRadiation instance with the results.
     */
    modulo(absorbedDoseOfIonizingRadiation) {
        return new AbsorbedDoseOfIonizingRadiation(this.value % absorbedDoseOfIonizingRadiation.BaseValue);
    }
    /**
     * Pow the given AbsorbedDoseOfIonizingRadiation with the current AbsorbedDoseOfIonizingRadiation.
     * @param absorbedDoseOfIonizingRadiation The other AbsorbedDoseOfIonizingRadiation.
     * @returns A new AbsorbedDoseOfIonizingRadiation instance with the results.
     */
    pow(absorbedDoseOfIonizingRadiation) {
        return new AbsorbedDoseOfIonizingRadiation(this.value ** absorbedDoseOfIonizingRadiation.BaseValue);
    }
}
exports.AbsorbedDoseOfIonizingRadiation = AbsorbedDoseOfIonizingRadiation;
