// unitsnet-js/dist/angle.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Angle = exports.AngleUnits = void 0;
/** AngleUnits enumeration */
var AngleUnits;
(function (AngleUnits) {
    /** */
    AngleUnits[AngleUnits["Radians"] = 0] = "Radians";
    /** */
    AngleUnits[AngleUnits["Degrees"] = 1] = "Degrees";
    /** */
    AngleUnits[AngleUnits["Arcminutes"] = 2] = "Arcminutes";
    /** */
    AngleUnits[AngleUnits["Arcseconds"] = 3] = "Arcseconds";
    /** */
    AngleUnits[AngleUnits["Gradians"] = 4] = "Gradians";
    /** */
    AngleUnits[AngleUnits["NatoMils"] = 5] = "NatoMils";
    /** */
    AngleUnits[AngleUnits["Revolutions"] = 6] = "Revolutions";
    /** */
    AngleUnits[AngleUnits["Tilt"] = 7] = "Tilt";
    /** */
    AngleUnits[AngleUnits["Nanoradians"] = 8] = "Nanoradians";
    /** */
    AngleUnits[AngleUnits["Microradians"] = 9] = "Microradians";
    /** */
    AngleUnits[AngleUnits["Milliradians"] = 10] = "Milliradians";
    /** */
    AngleUnits[AngleUnits["Centiradians"] = 11] = "Centiradians";
    /** */
    AngleUnits[AngleUnits["Deciradians"] = 12] = "Deciradians";
    /** */
    AngleUnits[AngleUnits["Nanodegrees"] = 13] = "Nanodegrees";
    /** */
    AngleUnits[AngleUnits["Microdegrees"] = 14] = "Microdegrees";
    /** */
    AngleUnits[AngleUnits["Millidegrees"] = 15] = "Millidegrees";
})(AngleUnits = exports.AngleUnits || (exports.AngleUnits = {}));
/** In geometry, an angle is the figure formed by two rays, called the sides of the angle, sharing a common endpoint, called the vertex of the angle. */
class Angle {
    /**
     * Create a new Angle.
     * @param value The value.
     * @param fromUnit The ‘Angle’ unit to create from.
     * The default unit is Degrees
     */
    constructor(value, fromUnit = AngleUnits.Degrees) {
        this.radiansLazy = null;
        this.degreesLazy = null;
        this.arcminutesLazy = null;
        this.arcsecondsLazy = null;
        this.gradiansLazy = null;
        this.natomilsLazy = null;
        this.revolutionsLazy = null;
        this.tiltLazy = null;
        this.nanoradiansLazy = null;
        this.microradiansLazy = null;
        this.milliradiansLazy = null;
        this.centiradiansLazy = null;
        this.deciradiansLazy = null;
        this.nanodegreesLazy = null;
        this.microdegreesLazy = null;
        this.millidegreesLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of Angle is Degrees.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Radians() {
        if (this.radiansLazy !== null) {
            return this.radiansLazy;
        }
        return this.radiansLazy = this.convertFromBase(AngleUnits.Radians);
    }
    /** */
    get Degrees() {
        if (this.degreesLazy !== null) {
            return this.degreesLazy;
        }
        return this.degreesLazy = this.convertFromBase(AngleUnits.Degrees);
    }
    /** */
    get Arcminutes() {
        if (this.arcminutesLazy !== null) {
            return this.arcminutesLazy;
        }
        return this.arcminutesLazy = this.convertFromBase(AngleUnits.Arcminutes);
    }
    /** */
    get Arcseconds() {
        if (this.arcsecondsLazy !== null) {
            return this.arcsecondsLazy;
        }
        return this.arcsecondsLazy = this.convertFromBase(AngleUnits.Arcseconds);
    }
    /** */
    get Gradians() {
        if (this.gradiansLazy !== null) {
            return this.gradiansLazy;
        }
        return this.gradiansLazy = this.convertFromBase(AngleUnits.Gradians);
    }
    /** */
    get NatoMils() {
        if (this.natomilsLazy !== null) {
            return this.natomilsLazy;
        }
        return this.natomilsLazy = this.convertFromBase(AngleUnits.NatoMils);
    }
    /** */
    get Revolutions() {
        if (this.revolutionsLazy !== null) {
            return this.revolutionsLazy;
        }
        return this.revolutionsLazy = this.convertFromBase(AngleUnits.Revolutions);
    }
    /** */
    get Tilt() {
        if (this.tiltLazy !== null) {
            return this.tiltLazy;
        }
        return this.tiltLazy = this.convertFromBase(AngleUnits.Tilt);
    }
    /** */
    get Nanoradians() {
        if (this.nanoradiansLazy !== null) {
            return this.nanoradiansLazy;
        }
        return this.nanoradiansLazy = this.convertFromBase(AngleUnits.Nanoradians);
    }
    /** */
    get Microradians() {
        if (this.microradiansLazy !== null) {
            return this.microradiansLazy;
        }
        return this.microradiansLazy = this.convertFromBase(AngleUnits.Microradians);
    }
    /** */
    get Milliradians() {
        if (this.milliradiansLazy !== null) {
            return this.milliradiansLazy;
        }
        return this.milliradiansLazy = this.convertFromBase(AngleUnits.Milliradians);
    }
    /** */
    get Centiradians() {
        if (this.centiradiansLazy !== null) {
            return this.centiradiansLazy;
        }
        return this.centiradiansLazy = this.convertFromBase(AngleUnits.Centiradians);
    }
    /** */
    get Deciradians() {
        if (this.deciradiansLazy !== null) {
            return this.deciradiansLazy;
        }
        return this.deciradiansLazy = this.convertFromBase(AngleUnits.Deciradians);
    }
    /** */
    get Nanodegrees() {
        if (this.nanodegreesLazy !== null) {
            return this.nanodegreesLazy;
        }
        return this.nanodegreesLazy = this.convertFromBase(AngleUnits.Nanodegrees);
    }
    /** */
    get Microdegrees() {
        if (this.microdegreesLazy !== null) {
            return this.microdegreesLazy;
        }
        return this.microdegreesLazy = this.convertFromBase(AngleUnits.Microdegrees);
    }
    /** */
    get Millidegrees() {
        if (this.millidegreesLazy !== null) {
            return this.millidegreesLazy;
        }
        return this.millidegreesLazy = this.convertFromBase(AngleUnits.Millidegrees);
    }
    /**
     * Create a new Angle instance from a Radians
     *
     * @param value The unit as Radians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromRadians(value) {
        return new Angle(value, AngleUnits.Radians);
    }
    /**
     * Create a new Angle instance from a Degrees
     *
     * @param value The unit as Degrees to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromDegrees(value) {
        return new Angle(value, AngleUnits.Degrees);
    }
    /**
     * Create a new Angle instance from a Arcminutes
     *
     * @param value The unit as Arcminutes to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromArcminutes(value) {
        return new Angle(value, AngleUnits.Arcminutes);
    }
    /**
     * Create a new Angle instance from a Arcseconds
     *
     * @param value The unit as Arcseconds to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromArcseconds(value) {
        return new Angle(value, AngleUnits.Arcseconds);
    }
    /**
     * Create a new Angle instance from a Gradians
     *
     * @param value The unit as Gradians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromGradians(value) {
        return new Angle(value, AngleUnits.Gradians);
    }
    /**
     * Create a new Angle instance from a NatoMils
     *
     * @param value The unit as NatoMils to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromNatoMils(value) {
        return new Angle(value, AngleUnits.NatoMils);
    }
    /**
     * Create a new Angle instance from a Revolutions
     *
     * @param value The unit as Revolutions to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromRevolutions(value) {
        return new Angle(value, AngleUnits.Revolutions);
    }
    /**
     * Create a new Angle instance from a Tilt
     *
     * @param value The unit as Tilt to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromTilt(value) {
        return new Angle(value, AngleUnits.Tilt);
    }
    /**
     * Create a new Angle instance from a Nanoradians
     *
     * @param value The unit as Nanoradians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromNanoradians(value) {
        return new Angle(value, AngleUnits.Nanoradians);
    }
    /**
     * Create a new Angle instance from a Microradians
     *
     * @param value The unit as Microradians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromMicroradians(value) {
        return new Angle(value, AngleUnits.Microradians);
    }
    /**
     * Create a new Angle instance from a Milliradians
     *
     * @param value The unit as Milliradians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromMilliradians(value) {
        return new Angle(value, AngleUnits.Milliradians);
    }
    /**
     * Create a new Angle instance from a Centiradians
     *
     * @param value The unit as Centiradians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromCentiradians(value) {
        return new Angle(value, AngleUnits.Centiradians);
    }
    /**
     * Create a new Angle instance from a Deciradians
     *
     * @param value The unit as Deciradians to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromDeciradians(value) {
        return new Angle(value, AngleUnits.Deciradians);
    }
    /**
     * Create a new Angle instance from a Nanodegrees
     *
     * @param value The unit as Nanodegrees to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromNanodegrees(value) {
        return new Angle(value, AngleUnits.Nanodegrees);
    }
    /**
     * Create a new Angle instance from a Microdegrees
     *
     * @param value The unit as Microdegrees to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromMicrodegrees(value) {
        return new Angle(value, AngleUnits.Microdegrees);
    }
    /**
     * Create a new Angle instance from a Millidegrees
     *
     * @param value The unit as Millidegrees to create a new Angle from.
     * @returns The new Angle instance.
     */
    static FromMillidegrees(value) {
        return new Angle(value, AngleUnits.Millidegrees);
    }
    /**
     * Convert Angle to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case AngleUnits.Radians: return this.Radians;
            case AngleUnits.Degrees: return this.Degrees;
            case AngleUnits.Arcminutes: return this.Arcminutes;
            case AngleUnits.Arcseconds: return this.Arcseconds;
            case AngleUnits.Gradians: return this.Gradians;
            case AngleUnits.NatoMils: return this.NatoMils;
            case AngleUnits.Revolutions: return this.Revolutions;
            case AngleUnits.Tilt: return this.Tilt;
            case AngleUnits.Nanoradians: return this.Nanoradians;
            case AngleUnits.Microradians: return this.Microradians;
            case AngleUnits.Milliradians: return this.Milliradians;
            case AngleUnits.Centiradians: return this.Centiradians;
            case AngleUnits.Deciradians: return this.Deciradians;
            case AngleUnits.Nanodegrees: return this.Nanodegrees;
            case AngleUnits.Microdegrees: return this.Microdegrees;
            case AngleUnits.Millidegrees: return this.Millidegrees;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case AngleUnits.Radians:
                return this.value / 180 * Math.PI;
            case AngleUnits.Degrees:
                return this.value;
            case AngleUnits.Arcminutes:
                return this.value * 60;
            case AngleUnits.Arcseconds:
                return this.value * 3600;
            case AngleUnits.Gradians:
                return this.value / 0.9;
            case AngleUnits.NatoMils:
                return this.value * 160 / 9;
            case AngleUnits.Revolutions:
                return this.value / 360;
            case AngleUnits.Tilt:
                return Math.sin(this.value / 180 * Math.PI);
            case AngleUnits.Nanoradians:
                return (this.value / 180 * Math.PI) / 1e-9;
            case AngleUnits.Microradians:
                return (this.value / 180 * Math.PI) / 0.000001;
            case AngleUnits.Milliradians:
                return (this.value / 180 * Math.PI) / 0.001;
            case AngleUnits.Centiradians:
                return (this.value / 180 * Math.PI) / 0.01;
            case AngleUnits.Deciradians:
                return (this.value / 180 * Math.PI) / 0.1;
            case AngleUnits.Nanodegrees:
                return (this.value) / 1e-9;
            case AngleUnits.Microdegrees:
                return (this.value) / 0.000001;
            case AngleUnits.Millidegrees:
                return (this.value) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case AngleUnits.Radians:
                return value * 180 / Math.PI;
            case AngleUnits.Degrees:
                return value;
            case AngleUnits.Arcminutes:
                return value / 60;
            case AngleUnits.Arcseconds:
                return value / 3600;
            case AngleUnits.Gradians:
                return value * 0.9;
            case AngleUnits.NatoMils:
                return value * 9 / 160;
            case AngleUnits.Revolutions:
                return value * 360;
            case AngleUnits.Tilt:
                return Math.asin(value) * 180 / Math.PI;
            case AngleUnits.Nanoradians:
                return (value * 180 / Math.PI) * 1e-9;
            case AngleUnits.Microradians:
                return (value * 180 / Math.PI) * 0.000001;
            case AngleUnits.Milliradians:
                return (value * 180 / Math.PI) * 0.001;
            case AngleUnits.Centiradians:
                return (value * 180 / Math.PI) * 0.01;
            case AngleUnits.Deciradians:
                return (value * 180 / Math.PI) * 0.1;
            case AngleUnits.Nanodegrees:
                return (value) * 1e-9;
            case AngleUnits.Microdegrees:
                return (value) * 0.000001;
            case AngleUnits.Millidegrees:
                return (value) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the Angle to string.
     * Note! the default format for Angle is Degrees.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the Angle.
     * @returns The string format of the Angle.
     */
    toString(unit = AngleUnits.Degrees) {
        switch (unit) {
            case AngleUnits.Radians:
                return this.Radians + ` rad`;
            case AngleUnits.Degrees:
                return this.Degrees + ` °`;
            case AngleUnits.Arcminutes:
                return this.Arcminutes + ` '`;
            case AngleUnits.Arcseconds:
                return this.Arcseconds + ` ″`;
            case AngleUnits.Gradians:
                return this.Gradians + ` g`;
            case AngleUnits.NatoMils:
                return this.NatoMils + ` mil`;
            case AngleUnits.Revolutions:
                return this.Revolutions + ` r`;
            case AngleUnits.Tilt:
                return this.Tilt + ` sin(θ)`;
            case AngleUnits.Nanoradians:
                return this.Nanoradians + ` nrad`;
            case AngleUnits.Microradians:
                return this.Microradians + ` μrad`;
            case AngleUnits.Milliradians:
                return this.Milliradians + ` mrad`;
            case AngleUnits.Centiradians:
                return this.Centiradians + ` crad`;
            case AngleUnits.Deciradians:
                return this.Deciradians + ` drad`;
            case AngleUnits.Nanodegrees:
                return this.Nanodegrees + ` n°`;
            case AngleUnits.Microdegrees:
                return this.Microdegrees + ` μ°`;
            case AngleUnits.Millidegrees:
                return this.Millidegrees + ` m°`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get Angle unit abbreviation.
     * Note! the default abbreviation for Angle is Degrees.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the Angle.
     * @returns The abbreviation string of Angle.
     */
    getUnitAbbreviation(unitAbbreviation = AngleUnits.Degrees) {
        switch (unitAbbreviation) {
            case AngleUnits.Radians:
                return `rad`;
            case AngleUnits.Degrees:
                return `°`;
            case AngleUnits.Arcminutes:
                return `'`;
            case AngleUnits.Arcseconds:
                return `″`;
            case AngleUnits.Gradians:
                return `g`;
            case AngleUnits.NatoMils:
                return `mil`;
            case AngleUnits.Revolutions:
                return `r`;
            case AngleUnits.Tilt:
                return `sin(θ)`;
            case AngleUnits.Nanoradians:
                return `nrad`;
            case AngleUnits.Microradians:
                return `μrad`;
            case AngleUnits.Milliradians:
                return `mrad`;
            case AngleUnits.Centiradians:
                return `crad`;
            case AngleUnits.Deciradians:
                return `drad`;
            case AngleUnits.Nanodegrees:
                return `n°`;
            case AngleUnits.Microdegrees:
                return `μ°`;
            case AngleUnits.Millidegrees:
                return `m°`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given Angle are equals to the current Angle.
     * @param angle The other Angle.
     * @returns True if the given Angle are equal to the current Angle.
     */
    equals(angle) {
        return this.value === angle.BaseValue;
    }
    /**
     * Compare the given Angle against the current Angle.
     * @param angle The other Angle.
     * @returns 0 if they are equal, -1 if the current Angle is less then other, 1 if the current Angle is greater then other.
     */
    compareTo(angle) {
        if (this.value > angle.BaseValue)
            return 1;
        if (this.value < angle.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given Angle with the current Angle.
     * @param angle The other Angle.
     * @returns A new Angle instance with the results.
     */
    add(angle) {
        return new Angle(this.value + angle.BaseValue);
    }
    /**
     * Subtract the given Angle with the current Angle.
     * @param angle The other Angle.
     * @returns A new Angle instance with the results.
     */
    subtract(angle) {
        return new Angle(this.value - angle.BaseValue);
    }
    /**
     * Multiply the given Angle with the current Angle.
     * @param angle The other Angle.
     * @returns A new Angle instance with the results.
     */
    multiply(angle) {
        return new Angle(this.value * angle.BaseValue);
    }
    /**
     * Divide the given Angle with the current Angle.
     * @param angle The other Angle.
     * @returns A new Angle instance with the results.
     */
    divide(angle) {
        return new Angle(this.value / angle.BaseValue);
    }
    /**
     * Modulo the given Angle with the current Angle.
     * @param angle The other Angle.
     * @returns A new Angle instance with the results.
     */
    modulo(angle) {
        return new Angle(this.value % angle.BaseValue);
    }
    /**
     * Pow the given Angle with the current Angle.
     * @param angle The other Angle.
     * @returns A new Angle instance with the results.
     */
    pow(angle) {
        return new Angle(this.value ** angle.BaseValue);
    }
}
exports.Angle = Angle;
