// unitsnet-js/dist/areadensity.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AreaDensity = exports.AreaDensityUnits = void 0;
/** AreaDensityUnits enumeration */
var AreaDensityUnits;
(function (AreaDensityUnits) {
    /** */
    AreaDensityUnits[AreaDensityUnits["KilogramsPerSquareMeter"] = 0] = "KilogramsPerSquareMeter";
    /** Also known as grammage for paper industry. In fiber industry used with abbreviation 'gsm'. */
    AreaDensityUnits[AreaDensityUnits["GramsPerSquareMeter"] = 1] = "GramsPerSquareMeter";
    /** */
    AreaDensityUnits[AreaDensityUnits["MilligramsPerSquareMeter"] = 2] = "MilligramsPerSquareMeter";
})(AreaDensityUnits = exports.AreaDensityUnits || (exports.AreaDensityUnits = {}));
/** The area density of a two-dimensional object is calculated as the mass per unit area. For paper this is also called grammage. */
class AreaDensity {
    /**
     * Create a new AreaDensity.
     * @param value The value.
     * @param fromUnit The ‘AreaDensity’ unit to create from.
     * The default unit is KilogramsPerSquareMeter
     */
    constructor(value, fromUnit = AreaDensityUnits.KilogramsPerSquareMeter) {
        this.kilogramspersquaremeterLazy = null;
        this.gramspersquaremeterLazy = null;
        this.milligramspersquaremeterLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of AreaDensity is KilogramsPerSquareMeter.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get KilogramsPerSquareMeter() {
        if (this.kilogramspersquaremeterLazy !== null) {
            return this.kilogramspersquaremeterLazy;
        }
        return this.kilogramspersquaremeterLazy = this.convertFromBase(AreaDensityUnits.KilogramsPerSquareMeter);
    }
    /** Also known as grammage for paper industry. In fiber industry used with abbreviation 'gsm'. */
    get GramsPerSquareMeter() {
        if (this.gramspersquaremeterLazy !== null) {
            return this.gramspersquaremeterLazy;
        }
        return this.gramspersquaremeterLazy = this.convertFromBase(AreaDensityUnits.GramsPerSquareMeter);
    }
    /** */
    get MilligramsPerSquareMeter() {
        if (this.milligramspersquaremeterLazy !== null) {
            return this.milligramspersquaremeterLazy;
        }
        return this.milligramspersquaremeterLazy = this.convertFromBase(AreaDensityUnits.MilligramsPerSquareMeter);
    }
    /**
     * Create a new AreaDensity instance from a KilogramsPerSquareMeter
     *
     * @param value The unit as KilogramsPerSquareMeter to create a new AreaDensity from.
     * @returns The new AreaDensity instance.
     */
    static FromKilogramsPerSquareMeter(value) {
        return new AreaDensity(value, AreaDensityUnits.KilogramsPerSquareMeter);
    }
    /**
     * Create a new AreaDensity instance from a GramsPerSquareMeter
     * Also known as grammage for paper industry. In fiber industry used with abbreviation 'gsm'.
     * @param value The unit as GramsPerSquareMeter to create a new AreaDensity from.
     * @returns The new AreaDensity instance.
     */
    static FromGramsPerSquareMeter(value) {
        return new AreaDensity(value, AreaDensityUnits.GramsPerSquareMeter);
    }
    /**
     * Create a new AreaDensity instance from a MilligramsPerSquareMeter
     *
     * @param value The unit as MilligramsPerSquareMeter to create a new AreaDensity from.
     * @returns The new AreaDensity instance.
     */
    static FromMilligramsPerSquareMeter(value) {
        return new AreaDensity(value, AreaDensityUnits.MilligramsPerSquareMeter);
    }
    /**
     * Convert AreaDensity to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case AreaDensityUnits.KilogramsPerSquareMeter: return this.KilogramsPerSquareMeter;
            case AreaDensityUnits.GramsPerSquareMeter: return this.GramsPerSquareMeter;
            case AreaDensityUnits.MilligramsPerSquareMeter: return this.MilligramsPerSquareMeter;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case AreaDensityUnits.KilogramsPerSquareMeter:
                return this.value;
            case AreaDensityUnits.GramsPerSquareMeter:
                return this.value * 1000;
            case AreaDensityUnits.MilligramsPerSquareMeter:
                return this.value * 1000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case AreaDensityUnits.KilogramsPerSquareMeter:
                return value;
            case AreaDensityUnits.GramsPerSquareMeter:
                return value / 1000;
            case AreaDensityUnits.MilligramsPerSquareMeter:
                return value / 1000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the AreaDensity to string.
     * Note! the default format for AreaDensity is KilogramsPerSquareMeter.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the AreaDensity.
     * @returns The string format of the AreaDensity.
     */
    toString(unit = AreaDensityUnits.KilogramsPerSquareMeter) {
        switch (unit) {
            case AreaDensityUnits.KilogramsPerSquareMeter:
                return this.KilogramsPerSquareMeter + ` kg/m²`;
            case AreaDensityUnits.GramsPerSquareMeter:
                return this.GramsPerSquareMeter + ` g/m²`;
            case AreaDensityUnits.MilligramsPerSquareMeter:
                return this.MilligramsPerSquareMeter + ` mg/m²`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get AreaDensity unit abbreviation.
     * Note! the default abbreviation for AreaDensity is KilogramsPerSquareMeter.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the AreaDensity.
     * @returns The abbreviation string of AreaDensity.
     */
    getUnitAbbreviation(unitAbbreviation = AreaDensityUnits.KilogramsPerSquareMeter) {
        switch (unitAbbreviation) {
            case AreaDensityUnits.KilogramsPerSquareMeter:
                return `kg/m²`;
            case AreaDensityUnits.GramsPerSquareMeter:
                return `g/m²`;
            case AreaDensityUnits.MilligramsPerSquareMeter:
                return `mg/m²`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given AreaDensity are equals to the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns True if the given AreaDensity are equal to the current AreaDensity.
     */
    equals(areaDensity) {
        return this.value === areaDensity.BaseValue;
    }
    /**
     * Compare the given AreaDensity against the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns 0 if they are equal, -1 if the current AreaDensity is less then other, 1 if the current AreaDensity is greater then other.
     */
    compareTo(areaDensity) {
        if (this.value > areaDensity.BaseValue)
            return 1;
        if (this.value < areaDensity.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given AreaDensity with the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns A new AreaDensity instance with the results.
     */
    add(areaDensity) {
        return new AreaDensity(this.value + areaDensity.BaseValue);
    }
    /**
     * Subtract the given AreaDensity with the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns A new AreaDensity instance with the results.
     */
    subtract(areaDensity) {
        return new AreaDensity(this.value - areaDensity.BaseValue);
    }
    /**
     * Multiply the given AreaDensity with the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns A new AreaDensity instance with the results.
     */
    multiply(areaDensity) {
        return new AreaDensity(this.value * areaDensity.BaseValue);
    }
    /**
     * Divide the given AreaDensity with the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns A new AreaDensity instance with the results.
     */
    divide(areaDensity) {
        return new AreaDensity(this.value / areaDensity.BaseValue);
    }
    /**
     * Modulo the given AreaDensity with the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns A new AreaDensity instance with the results.
     */
    modulo(areaDensity) {
        return new AreaDensity(this.value % areaDensity.BaseValue);
    }
    /**
     * Pow the given AreaDensity with the current AreaDensity.
     * @param areaDensity The other AreaDensity.
     * @returns A new AreaDensity instance with the results.
     */
    pow(areaDensity) {
        return new AreaDensity(this.value ** areaDensity.BaseValue);
    }
}
exports.AreaDensity = AreaDensity;
