// unitsnet-js/dist/electricresistivity.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElectricResistivity = exports.ElectricResistivityUnits = void 0;
/** ElectricResistivityUnits enumeration */
var ElectricResistivityUnits;
(function (ElectricResistivityUnits) {
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["OhmMeters"] = 0] = "OhmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["OhmsCentimeter"] = 1] = "OhmsCentimeter";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["PicoohmMeters"] = 2] = "PicoohmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["NanoohmMeters"] = 3] = "NanoohmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["MicroohmMeters"] = 4] = "MicroohmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["MilliohmMeters"] = 5] = "MilliohmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["KiloohmMeters"] = 6] = "KiloohmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["MegaohmMeters"] = 7] = "MegaohmMeters";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["PicoohmsCentimeter"] = 8] = "PicoohmsCentimeter";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["NanoohmsCentimeter"] = 9] = "NanoohmsCentimeter";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["MicroohmsCentimeter"] = 10] = "MicroohmsCentimeter";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["MilliohmsCentimeter"] = 11] = "MilliohmsCentimeter";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["KiloohmsCentimeter"] = 12] = "KiloohmsCentimeter";
    /** */
    ElectricResistivityUnits[ElectricResistivityUnits["MegaohmsCentimeter"] = 13] = "MegaohmsCentimeter";
})(ElectricResistivityUnits = exports.ElectricResistivityUnits || (exports.ElectricResistivityUnits = {}));
/** Electrical resistivity (also known as resistivity, specific electrical resistance, or volume resistivity) is a fundamental property that quantifies how strongly a given material opposes the flow of electric current. */
class ElectricResistivity {
    /**
     * Create a new ElectricResistivity.
     * @param value The value.
     * @param fromUnit The ‘ElectricResistivity’ unit to create from.
     * The default unit is OhmMeters
     */
    constructor(value, fromUnit = ElectricResistivityUnits.OhmMeters) {
        this.ohmmetersLazy = null;
        this.ohmscentimeterLazy = null;
        this.picoohmmetersLazy = null;
        this.nanoohmmetersLazy = null;
        this.microohmmetersLazy = null;
        this.milliohmmetersLazy = null;
        this.kiloohmmetersLazy = null;
        this.megaohmmetersLazy = null;
        this.picoohmscentimeterLazy = null;
        this.nanoohmscentimeterLazy = null;
        this.microohmscentimeterLazy = null;
        this.milliohmscentimeterLazy = null;
        this.kiloohmscentimeterLazy = null;
        this.megaohmscentimeterLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of ElectricResistivity is OhmMeters.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get OhmMeters() {
        if (this.ohmmetersLazy !== null) {
            return this.ohmmetersLazy;
        }
        return this.ohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.OhmMeters);
    }
    /** */
    get OhmsCentimeter() {
        if (this.ohmscentimeterLazy !== null) {
            return this.ohmscentimeterLazy;
        }
        return this.ohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.OhmsCentimeter);
    }
    /** */
    get PicoohmMeters() {
        if (this.picoohmmetersLazy !== null) {
            return this.picoohmmetersLazy;
        }
        return this.picoohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.PicoohmMeters);
    }
    /** */
    get NanoohmMeters() {
        if (this.nanoohmmetersLazy !== null) {
            return this.nanoohmmetersLazy;
        }
        return this.nanoohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.NanoohmMeters);
    }
    /** */
    get MicroohmMeters() {
        if (this.microohmmetersLazy !== null) {
            return this.microohmmetersLazy;
        }
        return this.microohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.MicroohmMeters);
    }
    /** */
    get MilliohmMeters() {
        if (this.milliohmmetersLazy !== null) {
            return this.milliohmmetersLazy;
        }
        return this.milliohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.MilliohmMeters);
    }
    /** */
    get KiloohmMeters() {
        if (this.kiloohmmetersLazy !== null) {
            return this.kiloohmmetersLazy;
        }
        return this.kiloohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.KiloohmMeters);
    }
    /** */
    get MegaohmMeters() {
        if (this.megaohmmetersLazy !== null) {
            return this.megaohmmetersLazy;
        }
        return this.megaohmmetersLazy = this.convertFromBase(ElectricResistivityUnits.MegaohmMeters);
    }
    /** */
    get PicoohmsCentimeter() {
        if (this.picoohmscentimeterLazy !== null) {
            return this.picoohmscentimeterLazy;
        }
        return this.picoohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.PicoohmsCentimeter);
    }
    /** */
    get NanoohmsCentimeter() {
        if (this.nanoohmscentimeterLazy !== null) {
            return this.nanoohmscentimeterLazy;
        }
        return this.nanoohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.NanoohmsCentimeter);
    }
    /** */
    get MicroohmsCentimeter() {
        if (this.microohmscentimeterLazy !== null) {
            return this.microohmscentimeterLazy;
        }
        return this.microohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.MicroohmsCentimeter);
    }
    /** */
    get MilliohmsCentimeter() {
        if (this.milliohmscentimeterLazy !== null) {
            return this.milliohmscentimeterLazy;
        }
        return this.milliohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.MilliohmsCentimeter);
    }
    /** */
    get KiloohmsCentimeter() {
        if (this.kiloohmscentimeterLazy !== null) {
            return this.kiloohmscentimeterLazy;
        }
        return this.kiloohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.KiloohmsCentimeter);
    }
    /** */
    get MegaohmsCentimeter() {
        if (this.megaohmscentimeterLazy !== null) {
            return this.megaohmscentimeterLazy;
        }
        return this.megaohmscentimeterLazy = this.convertFromBase(ElectricResistivityUnits.MegaohmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a OhmMeters
     *
     * @param value The unit as OhmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromOhmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.OhmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a OhmsCentimeter
     *
     * @param value The unit as OhmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromOhmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.OhmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a PicoohmMeters
     *
     * @param value The unit as PicoohmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromPicoohmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.PicoohmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a NanoohmMeters
     *
     * @param value The unit as NanoohmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromNanoohmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.NanoohmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a MicroohmMeters
     *
     * @param value The unit as MicroohmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromMicroohmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.MicroohmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a MilliohmMeters
     *
     * @param value The unit as MilliohmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromMilliohmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.MilliohmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a KiloohmMeters
     *
     * @param value The unit as KiloohmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromKiloohmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.KiloohmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a MegaohmMeters
     *
     * @param value The unit as MegaohmMeters to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromMegaohmMeters(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.MegaohmMeters);
    }
    /**
     * Create a new ElectricResistivity instance from a PicoohmsCentimeter
     *
     * @param value The unit as PicoohmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromPicoohmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.PicoohmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a NanoohmsCentimeter
     *
     * @param value The unit as NanoohmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromNanoohmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.NanoohmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a MicroohmsCentimeter
     *
     * @param value The unit as MicroohmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromMicroohmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.MicroohmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a MilliohmsCentimeter
     *
     * @param value The unit as MilliohmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromMilliohmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.MilliohmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a KiloohmsCentimeter
     *
     * @param value The unit as KiloohmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromKiloohmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.KiloohmsCentimeter);
    }
    /**
     * Create a new ElectricResistivity instance from a MegaohmsCentimeter
     *
     * @param value The unit as MegaohmsCentimeter to create a new ElectricResistivity from.
     * @returns The new ElectricResistivity instance.
     */
    static FromMegaohmsCentimeter(value) {
        return new ElectricResistivity(value, ElectricResistivityUnits.MegaohmsCentimeter);
    }
    /**
     * Convert ElectricResistivity to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case ElectricResistivityUnits.OhmMeters: return this.OhmMeters;
            case ElectricResistivityUnits.OhmsCentimeter: return this.OhmsCentimeter;
            case ElectricResistivityUnits.PicoohmMeters: return this.PicoohmMeters;
            case ElectricResistivityUnits.NanoohmMeters: return this.NanoohmMeters;
            case ElectricResistivityUnits.MicroohmMeters: return this.MicroohmMeters;
            case ElectricResistivityUnits.MilliohmMeters: return this.MilliohmMeters;
            case ElectricResistivityUnits.KiloohmMeters: return this.KiloohmMeters;
            case ElectricResistivityUnits.MegaohmMeters: return this.MegaohmMeters;
            case ElectricResistivityUnits.PicoohmsCentimeter: return this.PicoohmsCentimeter;
            case ElectricResistivityUnits.NanoohmsCentimeter: return this.NanoohmsCentimeter;
            case ElectricResistivityUnits.MicroohmsCentimeter: return this.MicroohmsCentimeter;
            case ElectricResistivityUnits.MilliohmsCentimeter: return this.MilliohmsCentimeter;
            case ElectricResistivityUnits.KiloohmsCentimeter: return this.KiloohmsCentimeter;
            case ElectricResistivityUnits.MegaohmsCentimeter: return this.MegaohmsCentimeter;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case ElectricResistivityUnits.OhmMeters:
                return this.value;
            case ElectricResistivityUnits.OhmsCentimeter:
                return this.value * 100;
            case ElectricResistivityUnits.PicoohmMeters:
                return (this.value) / 1e-12;
            case ElectricResistivityUnits.NanoohmMeters:
                return (this.value) / 1e-9;
            case ElectricResistivityUnits.MicroohmMeters:
                return (this.value) / 0.000001;
            case ElectricResistivityUnits.MilliohmMeters:
                return (this.value) / 0.001;
            case ElectricResistivityUnits.KiloohmMeters:
                return (this.value) / 1000;
            case ElectricResistivityUnits.MegaohmMeters:
                return (this.value) / 1000000;
            case ElectricResistivityUnits.PicoohmsCentimeter:
                return (this.value * 100) / 1e-12;
            case ElectricResistivityUnits.NanoohmsCentimeter:
                return (this.value * 100) / 1e-9;
            case ElectricResistivityUnits.MicroohmsCentimeter:
                return (this.value * 100) / 0.000001;
            case ElectricResistivityUnits.MilliohmsCentimeter:
                return (this.value * 100) / 0.001;
            case ElectricResistivityUnits.KiloohmsCentimeter:
                return (this.value * 100) / 1000;
            case ElectricResistivityUnits.MegaohmsCentimeter:
                return (this.value * 100) / 1000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case ElectricResistivityUnits.OhmMeters:
                return value;
            case ElectricResistivityUnits.OhmsCentimeter:
                return value / 100;
            case ElectricResistivityUnits.PicoohmMeters:
                return (value) * 1e-12;
            case ElectricResistivityUnits.NanoohmMeters:
                return (value) * 1e-9;
            case ElectricResistivityUnits.MicroohmMeters:
                return (value) * 0.000001;
            case ElectricResistivityUnits.MilliohmMeters:
                return (value) * 0.001;
            case ElectricResistivityUnits.KiloohmMeters:
                return (value) * 1000;
            case ElectricResistivityUnits.MegaohmMeters:
                return (value) * 1000000;
            case ElectricResistivityUnits.PicoohmsCentimeter:
                return (value / 100) * 1e-12;
            case ElectricResistivityUnits.NanoohmsCentimeter:
                return (value / 100) * 1e-9;
            case ElectricResistivityUnits.MicroohmsCentimeter:
                return (value / 100) * 0.000001;
            case ElectricResistivityUnits.MilliohmsCentimeter:
                return (value / 100) * 0.001;
            case ElectricResistivityUnits.KiloohmsCentimeter:
                return (value / 100) * 1000;
            case ElectricResistivityUnits.MegaohmsCentimeter:
                return (value / 100) * 1000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the ElectricResistivity to string.
     * Note! the default format for ElectricResistivity is OhmMeters.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the ElectricResistivity.
     * @returns The string format of the ElectricResistivity.
     */
    toString(unit = ElectricResistivityUnits.OhmMeters) {
        switch (unit) {
            case ElectricResistivityUnits.OhmMeters:
                return this.OhmMeters + ` Ω·m`;
            case ElectricResistivityUnits.OhmsCentimeter:
                return this.OhmsCentimeter + ` Ω·cm`;
            case ElectricResistivityUnits.PicoohmMeters:
                return this.PicoohmMeters + ` pΩ·m`;
            case ElectricResistivityUnits.NanoohmMeters:
                return this.NanoohmMeters + ` nΩ·m`;
            case ElectricResistivityUnits.MicroohmMeters:
                return this.MicroohmMeters + ` μΩ·m`;
            case ElectricResistivityUnits.MilliohmMeters:
                return this.MilliohmMeters + ` mΩ·m`;
            case ElectricResistivityUnits.KiloohmMeters:
                return this.KiloohmMeters + ` kΩ·m`;
            case ElectricResistivityUnits.MegaohmMeters:
                return this.MegaohmMeters + ` MΩ·m`;
            case ElectricResistivityUnits.PicoohmsCentimeter:
                return this.PicoohmsCentimeter + ` pΩ·cm`;
            case ElectricResistivityUnits.NanoohmsCentimeter:
                return this.NanoohmsCentimeter + ` nΩ·cm`;
            case ElectricResistivityUnits.MicroohmsCentimeter:
                return this.MicroohmsCentimeter + ` μΩ·cm`;
            case ElectricResistivityUnits.MilliohmsCentimeter:
                return this.MilliohmsCentimeter + ` mΩ·cm`;
            case ElectricResistivityUnits.KiloohmsCentimeter:
                return this.KiloohmsCentimeter + ` kΩ·cm`;
            case ElectricResistivityUnits.MegaohmsCentimeter:
                return this.MegaohmsCentimeter + ` MΩ·cm`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get ElectricResistivity unit abbreviation.
     * Note! the default abbreviation for ElectricResistivity is OhmMeters.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the ElectricResistivity.
     * @returns The abbreviation string of ElectricResistivity.
     */
    getUnitAbbreviation(unitAbbreviation = ElectricResistivityUnits.OhmMeters) {
        switch (unitAbbreviation) {
            case ElectricResistivityUnits.OhmMeters:
                return `Ω·m`;
            case ElectricResistivityUnits.OhmsCentimeter:
                return `Ω·cm`;
            case ElectricResistivityUnits.PicoohmMeters:
                return `pΩ·m`;
            case ElectricResistivityUnits.NanoohmMeters:
                return `nΩ·m`;
            case ElectricResistivityUnits.MicroohmMeters:
                return `μΩ·m`;
            case ElectricResistivityUnits.MilliohmMeters:
                return `mΩ·m`;
            case ElectricResistivityUnits.KiloohmMeters:
                return `kΩ·m`;
            case ElectricResistivityUnits.MegaohmMeters:
                return `MΩ·m`;
            case ElectricResistivityUnits.PicoohmsCentimeter:
                return `pΩ·cm`;
            case ElectricResistivityUnits.NanoohmsCentimeter:
                return `nΩ·cm`;
            case ElectricResistivityUnits.MicroohmsCentimeter:
                return `μΩ·cm`;
            case ElectricResistivityUnits.MilliohmsCentimeter:
                return `mΩ·cm`;
            case ElectricResistivityUnits.KiloohmsCentimeter:
                return `kΩ·cm`;
            case ElectricResistivityUnits.MegaohmsCentimeter:
                return `MΩ·cm`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given ElectricResistivity are equals to the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns True if the given ElectricResistivity are equal to the current ElectricResistivity.
     */
    equals(electricResistivity) {
        return this.value === electricResistivity.BaseValue;
    }
    /**
     * Compare the given ElectricResistivity against the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns 0 if they are equal, -1 if the current ElectricResistivity is less then other, 1 if the current ElectricResistivity is greater then other.
     */
    compareTo(electricResistivity) {
        if (this.value > electricResistivity.BaseValue)
            return 1;
        if (this.value < electricResistivity.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given ElectricResistivity with the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns A new ElectricResistivity instance with the results.
     */
    add(electricResistivity) {
        return new ElectricResistivity(this.value + electricResistivity.BaseValue);
    }
    /**
     * Subtract the given ElectricResistivity with the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns A new ElectricResistivity instance with the results.
     */
    subtract(electricResistivity) {
        return new ElectricResistivity(this.value - electricResistivity.BaseValue);
    }
    /**
     * Multiply the given ElectricResistivity with the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns A new ElectricResistivity instance with the results.
     */
    multiply(electricResistivity) {
        return new ElectricResistivity(this.value * electricResistivity.BaseValue);
    }
    /**
     * Divide the given ElectricResistivity with the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns A new ElectricResistivity instance with the results.
     */
    divide(electricResistivity) {
        return new ElectricResistivity(this.value / electricResistivity.BaseValue);
    }
    /**
     * Modulo the given ElectricResistivity with the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns A new ElectricResistivity instance with the results.
     */
    modulo(electricResistivity) {
        return new ElectricResistivity(this.value % electricResistivity.BaseValue);
    }
    /**
     * Pow the given ElectricResistivity with the current ElectricResistivity.
     * @param electricResistivity The other ElectricResistivity.
     * @returns A new ElectricResistivity instance with the results.
     */
    pow(electricResistivity) {
        return new ElectricResistivity(this.value ** electricResistivity.BaseValue);
    }
}
exports.ElectricResistivity = ElectricResistivity;
