// unitsnet-js/dist/electricadmittance.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElectricAdmittance = exports.ElectricAdmittanceUnits = void 0;
/** ElectricAdmittanceUnits enumeration */
var ElectricAdmittanceUnits;
(function (ElectricAdmittanceUnits) {
    /** */
    ElectricAdmittanceUnits[ElectricAdmittanceUnits["Siemens"] = 0] = "Siemens";
    /** */
    ElectricAdmittanceUnits[ElectricAdmittanceUnits["Nanosiemens"] = 1] = "Nanosiemens";
    /** */
    ElectricAdmittanceUnits[ElectricAdmittanceUnits["Microsiemens"] = 2] = "Microsiemens";
    /** */
    ElectricAdmittanceUnits[ElectricAdmittanceUnits["Millisiemens"] = 3] = "Millisiemens";
})(ElectricAdmittanceUnits = exports.ElectricAdmittanceUnits || (exports.ElectricAdmittanceUnits = {}));
/** Electric admittance is a measure of how easily a circuit or device will allow a current to flow. It is defined as the inverse of impedance. The SI unit of admittance is the siemens (symbol S). */
class ElectricAdmittance {
    /**
     * Create a new ElectricAdmittance.
     * @param value The value.
     * @param fromUnit The ‘ElectricAdmittance’ unit to create from.
     * The default unit is Siemens
     */
    constructor(value, fromUnit = ElectricAdmittanceUnits.Siemens) {
        this.siemensLazy = null;
        this.nanosiemensLazy = null;
        this.microsiemensLazy = null;
        this.millisiemensLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of ElectricAdmittance is Siemens.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Siemens() {
        if (this.siemensLazy !== null) {
            return this.siemensLazy;
        }
        return this.siemensLazy = this.convertFromBase(ElectricAdmittanceUnits.Siemens);
    }
    /** */
    get Nanosiemens() {
        if (this.nanosiemensLazy !== null) {
            return this.nanosiemensLazy;
        }
        return this.nanosiemensLazy = this.convertFromBase(ElectricAdmittanceUnits.Nanosiemens);
    }
    /** */
    get Microsiemens() {
        if (this.microsiemensLazy !== null) {
            return this.microsiemensLazy;
        }
        return this.microsiemensLazy = this.convertFromBase(ElectricAdmittanceUnits.Microsiemens);
    }
    /** */
    get Millisiemens() {
        if (this.millisiemensLazy !== null) {
            return this.millisiemensLazy;
        }
        return this.millisiemensLazy = this.convertFromBase(ElectricAdmittanceUnits.Millisiemens);
    }
    /**
     * Create a new ElectricAdmittance instance from a Siemens
     *
     * @param value The unit as Siemens to create a new ElectricAdmittance from.
     * @returns The new ElectricAdmittance instance.
     */
    static FromSiemens(value) {
        return new ElectricAdmittance(value, ElectricAdmittanceUnits.Siemens);
    }
    /**
     * Create a new ElectricAdmittance instance from a Nanosiemens
     *
     * @param value The unit as Nanosiemens to create a new ElectricAdmittance from.
     * @returns The new ElectricAdmittance instance.
     */
    static FromNanosiemens(value) {
        return new ElectricAdmittance(value, ElectricAdmittanceUnits.Nanosiemens);
    }
    /**
     * Create a new ElectricAdmittance instance from a Microsiemens
     *
     * @param value The unit as Microsiemens to create a new ElectricAdmittance from.
     * @returns The new ElectricAdmittance instance.
     */
    static FromMicrosiemens(value) {
        return new ElectricAdmittance(value, ElectricAdmittanceUnits.Microsiemens);
    }
    /**
     * Create a new ElectricAdmittance instance from a Millisiemens
     *
     * @param value The unit as Millisiemens to create a new ElectricAdmittance from.
     * @returns The new ElectricAdmittance instance.
     */
    static FromMillisiemens(value) {
        return new ElectricAdmittance(value, ElectricAdmittanceUnits.Millisiemens);
    }
    /**
     * Convert ElectricAdmittance to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case ElectricAdmittanceUnits.Siemens: return this.Siemens;
            case ElectricAdmittanceUnits.Nanosiemens: return this.Nanosiemens;
            case ElectricAdmittanceUnits.Microsiemens: return this.Microsiemens;
            case ElectricAdmittanceUnits.Millisiemens: return this.Millisiemens;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case ElectricAdmittanceUnits.Siemens:
                return this.value;
            case ElectricAdmittanceUnits.Nanosiemens:
                return (this.value) / 1e-9;
            case ElectricAdmittanceUnits.Microsiemens:
                return (this.value) / 0.000001;
            case ElectricAdmittanceUnits.Millisiemens:
                return (this.value) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case ElectricAdmittanceUnits.Siemens:
                return value;
            case ElectricAdmittanceUnits.Nanosiemens:
                return (value) * 1e-9;
            case ElectricAdmittanceUnits.Microsiemens:
                return (value) * 0.000001;
            case ElectricAdmittanceUnits.Millisiemens:
                return (value) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the ElectricAdmittance to string.
     * Note! the default format for ElectricAdmittance is Siemens.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the ElectricAdmittance.
     * @returns The string format of the ElectricAdmittance.
     */
    toString(unit = ElectricAdmittanceUnits.Siemens) {
        switch (unit) {
            case ElectricAdmittanceUnits.Siemens:
                return this.Siemens + ` S`;
            case ElectricAdmittanceUnits.Nanosiemens:
                return this.Nanosiemens + ` nS`;
            case ElectricAdmittanceUnits.Microsiemens:
                return this.Microsiemens + ` μS`;
            case ElectricAdmittanceUnits.Millisiemens:
                return this.Millisiemens + ` mS`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get ElectricAdmittance unit abbreviation.
     * Note! the default abbreviation for ElectricAdmittance is Siemens.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the ElectricAdmittance.
     * @returns The abbreviation string of ElectricAdmittance.
     */
    getUnitAbbreviation(unitAbbreviation = ElectricAdmittanceUnits.Siemens) {
        switch (unitAbbreviation) {
            case ElectricAdmittanceUnits.Siemens:
                return `S`;
            case ElectricAdmittanceUnits.Nanosiemens:
                return `nS`;
            case ElectricAdmittanceUnits.Microsiemens:
                return `μS`;
            case ElectricAdmittanceUnits.Millisiemens:
                return `mS`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given ElectricAdmittance are equals to the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns True if the given ElectricAdmittance are equal to the current ElectricAdmittance.
     */
    equals(electricAdmittance) {
        return this.value === electricAdmittance.BaseValue;
    }
    /**
     * Compare the given ElectricAdmittance against the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns 0 if they are equal, -1 if the current ElectricAdmittance is less then other, 1 if the current ElectricAdmittance is greater then other.
     */
    compareTo(electricAdmittance) {
        if (this.value > electricAdmittance.BaseValue)
            return 1;
        if (this.value < electricAdmittance.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given ElectricAdmittance with the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns A new ElectricAdmittance instance with the results.
     */
    add(electricAdmittance) {
        return new ElectricAdmittance(this.value + electricAdmittance.BaseValue);
    }
    /**
     * Subtract the given ElectricAdmittance with the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns A new ElectricAdmittance instance with the results.
     */
    subtract(electricAdmittance) {
        return new ElectricAdmittance(this.value - electricAdmittance.BaseValue);
    }
    /**
     * Multiply the given ElectricAdmittance with the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns A new ElectricAdmittance instance with the results.
     */
    multiply(electricAdmittance) {
        return new ElectricAdmittance(this.value * electricAdmittance.BaseValue);
    }
    /**
     * Divide the given ElectricAdmittance with the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns A new ElectricAdmittance instance with the results.
     */
    divide(electricAdmittance) {
        return new ElectricAdmittance(this.value / electricAdmittance.BaseValue);
    }
    /**
     * Modulo the given ElectricAdmittance with the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns A new ElectricAdmittance instance with the results.
     */
    modulo(electricAdmittance) {
        return new ElectricAdmittance(this.value % electricAdmittance.BaseValue);
    }
    /**
     * Pow the given ElectricAdmittance with the current ElectricAdmittance.
     * @param electricAdmittance The other ElectricAdmittance.
     * @returns A new ElectricAdmittance instance with the results.
     */
    pow(electricAdmittance) {
        return new ElectricAdmittance(this.value ** electricAdmittance.BaseValue);
    }
}
exports.ElectricAdmittance = ElectricAdmittance;
