// unitsnet-js/dist/electricresistance.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElectricResistance = exports.ElectricResistanceUnits = void 0;
/** ElectricResistanceUnits enumeration */
var ElectricResistanceUnits;
(function (ElectricResistanceUnits) {
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Ohms"] = 0] = "Ohms";
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Microohms"] = 1] = "Microohms";
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Milliohms"] = 2] = "Milliohms";
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Kiloohms"] = 3] = "Kiloohms";
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Megaohms"] = 4] = "Megaohms";
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Gigaohms"] = 5] = "Gigaohms";
    /** */
    ElectricResistanceUnits[ElectricResistanceUnits["Teraohms"] = 6] = "Teraohms";
})(ElectricResistanceUnits = exports.ElectricResistanceUnits || (exports.ElectricResistanceUnits = {}));
/** The electrical resistance of an electrical conductor is the opposition to the passage of an electric current through that conductor. */
class ElectricResistance {
    /**
     * Create a new ElectricResistance.
     * @param value The value.
     * @param fromUnit The ‘ElectricResistance’ unit to create from.
     * The default unit is Ohms
     */
    constructor(value, fromUnit = ElectricResistanceUnits.Ohms) {
        this.ohmsLazy = null;
        this.microohmsLazy = null;
        this.milliohmsLazy = null;
        this.kiloohmsLazy = null;
        this.megaohmsLazy = null;
        this.gigaohmsLazy = null;
        this.teraohmsLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of ElectricResistance is Ohms.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Ohms() {
        if (this.ohmsLazy !== null) {
            return this.ohmsLazy;
        }
        return this.ohmsLazy = this.convertFromBase(ElectricResistanceUnits.Ohms);
    }
    /** */
    get Microohms() {
        if (this.microohmsLazy !== null) {
            return this.microohmsLazy;
        }
        return this.microohmsLazy = this.convertFromBase(ElectricResistanceUnits.Microohms);
    }
    /** */
    get Milliohms() {
        if (this.milliohmsLazy !== null) {
            return this.milliohmsLazy;
        }
        return this.milliohmsLazy = this.convertFromBase(ElectricResistanceUnits.Milliohms);
    }
    /** */
    get Kiloohms() {
        if (this.kiloohmsLazy !== null) {
            return this.kiloohmsLazy;
        }
        return this.kiloohmsLazy = this.convertFromBase(ElectricResistanceUnits.Kiloohms);
    }
    /** */
    get Megaohms() {
        if (this.megaohmsLazy !== null) {
            return this.megaohmsLazy;
        }
        return this.megaohmsLazy = this.convertFromBase(ElectricResistanceUnits.Megaohms);
    }
    /** */
    get Gigaohms() {
        if (this.gigaohmsLazy !== null) {
            return this.gigaohmsLazy;
        }
        return this.gigaohmsLazy = this.convertFromBase(ElectricResistanceUnits.Gigaohms);
    }
    /** */
    get Teraohms() {
        if (this.teraohmsLazy !== null) {
            return this.teraohmsLazy;
        }
        return this.teraohmsLazy = this.convertFromBase(ElectricResistanceUnits.Teraohms);
    }
    /**
     * Create a new ElectricResistance instance from a Ohms
     *
     * @param value The unit as Ohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromOhms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Ohms);
    }
    /**
     * Create a new ElectricResistance instance from a Microohms
     *
     * @param value The unit as Microohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromMicroohms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Microohms);
    }
    /**
     * Create a new ElectricResistance instance from a Milliohms
     *
     * @param value The unit as Milliohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromMilliohms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Milliohms);
    }
    /**
     * Create a new ElectricResistance instance from a Kiloohms
     *
     * @param value The unit as Kiloohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromKiloohms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Kiloohms);
    }
    /**
     * Create a new ElectricResistance instance from a Megaohms
     *
     * @param value The unit as Megaohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromMegaohms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Megaohms);
    }
    /**
     * Create a new ElectricResistance instance from a Gigaohms
     *
     * @param value The unit as Gigaohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromGigaohms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Gigaohms);
    }
    /**
     * Create a new ElectricResistance instance from a Teraohms
     *
     * @param value The unit as Teraohms to create a new ElectricResistance from.
     * @returns The new ElectricResistance instance.
     */
    static FromTeraohms(value) {
        return new ElectricResistance(value, ElectricResistanceUnits.Teraohms);
    }
    /**
     * Convert ElectricResistance to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case ElectricResistanceUnits.Ohms: return this.Ohms;
            case ElectricResistanceUnits.Microohms: return this.Microohms;
            case ElectricResistanceUnits.Milliohms: return this.Milliohms;
            case ElectricResistanceUnits.Kiloohms: return this.Kiloohms;
            case ElectricResistanceUnits.Megaohms: return this.Megaohms;
            case ElectricResistanceUnits.Gigaohms: return this.Gigaohms;
            case ElectricResistanceUnits.Teraohms: return this.Teraohms;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case ElectricResistanceUnits.Ohms:
                return this.value;
            case ElectricResistanceUnits.Microohms:
                return (this.value) / 0.000001;
            case ElectricResistanceUnits.Milliohms:
                return (this.value) / 0.001;
            case ElectricResistanceUnits.Kiloohms:
                return (this.value) / 1000;
            case ElectricResistanceUnits.Megaohms:
                return (this.value) / 1000000;
            case ElectricResistanceUnits.Gigaohms:
                return (this.value) / 1000000000;
            case ElectricResistanceUnits.Teraohms:
                return (this.value) / 1000000000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case ElectricResistanceUnits.Ohms:
                return value;
            case ElectricResistanceUnits.Microohms:
                return (value) * 0.000001;
            case ElectricResistanceUnits.Milliohms:
                return (value) * 0.001;
            case ElectricResistanceUnits.Kiloohms:
                return (value) * 1000;
            case ElectricResistanceUnits.Megaohms:
                return (value) * 1000000;
            case ElectricResistanceUnits.Gigaohms:
                return (value) * 1000000000;
            case ElectricResistanceUnits.Teraohms:
                return (value) * 1000000000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the ElectricResistance to string.
     * Note! the default format for ElectricResistance is Ohms.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the ElectricResistance.
     * @returns The string format of the ElectricResistance.
     */
    toString(unit = ElectricResistanceUnits.Ohms) {
        switch (unit) {
            case ElectricResistanceUnits.Ohms:
                return this.Ohms + ` Ω`;
            case ElectricResistanceUnits.Microohms:
                return this.Microohms + ` μΩ`;
            case ElectricResistanceUnits.Milliohms:
                return this.Milliohms + ` mΩ`;
            case ElectricResistanceUnits.Kiloohms:
                return this.Kiloohms + ` kΩ`;
            case ElectricResistanceUnits.Megaohms:
                return this.Megaohms + ` MΩ`;
            case ElectricResistanceUnits.Gigaohms:
                return this.Gigaohms + ` GΩ`;
            case ElectricResistanceUnits.Teraohms:
                return this.Teraohms + ` TΩ`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get ElectricResistance unit abbreviation.
     * Note! the default abbreviation for ElectricResistance is Ohms.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the ElectricResistance.
     * @returns The abbreviation string of ElectricResistance.
     */
    getUnitAbbreviation(unitAbbreviation = ElectricResistanceUnits.Ohms) {
        switch (unitAbbreviation) {
            case ElectricResistanceUnits.Ohms:
                return `Ω`;
            case ElectricResistanceUnits.Microohms:
                return `μΩ`;
            case ElectricResistanceUnits.Milliohms:
                return `mΩ`;
            case ElectricResistanceUnits.Kiloohms:
                return `kΩ`;
            case ElectricResistanceUnits.Megaohms:
                return `MΩ`;
            case ElectricResistanceUnits.Gigaohms:
                return `GΩ`;
            case ElectricResistanceUnits.Teraohms:
                return `TΩ`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given ElectricResistance are equals to the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns True if the given ElectricResistance are equal to the current ElectricResistance.
     */
    equals(electricResistance) {
        return this.value === electricResistance.BaseValue;
    }
    /**
     * Compare the given ElectricResistance against the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns 0 if they are equal, -1 if the current ElectricResistance is less then other, 1 if the current ElectricResistance is greater then other.
     */
    compareTo(electricResistance) {
        if (this.value > electricResistance.BaseValue)
            return 1;
        if (this.value < electricResistance.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given ElectricResistance with the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns A new ElectricResistance instance with the results.
     */
    add(electricResistance) {
        return new ElectricResistance(this.value + electricResistance.BaseValue);
    }
    /**
     * Subtract the given ElectricResistance with the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns A new ElectricResistance instance with the results.
     */
    subtract(electricResistance) {
        return new ElectricResistance(this.value - electricResistance.BaseValue);
    }
    /**
     * Multiply the given ElectricResistance with the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns A new ElectricResistance instance with the results.
     */
    multiply(electricResistance) {
        return new ElectricResistance(this.value * electricResistance.BaseValue);
    }
    /**
     * Divide the given ElectricResistance with the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns A new ElectricResistance instance with the results.
     */
    divide(electricResistance) {
        return new ElectricResistance(this.value / electricResistance.BaseValue);
    }
    /**
     * Modulo the given ElectricResistance with the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns A new ElectricResistance instance with the results.
     */
    modulo(electricResistance) {
        return new ElectricResistance(this.value % electricResistance.BaseValue);
    }
    /**
     * Pow the given ElectricResistance with the current ElectricResistance.
     * @param electricResistance The other ElectricResistance.
     * @returns A new ElectricResistance instance with the results.
     */
    pow(electricResistance) {
        return new ElectricResistance(this.value ** electricResistance.BaseValue);
    }
}
exports.ElectricResistance = ElectricResistance;
