// unitsnet-js/dist/electricconductance.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ElectricConductance = exports.ElectricConductanceUnits = void 0;
/** ElectricConductanceUnits enumeration */
var ElectricConductanceUnits;
(function (ElectricConductanceUnits) {
    /** */
    ElectricConductanceUnits[ElectricConductanceUnits["Siemens"] = 0] = "Siemens";
    /** */
    ElectricConductanceUnits[ElectricConductanceUnits["Nanosiemens"] = 1] = "Nanosiemens";
    /** */
    ElectricConductanceUnits[ElectricConductanceUnits["Microsiemens"] = 2] = "Microsiemens";
    /** */
    ElectricConductanceUnits[ElectricConductanceUnits["Millisiemens"] = 3] = "Millisiemens";
    /** */
    ElectricConductanceUnits[ElectricConductanceUnits["Kilosiemens"] = 4] = "Kilosiemens";
})(ElectricConductanceUnits = exports.ElectricConductanceUnits || (exports.ElectricConductanceUnits = {}));
/** The electrical conductance of an electrical conductor is a measure of the easeness to pass an electric current through that conductor. */
class ElectricConductance {
    /**
     * Create a new ElectricConductance.
     * @param value The value.
     * @param fromUnit The ‘ElectricConductance’ unit to create from.
     * The default unit is Siemens
     */
    constructor(value, fromUnit = ElectricConductanceUnits.Siemens) {
        this.siemensLazy = null;
        this.nanosiemensLazy = null;
        this.microsiemensLazy = null;
        this.millisiemensLazy = null;
        this.kilosiemensLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of ElectricConductance is Siemens.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Siemens() {
        if (this.siemensLazy !== null) {
            return this.siemensLazy;
        }
        return this.siemensLazy = this.convertFromBase(ElectricConductanceUnits.Siemens);
    }
    /** */
    get Nanosiemens() {
        if (this.nanosiemensLazy !== null) {
            return this.nanosiemensLazy;
        }
        return this.nanosiemensLazy = this.convertFromBase(ElectricConductanceUnits.Nanosiemens);
    }
    /** */
    get Microsiemens() {
        if (this.microsiemensLazy !== null) {
            return this.microsiemensLazy;
        }
        return this.microsiemensLazy = this.convertFromBase(ElectricConductanceUnits.Microsiemens);
    }
    /** */
    get Millisiemens() {
        if (this.millisiemensLazy !== null) {
            return this.millisiemensLazy;
        }
        return this.millisiemensLazy = this.convertFromBase(ElectricConductanceUnits.Millisiemens);
    }
    /** */
    get Kilosiemens() {
        if (this.kilosiemensLazy !== null) {
            return this.kilosiemensLazy;
        }
        return this.kilosiemensLazy = this.convertFromBase(ElectricConductanceUnits.Kilosiemens);
    }
    /**
     * Create a new ElectricConductance instance from a Siemens
     *
     * @param value The unit as Siemens to create a new ElectricConductance from.
     * @returns The new ElectricConductance instance.
     */
    static FromSiemens(value) {
        return new ElectricConductance(value, ElectricConductanceUnits.Siemens);
    }
    /**
     * Create a new ElectricConductance instance from a Nanosiemens
     *
     * @param value The unit as Nanosiemens to create a new ElectricConductance from.
     * @returns The new ElectricConductance instance.
     */
    static FromNanosiemens(value) {
        return new ElectricConductance(value, ElectricConductanceUnits.Nanosiemens);
    }
    /**
     * Create a new ElectricConductance instance from a Microsiemens
     *
     * @param value The unit as Microsiemens to create a new ElectricConductance from.
     * @returns The new ElectricConductance instance.
     */
    static FromMicrosiemens(value) {
        return new ElectricConductance(value, ElectricConductanceUnits.Microsiemens);
    }
    /**
     * Create a new ElectricConductance instance from a Millisiemens
     *
     * @param value The unit as Millisiemens to create a new ElectricConductance from.
     * @returns The new ElectricConductance instance.
     */
    static FromMillisiemens(value) {
        return new ElectricConductance(value, ElectricConductanceUnits.Millisiemens);
    }
    /**
     * Create a new ElectricConductance instance from a Kilosiemens
     *
     * @param value The unit as Kilosiemens to create a new ElectricConductance from.
     * @returns The new ElectricConductance instance.
     */
    static FromKilosiemens(value) {
        return new ElectricConductance(value, ElectricConductanceUnits.Kilosiemens);
    }
    /**
     * Convert ElectricConductance to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case ElectricConductanceUnits.Siemens: return this.Siemens;
            case ElectricConductanceUnits.Nanosiemens: return this.Nanosiemens;
            case ElectricConductanceUnits.Microsiemens: return this.Microsiemens;
            case ElectricConductanceUnits.Millisiemens: return this.Millisiemens;
            case ElectricConductanceUnits.Kilosiemens: return this.Kilosiemens;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case ElectricConductanceUnits.Siemens:
                return this.value;
            case ElectricConductanceUnits.Nanosiemens:
                return (this.value) / 1e-9;
            case ElectricConductanceUnits.Microsiemens:
                return (this.value) / 0.000001;
            case ElectricConductanceUnits.Millisiemens:
                return (this.value) / 0.001;
            case ElectricConductanceUnits.Kilosiemens:
                return (this.value) / 1000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case ElectricConductanceUnits.Siemens:
                return value;
            case ElectricConductanceUnits.Nanosiemens:
                return (value) * 1e-9;
            case ElectricConductanceUnits.Microsiemens:
                return (value) * 0.000001;
            case ElectricConductanceUnits.Millisiemens:
                return (value) * 0.001;
            case ElectricConductanceUnits.Kilosiemens:
                return (value) * 1000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the ElectricConductance to string.
     * Note! the default format for ElectricConductance is Siemens.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the ElectricConductance.
     * @returns The string format of the ElectricConductance.
     */
    toString(unit = ElectricConductanceUnits.Siemens) {
        switch (unit) {
            case ElectricConductanceUnits.Siemens:
                return this.Siemens + ` S`;
            case ElectricConductanceUnits.Nanosiemens:
                return this.Nanosiemens + ` nS`;
            case ElectricConductanceUnits.Microsiemens:
                return this.Microsiemens + ` μS`;
            case ElectricConductanceUnits.Millisiemens:
                return this.Millisiemens + ` mS`;
            case ElectricConductanceUnits.Kilosiemens:
                return this.Kilosiemens + ` kS`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get ElectricConductance unit abbreviation.
     * Note! the default abbreviation for ElectricConductance is Siemens.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the ElectricConductance.
     * @returns The abbreviation string of ElectricConductance.
     */
    getUnitAbbreviation(unitAbbreviation = ElectricConductanceUnits.Siemens) {
        switch (unitAbbreviation) {
            case ElectricConductanceUnits.Siemens:
                return `S`;
            case ElectricConductanceUnits.Nanosiemens:
                return `nS`;
            case ElectricConductanceUnits.Microsiemens:
                return `μS`;
            case ElectricConductanceUnits.Millisiemens:
                return `mS`;
            case ElectricConductanceUnits.Kilosiemens:
                return `kS`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given ElectricConductance are equals to the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns True if the given ElectricConductance are equal to the current ElectricConductance.
     */
    equals(electricConductance) {
        return this.value === electricConductance.BaseValue;
    }
    /**
     * Compare the given ElectricConductance against the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns 0 if they are equal, -1 if the current ElectricConductance is less then other, 1 if the current ElectricConductance is greater then other.
     */
    compareTo(electricConductance) {
        if (this.value > electricConductance.BaseValue)
            return 1;
        if (this.value < electricConductance.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given ElectricConductance with the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns A new ElectricConductance instance with the results.
     */
    add(electricConductance) {
        return new ElectricConductance(this.value + electricConductance.BaseValue);
    }
    /**
     * Subtract the given ElectricConductance with the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns A new ElectricConductance instance with the results.
     */
    subtract(electricConductance) {
        return new ElectricConductance(this.value - electricConductance.BaseValue);
    }
    /**
     * Multiply the given ElectricConductance with the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns A new ElectricConductance instance with the results.
     */
    multiply(electricConductance) {
        return new ElectricConductance(this.value * electricConductance.BaseValue);
    }
    /**
     * Divide the given ElectricConductance with the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns A new ElectricConductance instance with the results.
     */
    divide(electricConductance) {
        return new ElectricConductance(this.value / electricConductance.BaseValue);
    }
    /**
     * Modulo the given ElectricConductance with the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns A new ElectricConductance instance with the results.
     */
    modulo(electricConductance) {
        return new ElectricConductance(this.value % electricConductance.BaseValue);
    }
    /**
     * Pow the given ElectricConductance with the current ElectricConductance.
     * @param electricConductance The other ElectricConductance.
     * @returns A new ElectricConductance instance with the results.
     */
    pow(electricConductance) {
        return new ElectricConductance(this.value ** electricConductance.BaseValue);
    }
}
exports.ElectricConductance = ElectricConductance;
