// unitsnet-js/dist/energydensity.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnergyDensity = exports.EnergyDensityUnits = void 0;
/** EnergyDensityUnits enumeration */
var EnergyDensityUnits;
(function (EnergyDensityUnits) {
    /** */
    EnergyDensityUnits[EnergyDensityUnits["JoulesPerCubicMeter"] = 0] = "JoulesPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["WattHoursPerCubicMeter"] = 1] = "WattHoursPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["KilojoulesPerCubicMeter"] = 2] = "KilojoulesPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["MegajoulesPerCubicMeter"] = 3] = "MegajoulesPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["GigajoulesPerCubicMeter"] = 4] = "GigajoulesPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["TerajoulesPerCubicMeter"] = 5] = "TerajoulesPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["PetajoulesPerCubicMeter"] = 6] = "PetajoulesPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["KilowattHoursPerCubicMeter"] = 7] = "KilowattHoursPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["MegawattHoursPerCubicMeter"] = 8] = "MegawattHoursPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["GigawattHoursPerCubicMeter"] = 9] = "GigawattHoursPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["TerawattHoursPerCubicMeter"] = 10] = "TerawattHoursPerCubicMeter";
    /** */
    EnergyDensityUnits[EnergyDensityUnits["PetawattHoursPerCubicMeter"] = 11] = "PetawattHoursPerCubicMeter";
})(EnergyDensityUnits = exports.EnergyDensityUnits || (exports.EnergyDensityUnits = {}));
/** Energy density is the amount of energy stored in a substance per unit volume and is measured in J/m³. It is sometimes confused with energy per unit mass which is properly called specific energy. */
class EnergyDensity {
    /**
     * Create a new EnergyDensity.
     * @param value The value.
     * @param fromUnit The ‘EnergyDensity’ unit to create from.
     * The default unit is JoulesPerCubicMeter
     */
    constructor(value, fromUnit = EnergyDensityUnits.JoulesPerCubicMeter) {
        this.joulespercubicmeterLazy = null;
        this.watthourspercubicmeterLazy = null;
        this.kilojoulespercubicmeterLazy = null;
        this.megajoulespercubicmeterLazy = null;
        this.gigajoulespercubicmeterLazy = null;
        this.terajoulespercubicmeterLazy = null;
        this.petajoulespercubicmeterLazy = null;
        this.kilowatthourspercubicmeterLazy = null;
        this.megawatthourspercubicmeterLazy = null;
        this.gigawatthourspercubicmeterLazy = null;
        this.terawatthourspercubicmeterLazy = null;
        this.petawatthourspercubicmeterLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of EnergyDensity is JoulesPerCubicMeter.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get JoulesPerCubicMeter() {
        if (this.joulespercubicmeterLazy !== null) {
            return this.joulespercubicmeterLazy;
        }
        return this.joulespercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.JoulesPerCubicMeter);
    }
    /** */
    get WattHoursPerCubicMeter() {
        if (this.watthourspercubicmeterLazy !== null) {
            return this.watthourspercubicmeterLazy;
        }
        return this.watthourspercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.WattHoursPerCubicMeter);
    }
    /** */
    get KilojoulesPerCubicMeter() {
        if (this.kilojoulespercubicmeterLazy !== null) {
            return this.kilojoulespercubicmeterLazy;
        }
        return this.kilojoulespercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.KilojoulesPerCubicMeter);
    }
    /** */
    get MegajoulesPerCubicMeter() {
        if (this.megajoulespercubicmeterLazy !== null) {
            return this.megajoulespercubicmeterLazy;
        }
        return this.megajoulespercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.MegajoulesPerCubicMeter);
    }
    /** */
    get GigajoulesPerCubicMeter() {
        if (this.gigajoulespercubicmeterLazy !== null) {
            return this.gigajoulespercubicmeterLazy;
        }
        return this.gigajoulespercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.GigajoulesPerCubicMeter);
    }
    /** */
    get TerajoulesPerCubicMeter() {
        if (this.terajoulespercubicmeterLazy !== null) {
            return this.terajoulespercubicmeterLazy;
        }
        return this.terajoulespercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.TerajoulesPerCubicMeter);
    }
    /** */
    get PetajoulesPerCubicMeter() {
        if (this.petajoulespercubicmeterLazy !== null) {
            return this.petajoulespercubicmeterLazy;
        }
        return this.petajoulespercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.PetajoulesPerCubicMeter);
    }
    /** */
    get KilowattHoursPerCubicMeter() {
        if (this.kilowatthourspercubicmeterLazy !== null) {
            return this.kilowatthourspercubicmeterLazy;
        }
        return this.kilowatthourspercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.KilowattHoursPerCubicMeter);
    }
    /** */
    get MegawattHoursPerCubicMeter() {
        if (this.megawatthourspercubicmeterLazy !== null) {
            return this.megawatthourspercubicmeterLazy;
        }
        return this.megawatthourspercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.MegawattHoursPerCubicMeter);
    }
    /** */
    get GigawattHoursPerCubicMeter() {
        if (this.gigawatthourspercubicmeterLazy !== null) {
            return this.gigawatthourspercubicmeterLazy;
        }
        return this.gigawatthourspercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.GigawattHoursPerCubicMeter);
    }
    /** */
    get TerawattHoursPerCubicMeter() {
        if (this.terawatthourspercubicmeterLazy !== null) {
            return this.terawatthourspercubicmeterLazy;
        }
        return this.terawatthourspercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.TerawattHoursPerCubicMeter);
    }
    /** */
    get PetawattHoursPerCubicMeter() {
        if (this.petawatthourspercubicmeterLazy !== null) {
            return this.petawatthourspercubicmeterLazy;
        }
        return this.petawatthourspercubicmeterLazy = this.convertFromBase(EnergyDensityUnits.PetawattHoursPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a JoulesPerCubicMeter
     *
     * @param value The unit as JoulesPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromJoulesPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.JoulesPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a WattHoursPerCubicMeter
     *
     * @param value The unit as WattHoursPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromWattHoursPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.WattHoursPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a KilojoulesPerCubicMeter
     *
     * @param value The unit as KilojoulesPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromKilojoulesPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.KilojoulesPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a MegajoulesPerCubicMeter
     *
     * @param value The unit as MegajoulesPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromMegajoulesPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.MegajoulesPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a GigajoulesPerCubicMeter
     *
     * @param value The unit as GigajoulesPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromGigajoulesPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.GigajoulesPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a TerajoulesPerCubicMeter
     *
     * @param value The unit as TerajoulesPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromTerajoulesPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.TerajoulesPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a PetajoulesPerCubicMeter
     *
     * @param value The unit as PetajoulesPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromPetajoulesPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.PetajoulesPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a KilowattHoursPerCubicMeter
     *
     * @param value The unit as KilowattHoursPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromKilowattHoursPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.KilowattHoursPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a MegawattHoursPerCubicMeter
     *
     * @param value The unit as MegawattHoursPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromMegawattHoursPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.MegawattHoursPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a GigawattHoursPerCubicMeter
     *
     * @param value The unit as GigawattHoursPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromGigawattHoursPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.GigawattHoursPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a TerawattHoursPerCubicMeter
     *
     * @param value The unit as TerawattHoursPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromTerawattHoursPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.TerawattHoursPerCubicMeter);
    }
    /**
     * Create a new EnergyDensity instance from a PetawattHoursPerCubicMeter
     *
     * @param value The unit as PetawattHoursPerCubicMeter to create a new EnergyDensity from.
     * @returns The new EnergyDensity instance.
     */
    static FromPetawattHoursPerCubicMeter(value) {
        return new EnergyDensity(value, EnergyDensityUnits.PetawattHoursPerCubicMeter);
    }
    /**
     * Convert EnergyDensity to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case EnergyDensityUnits.JoulesPerCubicMeter: return this.JoulesPerCubicMeter;
            case EnergyDensityUnits.WattHoursPerCubicMeter: return this.WattHoursPerCubicMeter;
            case EnergyDensityUnits.KilojoulesPerCubicMeter: return this.KilojoulesPerCubicMeter;
            case EnergyDensityUnits.MegajoulesPerCubicMeter: return this.MegajoulesPerCubicMeter;
            case EnergyDensityUnits.GigajoulesPerCubicMeter: return this.GigajoulesPerCubicMeter;
            case EnergyDensityUnits.TerajoulesPerCubicMeter: return this.TerajoulesPerCubicMeter;
            case EnergyDensityUnits.PetajoulesPerCubicMeter: return this.PetajoulesPerCubicMeter;
            case EnergyDensityUnits.KilowattHoursPerCubicMeter: return this.KilowattHoursPerCubicMeter;
            case EnergyDensityUnits.MegawattHoursPerCubicMeter: return this.MegawattHoursPerCubicMeter;
            case EnergyDensityUnits.GigawattHoursPerCubicMeter: return this.GigawattHoursPerCubicMeter;
            case EnergyDensityUnits.TerawattHoursPerCubicMeter: return this.TerawattHoursPerCubicMeter;
            case EnergyDensityUnits.PetawattHoursPerCubicMeter: return this.PetawattHoursPerCubicMeter;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case EnergyDensityUnits.JoulesPerCubicMeter:
                return this.value;
            case EnergyDensityUnits.WattHoursPerCubicMeter:
                return this.value / 3.6e+3;
            case EnergyDensityUnits.KilojoulesPerCubicMeter:
                return (this.value) / 1000;
            case EnergyDensityUnits.MegajoulesPerCubicMeter:
                return (this.value) / 1000000;
            case EnergyDensityUnits.GigajoulesPerCubicMeter:
                return (this.value) / 1000000000;
            case EnergyDensityUnits.TerajoulesPerCubicMeter:
                return (this.value) / 1000000000000;
            case EnergyDensityUnits.PetajoulesPerCubicMeter:
                return (this.value) / 1000000000000000;
            case EnergyDensityUnits.KilowattHoursPerCubicMeter:
                return (this.value / 3.6e+3) / 1000;
            case EnergyDensityUnits.MegawattHoursPerCubicMeter:
                return (this.value / 3.6e+3) / 1000000;
            case EnergyDensityUnits.GigawattHoursPerCubicMeter:
                return (this.value / 3.6e+3) / 1000000000;
            case EnergyDensityUnits.TerawattHoursPerCubicMeter:
                return (this.value / 3.6e+3) / 1000000000000;
            case EnergyDensityUnits.PetawattHoursPerCubicMeter:
                return (this.value / 3.6e+3) / 1000000000000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case EnergyDensityUnits.JoulesPerCubicMeter:
                return value;
            case EnergyDensityUnits.WattHoursPerCubicMeter:
                return value * 3.6e+3;
            case EnergyDensityUnits.KilojoulesPerCubicMeter:
                return (value) * 1000;
            case EnergyDensityUnits.MegajoulesPerCubicMeter:
                return (value) * 1000000;
            case EnergyDensityUnits.GigajoulesPerCubicMeter:
                return (value) * 1000000000;
            case EnergyDensityUnits.TerajoulesPerCubicMeter:
                return (value) * 1000000000000;
            case EnergyDensityUnits.PetajoulesPerCubicMeter:
                return (value) * 1000000000000000;
            case EnergyDensityUnits.KilowattHoursPerCubicMeter:
                return (value * 3.6e+3) * 1000;
            case EnergyDensityUnits.MegawattHoursPerCubicMeter:
                return (value * 3.6e+3) * 1000000;
            case EnergyDensityUnits.GigawattHoursPerCubicMeter:
                return (value * 3.6e+3) * 1000000000;
            case EnergyDensityUnits.TerawattHoursPerCubicMeter:
                return (value * 3.6e+3) * 1000000000000;
            case EnergyDensityUnits.PetawattHoursPerCubicMeter:
                return (value * 3.6e+3) * 1000000000000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the EnergyDensity to string.
     * Note! the default format for EnergyDensity is JoulesPerCubicMeter.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the EnergyDensity.
     * @returns The string format of the EnergyDensity.
     */
    toString(unit = EnergyDensityUnits.JoulesPerCubicMeter) {
        switch (unit) {
            case EnergyDensityUnits.JoulesPerCubicMeter:
                return this.JoulesPerCubicMeter + ` J/m³`;
            case EnergyDensityUnits.WattHoursPerCubicMeter:
                return this.WattHoursPerCubicMeter + ` Wh/m³`;
            case EnergyDensityUnits.KilojoulesPerCubicMeter:
                return this.KilojoulesPerCubicMeter + ` kJ/m³`;
            case EnergyDensityUnits.MegajoulesPerCubicMeter:
                return this.MegajoulesPerCubicMeter + ` MJ/m³`;
            case EnergyDensityUnits.GigajoulesPerCubicMeter:
                return this.GigajoulesPerCubicMeter + ` GJ/m³`;
            case EnergyDensityUnits.TerajoulesPerCubicMeter:
                return this.TerajoulesPerCubicMeter + ` TJ/m³`;
            case EnergyDensityUnits.PetajoulesPerCubicMeter:
                return this.PetajoulesPerCubicMeter + ` PJ/m³`;
            case EnergyDensityUnits.KilowattHoursPerCubicMeter:
                return this.KilowattHoursPerCubicMeter + ` kWh/m³`;
            case EnergyDensityUnits.MegawattHoursPerCubicMeter:
                return this.MegawattHoursPerCubicMeter + ` MWh/m³`;
            case EnergyDensityUnits.GigawattHoursPerCubicMeter:
                return this.GigawattHoursPerCubicMeter + ` GWh/m³`;
            case EnergyDensityUnits.TerawattHoursPerCubicMeter:
                return this.TerawattHoursPerCubicMeter + ` TWh/m³`;
            case EnergyDensityUnits.PetawattHoursPerCubicMeter:
                return this.PetawattHoursPerCubicMeter + ` PWh/m³`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get EnergyDensity unit abbreviation.
     * Note! the default abbreviation for EnergyDensity is JoulesPerCubicMeter.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the EnergyDensity.
     * @returns The abbreviation string of EnergyDensity.
     */
    getUnitAbbreviation(unitAbbreviation = EnergyDensityUnits.JoulesPerCubicMeter) {
        switch (unitAbbreviation) {
            case EnergyDensityUnits.JoulesPerCubicMeter:
                return `J/m³`;
            case EnergyDensityUnits.WattHoursPerCubicMeter:
                return `Wh/m³`;
            case EnergyDensityUnits.KilojoulesPerCubicMeter:
                return `kJ/m³`;
            case EnergyDensityUnits.MegajoulesPerCubicMeter:
                return `MJ/m³`;
            case EnergyDensityUnits.GigajoulesPerCubicMeter:
                return `GJ/m³`;
            case EnergyDensityUnits.TerajoulesPerCubicMeter:
                return `TJ/m³`;
            case EnergyDensityUnits.PetajoulesPerCubicMeter:
                return `PJ/m³`;
            case EnergyDensityUnits.KilowattHoursPerCubicMeter:
                return `kWh/m³`;
            case EnergyDensityUnits.MegawattHoursPerCubicMeter:
                return `MWh/m³`;
            case EnergyDensityUnits.GigawattHoursPerCubicMeter:
                return `GWh/m³`;
            case EnergyDensityUnits.TerawattHoursPerCubicMeter:
                return `TWh/m³`;
            case EnergyDensityUnits.PetawattHoursPerCubicMeter:
                return `PWh/m³`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given EnergyDensity are equals to the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns True if the given EnergyDensity are equal to the current EnergyDensity.
     */
    equals(energyDensity) {
        return this.value === energyDensity.BaseValue;
    }
    /**
     * Compare the given EnergyDensity against the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns 0 if they are equal, -1 if the current EnergyDensity is less then other, 1 if the current EnergyDensity is greater then other.
     */
    compareTo(energyDensity) {
        if (this.value > energyDensity.BaseValue)
            return 1;
        if (this.value < energyDensity.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given EnergyDensity with the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns A new EnergyDensity instance with the results.
     */
    add(energyDensity) {
        return new EnergyDensity(this.value + energyDensity.BaseValue);
    }
    /**
     * Subtract the given EnergyDensity with the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns A new EnergyDensity instance with the results.
     */
    subtract(energyDensity) {
        return new EnergyDensity(this.value - energyDensity.BaseValue);
    }
    /**
     * Multiply the given EnergyDensity with the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns A new EnergyDensity instance with the results.
     */
    multiply(energyDensity) {
        return new EnergyDensity(this.value * energyDensity.BaseValue);
    }
    /**
     * Divide the given EnergyDensity with the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns A new EnergyDensity instance with the results.
     */
    divide(energyDensity) {
        return new EnergyDensity(this.value / energyDensity.BaseValue);
    }
    /**
     * Modulo the given EnergyDensity with the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns A new EnergyDensity instance with the results.
     */
    modulo(energyDensity) {
        return new EnergyDensity(this.value % energyDensity.BaseValue);
    }
    /**
     * Pow the given EnergyDensity with the current EnergyDensity.
     * @param energyDensity The other EnergyDensity.
     * @returns A new EnergyDensity instance with the results.
     */
    pow(energyDensity) {
        return new EnergyDensity(this.value ** energyDensity.BaseValue);
    }
}
exports.EnergyDensity = EnergyDensity;
