// unitsnet-js/dist/heattransfercoefficient.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.HeatTransferCoefficient = exports.HeatTransferCoefficientUnits = void 0;
/** HeatTransferCoefficientUnits enumeration */
var HeatTransferCoefficientUnits;
(function (HeatTransferCoefficientUnits) {
    /** */
    HeatTransferCoefficientUnits[HeatTransferCoefficientUnits["WattsPerSquareMeterKelvin"] = 0] = "WattsPerSquareMeterKelvin";
    /** */
    HeatTransferCoefficientUnits[HeatTransferCoefficientUnits["WattsPerSquareMeterCelsius"] = 1] = "WattsPerSquareMeterCelsius";
    /** */
    HeatTransferCoefficientUnits[HeatTransferCoefficientUnits["BtusPerHourSquareFootDegreeFahrenheit"] = 2] = "BtusPerHourSquareFootDegreeFahrenheit";
    /** */
    HeatTransferCoefficientUnits[HeatTransferCoefficientUnits["CaloriesPerHourSquareMeterDegreeCelsius"] = 3] = "CaloriesPerHourSquareMeterDegreeCelsius";
    /** */
    HeatTransferCoefficientUnits[HeatTransferCoefficientUnits["KilocaloriesPerHourSquareMeterDegreeCelsius"] = 4] = "KilocaloriesPerHourSquareMeterDegreeCelsius";
})(HeatTransferCoefficientUnits = exports.HeatTransferCoefficientUnits || (exports.HeatTransferCoefficientUnits = {}));
/** The heat transfer coefficient or film coefficient, or film effectiveness, in thermodynamics and in mechanics is the proportionality constant between the heat flux and the thermodynamic driving force for the flow of heat (i.e., the temperature difference, ΔT) */
class HeatTransferCoefficient {
    /**
     * Create a new HeatTransferCoefficient.
     * @param value The value.
     * @param fromUnit The ‘HeatTransferCoefficient’ unit to create from.
     * The default unit is WattsPerSquareMeterKelvin
     */
    constructor(value, fromUnit = HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin) {
        this.wattspersquaremeterkelvinLazy = null;
        this.wattspersquaremetercelsiusLazy = null;
        this.btusperhoursquarefootdegreefahrenheitLazy = null;
        this.caloriesperhoursquaremeterdegreecelsiusLazy = null;
        this.kilocaloriesperhoursquaremeterdegreecelsiusLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of HeatTransferCoefficient is WattsPerSquareMeterKelvin.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get WattsPerSquareMeterKelvin() {
        if (this.wattspersquaremeterkelvinLazy !== null) {
            return this.wattspersquaremeterkelvinLazy;
        }
        return this.wattspersquaremeterkelvinLazy = this.convertFromBase(HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin);
    }
    /** */
    get WattsPerSquareMeterCelsius() {
        if (this.wattspersquaremetercelsiusLazy !== null) {
            return this.wattspersquaremetercelsiusLazy;
        }
        return this.wattspersquaremetercelsiusLazy = this.convertFromBase(HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius);
    }
    /** */
    get BtusPerHourSquareFootDegreeFahrenheit() {
        if (this.btusperhoursquarefootdegreefahrenheitLazy !== null) {
            return this.btusperhoursquarefootdegreefahrenheitLazy;
        }
        return this.btusperhoursquarefootdegreefahrenheitLazy = this.convertFromBase(HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit);
    }
    /** */
    get CaloriesPerHourSquareMeterDegreeCelsius() {
        if (this.caloriesperhoursquaremeterdegreecelsiusLazy !== null) {
            return this.caloriesperhoursquaremeterdegreecelsiusLazy;
        }
        return this.caloriesperhoursquaremeterdegreecelsiusLazy = this.convertFromBase(HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius);
    }
    /** */
    get KilocaloriesPerHourSquareMeterDegreeCelsius() {
        if (this.kilocaloriesperhoursquaremeterdegreecelsiusLazy !== null) {
            return this.kilocaloriesperhoursquaremeterdegreecelsiusLazy;
        }
        return this.kilocaloriesperhoursquaremeterdegreecelsiusLazy = this.convertFromBase(HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius);
    }
    /**
     * Create a new HeatTransferCoefficient instance from a WattsPerSquareMeterKelvin
     *
     * @param value The unit as WattsPerSquareMeterKelvin to create a new HeatTransferCoefficient from.
     * @returns The new HeatTransferCoefficient instance.
     */
    static FromWattsPerSquareMeterKelvin(value) {
        return new HeatTransferCoefficient(value, HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin);
    }
    /**
     * Create a new HeatTransferCoefficient instance from a WattsPerSquareMeterCelsius
     *
     * @param value The unit as WattsPerSquareMeterCelsius to create a new HeatTransferCoefficient from.
     * @returns The new HeatTransferCoefficient instance.
     */
    static FromWattsPerSquareMeterCelsius(value) {
        return new HeatTransferCoefficient(value, HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius);
    }
    /**
     * Create a new HeatTransferCoefficient instance from a BtusPerHourSquareFootDegreeFahrenheit
     *
     * @param value The unit as BtusPerHourSquareFootDegreeFahrenheit to create a new HeatTransferCoefficient from.
     * @returns The new HeatTransferCoefficient instance.
     */
    static FromBtusPerHourSquareFootDegreeFahrenheit(value) {
        return new HeatTransferCoefficient(value, HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit);
    }
    /**
     * Create a new HeatTransferCoefficient instance from a CaloriesPerHourSquareMeterDegreeCelsius
     *
     * @param value The unit as CaloriesPerHourSquareMeterDegreeCelsius to create a new HeatTransferCoefficient from.
     * @returns The new HeatTransferCoefficient instance.
     */
    static FromCaloriesPerHourSquareMeterDegreeCelsius(value) {
        return new HeatTransferCoefficient(value, HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius);
    }
    /**
     * Create a new HeatTransferCoefficient instance from a KilocaloriesPerHourSquareMeterDegreeCelsius
     *
     * @param value The unit as KilocaloriesPerHourSquareMeterDegreeCelsius to create a new HeatTransferCoefficient from.
     * @returns The new HeatTransferCoefficient instance.
     */
    static FromKilocaloriesPerHourSquareMeterDegreeCelsius(value) {
        return new HeatTransferCoefficient(value, HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius);
    }
    /**
     * Convert HeatTransferCoefficient to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin: return this.WattsPerSquareMeterKelvin;
            case HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius: return this.WattsPerSquareMeterCelsius;
            case HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit: return this.BtusPerHourSquareFootDegreeFahrenheit;
            case HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius: return this.CaloriesPerHourSquareMeterDegreeCelsius;
            case HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius: return this.KilocaloriesPerHourSquareMeterDegreeCelsius;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin:
                return this.value;
            case HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius:
                return this.value;
            case HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit:
                return this.value / 5.6782633411134878;
            case HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius:
                return (this.value / 4.1868) * 3600;
            case HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius:
                return ((this.value / 4.1868) * 3600) / 1000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin:
                return value;
            case HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius:
                return value;
            case HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit:
                return value * 5.6782633411134878;
            case HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius:
                return (value * 4.1868) / 3600;
            case HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius:
                return ((value * 4.1868) / 3600) * 1000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the HeatTransferCoefficient to string.
     * Note! the default format for HeatTransferCoefficient is WattsPerSquareMeterKelvin.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the HeatTransferCoefficient.
     * @returns The string format of the HeatTransferCoefficient.
     */
    toString(unit = HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin) {
        switch (unit) {
            case HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin:
                return this.WattsPerSquareMeterKelvin + ` W/m²·K`;
            case HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius:
                return this.WattsPerSquareMeterCelsius + ` W/m²·°C`;
            case HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit:
                return this.BtusPerHourSquareFootDegreeFahrenheit + ` Btu/h·ft²·°F`;
            case HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius:
                return this.CaloriesPerHourSquareMeterDegreeCelsius + ` kcal/h·m²·°C`;
            case HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius:
                return this.KilocaloriesPerHourSquareMeterDegreeCelsius + ` kkcal/h·m²·°C`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get HeatTransferCoefficient unit abbreviation.
     * Note! the default abbreviation for HeatTransferCoefficient is WattsPerSquareMeterKelvin.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the HeatTransferCoefficient.
     * @returns The abbreviation string of HeatTransferCoefficient.
     */
    getUnitAbbreviation(unitAbbreviation = HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin) {
        switch (unitAbbreviation) {
            case HeatTransferCoefficientUnits.WattsPerSquareMeterKelvin:
                return `W/m²·K`;
            case HeatTransferCoefficientUnits.WattsPerSquareMeterCelsius:
                return `W/m²·°C`;
            case HeatTransferCoefficientUnits.BtusPerHourSquareFootDegreeFahrenheit:
                return `Btu/h·ft²·°F`;
            case HeatTransferCoefficientUnits.CaloriesPerHourSquareMeterDegreeCelsius:
                return `kcal/h·m²·°C`;
            case HeatTransferCoefficientUnits.KilocaloriesPerHourSquareMeterDegreeCelsius:
                return `kkcal/h·m²·°C`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given HeatTransferCoefficient are equals to the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns True if the given HeatTransferCoefficient are equal to the current HeatTransferCoefficient.
     */
    equals(heatTransferCoefficient) {
        return this.value === heatTransferCoefficient.BaseValue;
    }
    /**
     * Compare the given HeatTransferCoefficient against the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns 0 if they are equal, -1 if the current HeatTransferCoefficient is less then other, 1 if the current HeatTransferCoefficient is greater then other.
     */
    compareTo(heatTransferCoefficient) {
        if (this.value > heatTransferCoefficient.BaseValue)
            return 1;
        if (this.value < heatTransferCoefficient.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given HeatTransferCoefficient with the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns A new HeatTransferCoefficient instance with the results.
     */
    add(heatTransferCoefficient) {
        return new HeatTransferCoefficient(this.value + heatTransferCoefficient.BaseValue);
    }
    /**
     * Subtract the given HeatTransferCoefficient with the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns A new HeatTransferCoefficient instance with the results.
     */
    subtract(heatTransferCoefficient) {
        return new HeatTransferCoefficient(this.value - heatTransferCoefficient.BaseValue);
    }
    /**
     * Multiply the given HeatTransferCoefficient with the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns A new HeatTransferCoefficient instance with the results.
     */
    multiply(heatTransferCoefficient) {
        return new HeatTransferCoefficient(this.value * heatTransferCoefficient.BaseValue);
    }
    /**
     * Divide the given HeatTransferCoefficient with the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns A new HeatTransferCoefficient instance with the results.
     */
    divide(heatTransferCoefficient) {
        return new HeatTransferCoefficient(this.value / heatTransferCoefficient.BaseValue);
    }
    /**
     * Modulo the given HeatTransferCoefficient with the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns A new HeatTransferCoefficient instance with the results.
     */
    modulo(heatTransferCoefficient) {
        return new HeatTransferCoefficient(this.value % heatTransferCoefficient.BaseValue);
    }
    /**
     * Pow the given HeatTransferCoefficient with the current HeatTransferCoefficient.
     * @param heatTransferCoefficient The other HeatTransferCoefficient.
     * @returns A new HeatTransferCoefficient instance with the results.
     */
    pow(heatTransferCoefficient) {
        return new HeatTransferCoefficient(this.value ** heatTransferCoefficient.BaseValue);
    }
}
exports.HeatTransferCoefficient = HeatTransferCoefficient;
