// unitsnet-js/dist/jerk.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Jerk = exports.JerkUnits = void 0;
/** JerkUnits enumeration */
var JerkUnits;
(function (JerkUnits) {
    /** */
    JerkUnits[JerkUnits["MetersPerSecondCubed"] = 0] = "MetersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["InchesPerSecondCubed"] = 1] = "InchesPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["FeetPerSecondCubed"] = 2] = "FeetPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["StandardGravitiesPerSecond"] = 3] = "StandardGravitiesPerSecond";
    /** */
    JerkUnits[JerkUnits["NanometersPerSecondCubed"] = 4] = "NanometersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["MicrometersPerSecondCubed"] = 5] = "MicrometersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["MillimetersPerSecondCubed"] = 6] = "MillimetersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["CentimetersPerSecondCubed"] = 7] = "CentimetersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["DecimetersPerSecondCubed"] = 8] = "DecimetersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["KilometersPerSecondCubed"] = 9] = "KilometersPerSecondCubed";
    /** */
    JerkUnits[JerkUnits["MillistandardGravitiesPerSecond"] = 10] = "MillistandardGravitiesPerSecond";
})(JerkUnits = exports.JerkUnits || (exports.JerkUnits = {}));
/** Jerk or Jolt, in physics, is the rate at which the acceleration of an object changes over time. The SI unit for jerk is the Meter per second cubed (m/s³). Jerks are vector quantities (they have magnitude and direction) and add according to the parallelogram law. */
class Jerk {
    /**
     * Create a new Jerk.
     * @param value The value.
     * @param fromUnit The ‘Jerk’ unit to create from.
     * The default unit is MetersPerSecondCubed
     */
    constructor(value, fromUnit = JerkUnits.MetersPerSecondCubed) {
        this.meterspersecondcubedLazy = null;
        this.inchespersecondcubedLazy = null;
        this.feetpersecondcubedLazy = null;
        this.standardgravitiespersecondLazy = null;
        this.nanometerspersecondcubedLazy = null;
        this.micrometerspersecondcubedLazy = null;
        this.millimeterspersecondcubedLazy = null;
        this.centimeterspersecondcubedLazy = null;
        this.decimeterspersecondcubedLazy = null;
        this.kilometerspersecondcubedLazy = null;
        this.millistandardgravitiespersecondLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of Jerk is MetersPerSecondCubed.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get MetersPerSecondCubed() {
        if (this.meterspersecondcubedLazy !== null) {
            return this.meterspersecondcubedLazy;
        }
        return this.meterspersecondcubedLazy = this.convertFromBase(JerkUnits.MetersPerSecondCubed);
    }
    /** */
    get InchesPerSecondCubed() {
        if (this.inchespersecondcubedLazy !== null) {
            return this.inchespersecondcubedLazy;
        }
        return this.inchespersecondcubedLazy = this.convertFromBase(JerkUnits.InchesPerSecondCubed);
    }
    /** */
    get FeetPerSecondCubed() {
        if (this.feetpersecondcubedLazy !== null) {
            return this.feetpersecondcubedLazy;
        }
        return this.feetpersecondcubedLazy = this.convertFromBase(JerkUnits.FeetPerSecondCubed);
    }
    /** */
    get StandardGravitiesPerSecond() {
        if (this.standardgravitiespersecondLazy !== null) {
            return this.standardgravitiespersecondLazy;
        }
        return this.standardgravitiespersecondLazy = this.convertFromBase(JerkUnits.StandardGravitiesPerSecond);
    }
    /** */
    get NanometersPerSecondCubed() {
        if (this.nanometerspersecondcubedLazy !== null) {
            return this.nanometerspersecondcubedLazy;
        }
        return this.nanometerspersecondcubedLazy = this.convertFromBase(JerkUnits.NanometersPerSecondCubed);
    }
    /** */
    get MicrometersPerSecondCubed() {
        if (this.micrometerspersecondcubedLazy !== null) {
            return this.micrometerspersecondcubedLazy;
        }
        return this.micrometerspersecondcubedLazy = this.convertFromBase(JerkUnits.MicrometersPerSecondCubed);
    }
    /** */
    get MillimetersPerSecondCubed() {
        if (this.millimeterspersecondcubedLazy !== null) {
            return this.millimeterspersecondcubedLazy;
        }
        return this.millimeterspersecondcubedLazy = this.convertFromBase(JerkUnits.MillimetersPerSecondCubed);
    }
    /** */
    get CentimetersPerSecondCubed() {
        if (this.centimeterspersecondcubedLazy !== null) {
            return this.centimeterspersecondcubedLazy;
        }
        return this.centimeterspersecondcubedLazy = this.convertFromBase(JerkUnits.CentimetersPerSecondCubed);
    }
    /** */
    get DecimetersPerSecondCubed() {
        if (this.decimeterspersecondcubedLazy !== null) {
            return this.decimeterspersecondcubedLazy;
        }
        return this.decimeterspersecondcubedLazy = this.convertFromBase(JerkUnits.DecimetersPerSecondCubed);
    }
    /** */
    get KilometersPerSecondCubed() {
        if (this.kilometerspersecondcubedLazy !== null) {
            return this.kilometerspersecondcubedLazy;
        }
        return this.kilometerspersecondcubedLazy = this.convertFromBase(JerkUnits.KilometersPerSecondCubed);
    }
    /** */
    get MillistandardGravitiesPerSecond() {
        if (this.millistandardgravitiespersecondLazy !== null) {
            return this.millistandardgravitiespersecondLazy;
        }
        return this.millistandardgravitiespersecondLazy = this.convertFromBase(JerkUnits.MillistandardGravitiesPerSecond);
    }
    /**
     * Create a new Jerk instance from a MetersPerSecondCubed
     *
     * @param value The unit as MetersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromMetersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.MetersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a InchesPerSecondCubed
     *
     * @param value The unit as InchesPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromInchesPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.InchesPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a FeetPerSecondCubed
     *
     * @param value The unit as FeetPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromFeetPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.FeetPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a StandardGravitiesPerSecond
     *
     * @param value The unit as StandardGravitiesPerSecond to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromStandardGravitiesPerSecond(value) {
        return new Jerk(value, JerkUnits.StandardGravitiesPerSecond);
    }
    /**
     * Create a new Jerk instance from a NanometersPerSecondCubed
     *
     * @param value The unit as NanometersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromNanometersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.NanometersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a MicrometersPerSecondCubed
     *
     * @param value The unit as MicrometersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromMicrometersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.MicrometersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a MillimetersPerSecondCubed
     *
     * @param value The unit as MillimetersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromMillimetersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.MillimetersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a CentimetersPerSecondCubed
     *
     * @param value The unit as CentimetersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromCentimetersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.CentimetersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a DecimetersPerSecondCubed
     *
     * @param value The unit as DecimetersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromDecimetersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.DecimetersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a KilometersPerSecondCubed
     *
     * @param value The unit as KilometersPerSecondCubed to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromKilometersPerSecondCubed(value) {
        return new Jerk(value, JerkUnits.KilometersPerSecondCubed);
    }
    /**
     * Create a new Jerk instance from a MillistandardGravitiesPerSecond
     *
     * @param value The unit as MillistandardGravitiesPerSecond to create a new Jerk from.
     * @returns The new Jerk instance.
     */
    static FromMillistandardGravitiesPerSecond(value) {
        return new Jerk(value, JerkUnits.MillistandardGravitiesPerSecond);
    }
    /**
     * Convert Jerk to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case JerkUnits.MetersPerSecondCubed: return this.MetersPerSecondCubed;
            case JerkUnits.InchesPerSecondCubed: return this.InchesPerSecondCubed;
            case JerkUnits.FeetPerSecondCubed: return this.FeetPerSecondCubed;
            case JerkUnits.StandardGravitiesPerSecond: return this.StandardGravitiesPerSecond;
            case JerkUnits.NanometersPerSecondCubed: return this.NanometersPerSecondCubed;
            case JerkUnits.MicrometersPerSecondCubed: return this.MicrometersPerSecondCubed;
            case JerkUnits.MillimetersPerSecondCubed: return this.MillimetersPerSecondCubed;
            case JerkUnits.CentimetersPerSecondCubed: return this.CentimetersPerSecondCubed;
            case JerkUnits.DecimetersPerSecondCubed: return this.DecimetersPerSecondCubed;
            case JerkUnits.KilometersPerSecondCubed: return this.KilometersPerSecondCubed;
            case JerkUnits.MillistandardGravitiesPerSecond: return this.MillistandardGravitiesPerSecond;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case JerkUnits.MetersPerSecondCubed:
                return this.value;
            case JerkUnits.InchesPerSecondCubed:
                return this.value / 0.0254;
            case JerkUnits.FeetPerSecondCubed:
                return this.value / 0.304800;
            case JerkUnits.StandardGravitiesPerSecond:
                return this.value / 9.80665;
            case JerkUnits.NanometersPerSecondCubed:
                return (this.value) / 1e-9;
            case JerkUnits.MicrometersPerSecondCubed:
                return (this.value) / 0.000001;
            case JerkUnits.MillimetersPerSecondCubed:
                return (this.value) / 0.001;
            case JerkUnits.CentimetersPerSecondCubed:
                return (this.value) / 0.01;
            case JerkUnits.DecimetersPerSecondCubed:
                return (this.value) / 0.1;
            case JerkUnits.KilometersPerSecondCubed:
                return (this.value) / 1000;
            case JerkUnits.MillistandardGravitiesPerSecond:
                return (this.value / 9.80665) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case JerkUnits.MetersPerSecondCubed:
                return value;
            case JerkUnits.InchesPerSecondCubed:
                return value * 0.0254;
            case JerkUnits.FeetPerSecondCubed:
                return value * 0.304800;
            case JerkUnits.StandardGravitiesPerSecond:
                return value * 9.80665;
            case JerkUnits.NanometersPerSecondCubed:
                return (value) * 1e-9;
            case JerkUnits.MicrometersPerSecondCubed:
                return (value) * 0.000001;
            case JerkUnits.MillimetersPerSecondCubed:
                return (value) * 0.001;
            case JerkUnits.CentimetersPerSecondCubed:
                return (value) * 0.01;
            case JerkUnits.DecimetersPerSecondCubed:
                return (value) * 0.1;
            case JerkUnits.KilometersPerSecondCubed:
                return (value) * 1000;
            case JerkUnits.MillistandardGravitiesPerSecond:
                return (value * 9.80665) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the Jerk to string.
     * Note! the default format for Jerk is MetersPerSecondCubed.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the Jerk.
     * @returns The string format of the Jerk.
     */
    toString(unit = JerkUnits.MetersPerSecondCubed) {
        switch (unit) {
            case JerkUnits.MetersPerSecondCubed:
                return this.MetersPerSecondCubed + ` m/s³`;
            case JerkUnits.InchesPerSecondCubed:
                return this.InchesPerSecondCubed + ` in/s³`;
            case JerkUnits.FeetPerSecondCubed:
                return this.FeetPerSecondCubed + ` ft/s³`;
            case JerkUnits.StandardGravitiesPerSecond:
                return this.StandardGravitiesPerSecond + ` g/s`;
            case JerkUnits.NanometersPerSecondCubed:
                return this.NanometersPerSecondCubed + ` nm/s³`;
            case JerkUnits.MicrometersPerSecondCubed:
                return this.MicrometersPerSecondCubed + ` μm/s³`;
            case JerkUnits.MillimetersPerSecondCubed:
                return this.MillimetersPerSecondCubed + ` mm/s³`;
            case JerkUnits.CentimetersPerSecondCubed:
                return this.CentimetersPerSecondCubed + ` cm/s³`;
            case JerkUnits.DecimetersPerSecondCubed:
                return this.DecimetersPerSecondCubed + ` dm/s³`;
            case JerkUnits.KilometersPerSecondCubed:
                return this.KilometersPerSecondCubed + ` km/s³`;
            case JerkUnits.MillistandardGravitiesPerSecond:
                return this.MillistandardGravitiesPerSecond + ` mg/s`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get Jerk unit abbreviation.
     * Note! the default abbreviation for Jerk is MetersPerSecondCubed.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the Jerk.
     * @returns The abbreviation string of Jerk.
     */
    getUnitAbbreviation(unitAbbreviation = JerkUnits.MetersPerSecondCubed) {
        switch (unitAbbreviation) {
            case JerkUnits.MetersPerSecondCubed:
                return `m/s³`;
            case JerkUnits.InchesPerSecondCubed:
                return `in/s³`;
            case JerkUnits.FeetPerSecondCubed:
                return `ft/s³`;
            case JerkUnits.StandardGravitiesPerSecond:
                return `g/s`;
            case JerkUnits.NanometersPerSecondCubed:
                return `nm/s³`;
            case JerkUnits.MicrometersPerSecondCubed:
                return `μm/s³`;
            case JerkUnits.MillimetersPerSecondCubed:
                return `mm/s³`;
            case JerkUnits.CentimetersPerSecondCubed:
                return `cm/s³`;
            case JerkUnits.DecimetersPerSecondCubed:
                return `dm/s³`;
            case JerkUnits.KilometersPerSecondCubed:
                return `km/s³`;
            case JerkUnits.MillistandardGravitiesPerSecond:
                return `mg/s`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given Jerk are equals to the current Jerk.
     * @param jerk The other Jerk.
     * @returns True if the given Jerk are equal to the current Jerk.
     */
    equals(jerk) {
        return this.value === jerk.BaseValue;
    }
    /**
     * Compare the given Jerk against the current Jerk.
     * @param jerk The other Jerk.
     * @returns 0 if they are equal, -1 if the current Jerk is less then other, 1 if the current Jerk is greater then other.
     */
    compareTo(jerk) {
        if (this.value > jerk.BaseValue)
            return 1;
        if (this.value < jerk.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given Jerk with the current Jerk.
     * @param jerk The other Jerk.
     * @returns A new Jerk instance with the results.
     */
    add(jerk) {
        return new Jerk(this.value + jerk.BaseValue);
    }
    /**
     * Subtract the given Jerk with the current Jerk.
     * @param jerk The other Jerk.
     * @returns A new Jerk instance with the results.
     */
    subtract(jerk) {
        return new Jerk(this.value - jerk.BaseValue);
    }
    /**
     * Multiply the given Jerk with the current Jerk.
     * @param jerk The other Jerk.
     * @returns A new Jerk instance with the results.
     */
    multiply(jerk) {
        return new Jerk(this.value * jerk.BaseValue);
    }
    /**
     * Divide the given Jerk with the current Jerk.
     * @param jerk The other Jerk.
     * @returns A new Jerk instance with the results.
     */
    divide(jerk) {
        return new Jerk(this.value / jerk.BaseValue);
    }
    /**
     * Modulo the given Jerk with the current Jerk.
     * @param jerk The other Jerk.
     * @returns A new Jerk instance with the results.
     */
    modulo(jerk) {
        return new Jerk(this.value % jerk.BaseValue);
    }
    /**
     * Pow the given Jerk with the current Jerk.
     * @param jerk The other Jerk.
     * @returns A new Jerk instance with the results.
     */
    pow(jerk) {
        return new Jerk(this.value ** jerk.BaseValue);
    }
}
exports.Jerk = Jerk;
