// unitsnet-js/dist/leakrate.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LeakRate = exports.LeakRateUnits = void 0;
/** LeakRateUnits enumeration */
var LeakRateUnits;
(function (LeakRateUnits) {
    /** */
    LeakRateUnits[LeakRateUnits["PascalCubicMetersPerSecond"] = 0] = "PascalCubicMetersPerSecond";
    /** */
    LeakRateUnits[LeakRateUnits["MillibarLitersPerSecond"] = 1] = "MillibarLitersPerSecond";
    /** */
    LeakRateUnits[LeakRateUnits["TorrLitersPerSecond"] = 2] = "TorrLitersPerSecond";
})(LeakRateUnits = exports.LeakRateUnits || (exports.LeakRateUnits = {}));
/** A leakage rate of QL = 1 Pa-m³/s is given when the pressure in a closed, evacuated container with a volume of 1 m³ rises by 1 Pa per second or when the pressure in the container drops by 1 Pa in the event of overpressure. */
class LeakRate {
    /**
     * Create a new LeakRate.
     * @param value The value.
     * @param fromUnit The ‘LeakRate’ unit to create from.
     * The default unit is PascalCubicMetersPerSecond
     */
    constructor(value, fromUnit = LeakRateUnits.PascalCubicMetersPerSecond) {
        this.pascalcubicmeterspersecondLazy = null;
        this.millibarliterspersecondLazy = null;
        this.torrliterspersecondLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of LeakRate is PascalCubicMetersPerSecond.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get PascalCubicMetersPerSecond() {
        if (this.pascalcubicmeterspersecondLazy !== null) {
            return this.pascalcubicmeterspersecondLazy;
        }
        return this.pascalcubicmeterspersecondLazy = this.convertFromBase(LeakRateUnits.PascalCubicMetersPerSecond);
    }
    /** */
    get MillibarLitersPerSecond() {
        if (this.millibarliterspersecondLazy !== null) {
            return this.millibarliterspersecondLazy;
        }
        return this.millibarliterspersecondLazy = this.convertFromBase(LeakRateUnits.MillibarLitersPerSecond);
    }
    /** */
    get TorrLitersPerSecond() {
        if (this.torrliterspersecondLazy !== null) {
            return this.torrliterspersecondLazy;
        }
        return this.torrliterspersecondLazy = this.convertFromBase(LeakRateUnits.TorrLitersPerSecond);
    }
    /**
     * Create a new LeakRate instance from a PascalCubicMetersPerSecond
     *
     * @param value The unit as PascalCubicMetersPerSecond to create a new LeakRate from.
     * @returns The new LeakRate instance.
     */
    static FromPascalCubicMetersPerSecond(value) {
        return new LeakRate(value, LeakRateUnits.PascalCubicMetersPerSecond);
    }
    /**
     * Create a new LeakRate instance from a MillibarLitersPerSecond
     *
     * @param value The unit as MillibarLitersPerSecond to create a new LeakRate from.
     * @returns The new LeakRate instance.
     */
    static FromMillibarLitersPerSecond(value) {
        return new LeakRate(value, LeakRateUnits.MillibarLitersPerSecond);
    }
    /**
     * Create a new LeakRate instance from a TorrLitersPerSecond
     *
     * @param value The unit as TorrLitersPerSecond to create a new LeakRate from.
     * @returns The new LeakRate instance.
     */
    static FromTorrLitersPerSecond(value) {
        return new LeakRate(value, LeakRateUnits.TorrLitersPerSecond);
    }
    /**
     * Convert LeakRate to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case LeakRateUnits.PascalCubicMetersPerSecond: return this.PascalCubicMetersPerSecond;
            case LeakRateUnits.MillibarLitersPerSecond: return this.MillibarLitersPerSecond;
            case LeakRateUnits.TorrLitersPerSecond: return this.TorrLitersPerSecond;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case LeakRateUnits.PascalCubicMetersPerSecond:
                return this.value;
            case LeakRateUnits.MillibarLitersPerSecond:
                return this.value * 10;
            case LeakRateUnits.TorrLitersPerSecond:
                return this.value * 7.5;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case LeakRateUnits.PascalCubicMetersPerSecond:
                return value;
            case LeakRateUnits.MillibarLitersPerSecond:
                return value / 10;
            case LeakRateUnits.TorrLitersPerSecond:
                return value / 7.5;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the LeakRate to string.
     * Note! the default format for LeakRate is PascalCubicMetersPerSecond.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the LeakRate.
     * @returns The string format of the LeakRate.
     */
    toString(unit = LeakRateUnits.PascalCubicMetersPerSecond) {
        switch (unit) {
            case LeakRateUnits.PascalCubicMetersPerSecond:
                return this.PascalCubicMetersPerSecond + ` Pa·m³/s`;
            case LeakRateUnits.MillibarLitersPerSecond:
                return this.MillibarLitersPerSecond + ` mbar·l/s`;
            case LeakRateUnits.TorrLitersPerSecond:
                return this.TorrLitersPerSecond + ` Torr·l/s`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get LeakRate unit abbreviation.
     * Note! the default abbreviation for LeakRate is PascalCubicMetersPerSecond.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the LeakRate.
     * @returns The abbreviation string of LeakRate.
     */
    getUnitAbbreviation(unitAbbreviation = LeakRateUnits.PascalCubicMetersPerSecond) {
        switch (unitAbbreviation) {
            case LeakRateUnits.PascalCubicMetersPerSecond:
                return `Pa·m³/s`;
            case LeakRateUnits.MillibarLitersPerSecond:
                return `mbar·l/s`;
            case LeakRateUnits.TorrLitersPerSecond:
                return `Torr·l/s`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given LeakRate are equals to the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns True if the given LeakRate are equal to the current LeakRate.
     */
    equals(leakRate) {
        return this.value === leakRate.BaseValue;
    }
    /**
     * Compare the given LeakRate against the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns 0 if they are equal, -1 if the current LeakRate is less then other, 1 if the current LeakRate is greater then other.
     */
    compareTo(leakRate) {
        if (this.value > leakRate.BaseValue)
            return 1;
        if (this.value < leakRate.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given LeakRate with the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns A new LeakRate instance with the results.
     */
    add(leakRate) {
        return new LeakRate(this.value + leakRate.BaseValue);
    }
    /**
     * Subtract the given LeakRate with the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns A new LeakRate instance with the results.
     */
    subtract(leakRate) {
        return new LeakRate(this.value - leakRate.BaseValue);
    }
    /**
     * Multiply the given LeakRate with the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns A new LeakRate instance with the results.
     */
    multiply(leakRate) {
        return new LeakRate(this.value * leakRate.BaseValue);
    }
    /**
     * Divide the given LeakRate with the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns A new LeakRate instance with the results.
     */
    divide(leakRate) {
        return new LeakRate(this.value / leakRate.BaseValue);
    }
    /**
     * Modulo the given LeakRate with the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns A new LeakRate instance with the results.
     */
    modulo(leakRate) {
        return new LeakRate(this.value % leakRate.BaseValue);
    }
    /**
     * Pow the given LeakRate with the current LeakRate.
     * @param leakRate The other LeakRate.
     * @returns A new LeakRate instance with the results.
     */
    pow(leakRate) {
        return new LeakRate(this.value ** leakRate.BaseValue);
    }
}
exports.LeakRate = LeakRate;
