// unitsnet-js/dist/lineardensity.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LinearDensity = exports.LinearDensityUnits = void 0;
/** LinearDensityUnits enumeration */
var LinearDensityUnits;
(function (LinearDensityUnits) {
    /** */
    LinearDensityUnits[LinearDensityUnits["GramsPerMillimeter"] = 0] = "GramsPerMillimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["GramsPerCentimeter"] = 1] = "GramsPerCentimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["GramsPerMeter"] = 2] = "GramsPerMeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["PoundsPerInch"] = 3] = "PoundsPerInch";
    /** */
    LinearDensityUnits[LinearDensityUnits["PoundsPerFoot"] = 4] = "PoundsPerFoot";
    /** */
    LinearDensityUnits[LinearDensityUnits["MicrogramsPerMillimeter"] = 5] = "MicrogramsPerMillimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["MilligramsPerMillimeter"] = 6] = "MilligramsPerMillimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["KilogramsPerMillimeter"] = 7] = "KilogramsPerMillimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["MicrogramsPerCentimeter"] = 8] = "MicrogramsPerCentimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["MilligramsPerCentimeter"] = 9] = "MilligramsPerCentimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["KilogramsPerCentimeter"] = 10] = "KilogramsPerCentimeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["MicrogramsPerMeter"] = 11] = "MicrogramsPerMeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["MilligramsPerMeter"] = 12] = "MilligramsPerMeter";
    /** */
    LinearDensityUnits[LinearDensityUnits["KilogramsPerMeter"] = 13] = "KilogramsPerMeter";
})(LinearDensityUnits = exports.LinearDensityUnits || (exports.LinearDensityUnits = {}));
/** The Linear Density, or more precisely, the linear mass density, of a substance is its mass per unit length.  The term linear density is most often used when describing the characteristics of one-dimensional objects, although linear density can also be used to describe the density of a three-dimensional quantity along one particular dimension. */
class LinearDensity {
    /**
     * Create a new LinearDensity.
     * @param value The value.
     * @param fromUnit The ‘LinearDensity’ unit to create from.
     * The default unit is KilogramsPerMeter
     */
    constructor(value, fromUnit = LinearDensityUnits.KilogramsPerMeter) {
        this.gramspermillimeterLazy = null;
        this.gramspercentimeterLazy = null;
        this.gramspermeterLazy = null;
        this.poundsperinchLazy = null;
        this.poundsperfootLazy = null;
        this.microgramspermillimeterLazy = null;
        this.milligramspermillimeterLazy = null;
        this.kilogramspermillimeterLazy = null;
        this.microgramspercentimeterLazy = null;
        this.milligramspercentimeterLazy = null;
        this.kilogramspercentimeterLazy = null;
        this.microgramspermeterLazy = null;
        this.milligramspermeterLazy = null;
        this.kilogramspermeterLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of LinearDensity is KilogramsPerMeter.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get GramsPerMillimeter() {
        if (this.gramspermillimeterLazy !== null) {
            return this.gramspermillimeterLazy;
        }
        return this.gramspermillimeterLazy = this.convertFromBase(LinearDensityUnits.GramsPerMillimeter);
    }
    /** */
    get GramsPerCentimeter() {
        if (this.gramspercentimeterLazy !== null) {
            return this.gramspercentimeterLazy;
        }
        return this.gramspercentimeterLazy = this.convertFromBase(LinearDensityUnits.GramsPerCentimeter);
    }
    /** */
    get GramsPerMeter() {
        if (this.gramspermeterLazy !== null) {
            return this.gramspermeterLazy;
        }
        return this.gramspermeterLazy = this.convertFromBase(LinearDensityUnits.GramsPerMeter);
    }
    /** */
    get PoundsPerInch() {
        if (this.poundsperinchLazy !== null) {
            return this.poundsperinchLazy;
        }
        return this.poundsperinchLazy = this.convertFromBase(LinearDensityUnits.PoundsPerInch);
    }
    /** */
    get PoundsPerFoot() {
        if (this.poundsperfootLazy !== null) {
            return this.poundsperfootLazy;
        }
        return this.poundsperfootLazy = this.convertFromBase(LinearDensityUnits.PoundsPerFoot);
    }
    /** */
    get MicrogramsPerMillimeter() {
        if (this.microgramspermillimeterLazy !== null) {
            return this.microgramspermillimeterLazy;
        }
        return this.microgramspermillimeterLazy = this.convertFromBase(LinearDensityUnits.MicrogramsPerMillimeter);
    }
    /** */
    get MilligramsPerMillimeter() {
        if (this.milligramspermillimeterLazy !== null) {
            return this.milligramspermillimeterLazy;
        }
        return this.milligramspermillimeterLazy = this.convertFromBase(LinearDensityUnits.MilligramsPerMillimeter);
    }
    /** */
    get KilogramsPerMillimeter() {
        if (this.kilogramspermillimeterLazy !== null) {
            return this.kilogramspermillimeterLazy;
        }
        return this.kilogramspermillimeterLazy = this.convertFromBase(LinearDensityUnits.KilogramsPerMillimeter);
    }
    /** */
    get MicrogramsPerCentimeter() {
        if (this.microgramspercentimeterLazy !== null) {
            return this.microgramspercentimeterLazy;
        }
        return this.microgramspercentimeterLazy = this.convertFromBase(LinearDensityUnits.MicrogramsPerCentimeter);
    }
    /** */
    get MilligramsPerCentimeter() {
        if (this.milligramspercentimeterLazy !== null) {
            return this.milligramspercentimeterLazy;
        }
        return this.milligramspercentimeterLazy = this.convertFromBase(LinearDensityUnits.MilligramsPerCentimeter);
    }
    /** */
    get KilogramsPerCentimeter() {
        if (this.kilogramspercentimeterLazy !== null) {
            return this.kilogramspercentimeterLazy;
        }
        return this.kilogramspercentimeterLazy = this.convertFromBase(LinearDensityUnits.KilogramsPerCentimeter);
    }
    /** */
    get MicrogramsPerMeter() {
        if (this.microgramspermeterLazy !== null) {
            return this.microgramspermeterLazy;
        }
        return this.microgramspermeterLazy = this.convertFromBase(LinearDensityUnits.MicrogramsPerMeter);
    }
    /** */
    get MilligramsPerMeter() {
        if (this.milligramspermeterLazy !== null) {
            return this.milligramspermeterLazy;
        }
        return this.milligramspermeterLazy = this.convertFromBase(LinearDensityUnits.MilligramsPerMeter);
    }
    /** */
    get KilogramsPerMeter() {
        if (this.kilogramspermeterLazy !== null) {
            return this.kilogramspermeterLazy;
        }
        return this.kilogramspermeterLazy = this.convertFromBase(LinearDensityUnits.KilogramsPerMeter);
    }
    /**
     * Create a new LinearDensity instance from a GramsPerMillimeter
     *
     * @param value The unit as GramsPerMillimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromGramsPerMillimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.GramsPerMillimeter);
    }
    /**
     * Create a new LinearDensity instance from a GramsPerCentimeter
     *
     * @param value The unit as GramsPerCentimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromGramsPerCentimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.GramsPerCentimeter);
    }
    /**
     * Create a new LinearDensity instance from a GramsPerMeter
     *
     * @param value The unit as GramsPerMeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromGramsPerMeter(value) {
        return new LinearDensity(value, LinearDensityUnits.GramsPerMeter);
    }
    /**
     * Create a new LinearDensity instance from a PoundsPerInch
     *
     * @param value The unit as PoundsPerInch to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromPoundsPerInch(value) {
        return new LinearDensity(value, LinearDensityUnits.PoundsPerInch);
    }
    /**
     * Create a new LinearDensity instance from a PoundsPerFoot
     *
     * @param value The unit as PoundsPerFoot to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromPoundsPerFoot(value) {
        return new LinearDensity(value, LinearDensityUnits.PoundsPerFoot);
    }
    /**
     * Create a new LinearDensity instance from a MicrogramsPerMillimeter
     *
     * @param value The unit as MicrogramsPerMillimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromMicrogramsPerMillimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.MicrogramsPerMillimeter);
    }
    /**
     * Create a new LinearDensity instance from a MilligramsPerMillimeter
     *
     * @param value The unit as MilligramsPerMillimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromMilligramsPerMillimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.MilligramsPerMillimeter);
    }
    /**
     * Create a new LinearDensity instance from a KilogramsPerMillimeter
     *
     * @param value The unit as KilogramsPerMillimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromKilogramsPerMillimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.KilogramsPerMillimeter);
    }
    /**
     * Create a new LinearDensity instance from a MicrogramsPerCentimeter
     *
     * @param value The unit as MicrogramsPerCentimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromMicrogramsPerCentimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.MicrogramsPerCentimeter);
    }
    /**
     * Create a new LinearDensity instance from a MilligramsPerCentimeter
     *
     * @param value The unit as MilligramsPerCentimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromMilligramsPerCentimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.MilligramsPerCentimeter);
    }
    /**
     * Create a new LinearDensity instance from a KilogramsPerCentimeter
     *
     * @param value The unit as KilogramsPerCentimeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromKilogramsPerCentimeter(value) {
        return new LinearDensity(value, LinearDensityUnits.KilogramsPerCentimeter);
    }
    /**
     * Create a new LinearDensity instance from a MicrogramsPerMeter
     *
     * @param value The unit as MicrogramsPerMeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromMicrogramsPerMeter(value) {
        return new LinearDensity(value, LinearDensityUnits.MicrogramsPerMeter);
    }
    /**
     * Create a new LinearDensity instance from a MilligramsPerMeter
     *
     * @param value The unit as MilligramsPerMeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromMilligramsPerMeter(value) {
        return new LinearDensity(value, LinearDensityUnits.MilligramsPerMeter);
    }
    /**
     * Create a new LinearDensity instance from a KilogramsPerMeter
     *
     * @param value The unit as KilogramsPerMeter to create a new LinearDensity from.
     * @returns The new LinearDensity instance.
     */
    static FromKilogramsPerMeter(value) {
        return new LinearDensity(value, LinearDensityUnits.KilogramsPerMeter);
    }
    /**
     * Convert LinearDensity to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case LinearDensityUnits.GramsPerMillimeter: return this.GramsPerMillimeter;
            case LinearDensityUnits.GramsPerCentimeter: return this.GramsPerCentimeter;
            case LinearDensityUnits.GramsPerMeter: return this.GramsPerMeter;
            case LinearDensityUnits.PoundsPerInch: return this.PoundsPerInch;
            case LinearDensityUnits.PoundsPerFoot: return this.PoundsPerFoot;
            case LinearDensityUnits.MicrogramsPerMillimeter: return this.MicrogramsPerMillimeter;
            case LinearDensityUnits.MilligramsPerMillimeter: return this.MilligramsPerMillimeter;
            case LinearDensityUnits.KilogramsPerMillimeter: return this.KilogramsPerMillimeter;
            case LinearDensityUnits.MicrogramsPerCentimeter: return this.MicrogramsPerCentimeter;
            case LinearDensityUnits.MilligramsPerCentimeter: return this.MilligramsPerCentimeter;
            case LinearDensityUnits.KilogramsPerCentimeter: return this.KilogramsPerCentimeter;
            case LinearDensityUnits.MicrogramsPerMeter: return this.MicrogramsPerMeter;
            case LinearDensityUnits.MilligramsPerMeter: return this.MilligramsPerMeter;
            case LinearDensityUnits.KilogramsPerMeter: return this.KilogramsPerMeter;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case LinearDensityUnits.GramsPerMillimeter:
                return this.value;
            case LinearDensityUnits.GramsPerCentimeter:
                return this.value / 1e-1;
            case LinearDensityUnits.GramsPerMeter:
                return this.value / 1e-3;
            case LinearDensityUnits.PoundsPerInch:
                return this.value * 5.5997415e-2;
            case LinearDensityUnits.PoundsPerFoot:
                return this.value / 1.48816394;
            case LinearDensityUnits.MicrogramsPerMillimeter:
                return (this.value) / 0.000001;
            case LinearDensityUnits.MilligramsPerMillimeter:
                return (this.value) / 0.001;
            case LinearDensityUnits.KilogramsPerMillimeter:
                return (this.value) / 1000;
            case LinearDensityUnits.MicrogramsPerCentimeter:
                return (this.value / 1e-1) / 0.000001;
            case LinearDensityUnits.MilligramsPerCentimeter:
                return (this.value / 1e-1) / 0.001;
            case LinearDensityUnits.KilogramsPerCentimeter:
                return (this.value / 1e-1) / 1000;
            case LinearDensityUnits.MicrogramsPerMeter:
                return (this.value / 1e-3) / 0.000001;
            case LinearDensityUnits.MilligramsPerMeter:
                return (this.value / 1e-3) / 0.001;
            case LinearDensityUnits.KilogramsPerMeter:
                return (this.value / 1e-3) / 1000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case LinearDensityUnits.GramsPerMillimeter:
                return value;
            case LinearDensityUnits.GramsPerCentimeter:
                return value * 1e-1;
            case LinearDensityUnits.GramsPerMeter:
                return value * 1e-3;
            case LinearDensityUnits.PoundsPerInch:
                return value / 5.5997415e-2;
            case LinearDensityUnits.PoundsPerFoot:
                return value * 1.48816394;
            case LinearDensityUnits.MicrogramsPerMillimeter:
                return (value) * 0.000001;
            case LinearDensityUnits.MilligramsPerMillimeter:
                return (value) * 0.001;
            case LinearDensityUnits.KilogramsPerMillimeter:
                return (value) * 1000;
            case LinearDensityUnits.MicrogramsPerCentimeter:
                return (value * 1e-1) * 0.000001;
            case LinearDensityUnits.MilligramsPerCentimeter:
                return (value * 1e-1) * 0.001;
            case LinearDensityUnits.KilogramsPerCentimeter:
                return (value * 1e-1) * 1000;
            case LinearDensityUnits.MicrogramsPerMeter:
                return (value * 1e-3) * 0.000001;
            case LinearDensityUnits.MilligramsPerMeter:
                return (value * 1e-3) * 0.001;
            case LinearDensityUnits.KilogramsPerMeter:
                return (value * 1e-3) * 1000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the LinearDensity to string.
     * Note! the default format for LinearDensity is KilogramsPerMeter.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the LinearDensity.
     * @returns The string format of the LinearDensity.
     */
    toString(unit = LinearDensityUnits.KilogramsPerMeter) {
        switch (unit) {
            case LinearDensityUnits.GramsPerMillimeter:
                return this.GramsPerMillimeter + ` g/mm`;
            case LinearDensityUnits.GramsPerCentimeter:
                return this.GramsPerCentimeter + ` g/cm`;
            case LinearDensityUnits.GramsPerMeter:
                return this.GramsPerMeter + ` g/m`;
            case LinearDensityUnits.PoundsPerInch:
                return this.PoundsPerInch + ` lb/in`;
            case LinearDensityUnits.PoundsPerFoot:
                return this.PoundsPerFoot + ` lb/ft`;
            case LinearDensityUnits.MicrogramsPerMillimeter:
                return this.MicrogramsPerMillimeter + ` μg/mm`;
            case LinearDensityUnits.MilligramsPerMillimeter:
                return this.MilligramsPerMillimeter + ` mg/mm`;
            case LinearDensityUnits.KilogramsPerMillimeter:
                return this.KilogramsPerMillimeter + ` kg/mm`;
            case LinearDensityUnits.MicrogramsPerCentimeter:
                return this.MicrogramsPerCentimeter + ` μg/cm`;
            case LinearDensityUnits.MilligramsPerCentimeter:
                return this.MilligramsPerCentimeter + ` mg/cm`;
            case LinearDensityUnits.KilogramsPerCentimeter:
                return this.KilogramsPerCentimeter + ` kg/cm`;
            case LinearDensityUnits.MicrogramsPerMeter:
                return this.MicrogramsPerMeter + ` μg/m`;
            case LinearDensityUnits.MilligramsPerMeter:
                return this.MilligramsPerMeter + ` mg/m`;
            case LinearDensityUnits.KilogramsPerMeter:
                return this.KilogramsPerMeter + ` kg/m`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get LinearDensity unit abbreviation.
     * Note! the default abbreviation for LinearDensity is KilogramsPerMeter.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the LinearDensity.
     * @returns The abbreviation string of LinearDensity.
     */
    getUnitAbbreviation(unitAbbreviation = LinearDensityUnits.KilogramsPerMeter) {
        switch (unitAbbreviation) {
            case LinearDensityUnits.GramsPerMillimeter:
                return `g/mm`;
            case LinearDensityUnits.GramsPerCentimeter:
                return `g/cm`;
            case LinearDensityUnits.GramsPerMeter:
                return `g/m`;
            case LinearDensityUnits.PoundsPerInch:
                return `lb/in`;
            case LinearDensityUnits.PoundsPerFoot:
                return `lb/ft`;
            case LinearDensityUnits.MicrogramsPerMillimeter:
                return `μg/mm`;
            case LinearDensityUnits.MilligramsPerMillimeter:
                return `mg/mm`;
            case LinearDensityUnits.KilogramsPerMillimeter:
                return `kg/mm`;
            case LinearDensityUnits.MicrogramsPerCentimeter:
                return `μg/cm`;
            case LinearDensityUnits.MilligramsPerCentimeter:
                return `mg/cm`;
            case LinearDensityUnits.KilogramsPerCentimeter:
                return `kg/cm`;
            case LinearDensityUnits.MicrogramsPerMeter:
                return `μg/m`;
            case LinearDensityUnits.MilligramsPerMeter:
                return `mg/m`;
            case LinearDensityUnits.KilogramsPerMeter:
                return `kg/m`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given LinearDensity are equals to the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns True if the given LinearDensity are equal to the current LinearDensity.
     */
    equals(linearDensity) {
        return this.value === linearDensity.BaseValue;
    }
    /**
     * Compare the given LinearDensity against the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns 0 if they are equal, -1 if the current LinearDensity is less then other, 1 if the current LinearDensity is greater then other.
     */
    compareTo(linearDensity) {
        if (this.value > linearDensity.BaseValue)
            return 1;
        if (this.value < linearDensity.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given LinearDensity with the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns A new LinearDensity instance with the results.
     */
    add(linearDensity) {
        return new LinearDensity(this.value + linearDensity.BaseValue);
    }
    /**
     * Subtract the given LinearDensity with the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns A new LinearDensity instance with the results.
     */
    subtract(linearDensity) {
        return new LinearDensity(this.value - linearDensity.BaseValue);
    }
    /**
     * Multiply the given LinearDensity with the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns A new LinearDensity instance with the results.
     */
    multiply(linearDensity) {
        return new LinearDensity(this.value * linearDensity.BaseValue);
    }
    /**
     * Divide the given LinearDensity with the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns A new LinearDensity instance with the results.
     */
    divide(linearDensity) {
        return new LinearDensity(this.value / linearDensity.BaseValue);
    }
    /**
     * Modulo the given LinearDensity with the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns A new LinearDensity instance with the results.
     */
    modulo(linearDensity) {
        return new LinearDensity(this.value % linearDensity.BaseValue);
    }
    /**
     * Pow the given LinearDensity with the current LinearDensity.
     * @param linearDensity The other LinearDensity.
     * @returns A new LinearDensity instance with the results.
     */
    pow(linearDensity) {
        return new LinearDensity(this.value ** linearDensity.BaseValue);
    }
}
exports.LinearDensity = LinearDensity;
