// unitsnet-js/dist/magneticfield.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagneticField = exports.MagneticFieldUnits = void 0;
/** MagneticFieldUnits enumeration */
var MagneticFieldUnits;
(function (MagneticFieldUnits) {
    /** */
    MagneticFieldUnits[MagneticFieldUnits["Teslas"] = 0] = "Teslas";
    /** */
    MagneticFieldUnits[MagneticFieldUnits["Gausses"] = 1] = "Gausses";
    /** */
    MagneticFieldUnits[MagneticFieldUnits["Nanoteslas"] = 2] = "Nanoteslas";
    /** */
    MagneticFieldUnits[MagneticFieldUnits["Microteslas"] = 3] = "Microteslas";
    /** */
    MagneticFieldUnits[MagneticFieldUnits["Milliteslas"] = 4] = "Milliteslas";
    /** */
    MagneticFieldUnits[MagneticFieldUnits["Milligausses"] = 5] = "Milligausses";
})(MagneticFieldUnits = exports.MagneticFieldUnits || (exports.MagneticFieldUnits = {}));
/** A magnetic field is a force field that is created by moving electric charges (electric currents) and magnetic dipoles, and exerts a force on other nearby moving charges and magnetic dipoles. */
class MagneticField {
    /**
     * Create a new MagneticField.
     * @param value The value.
     * @param fromUnit The ‘MagneticField’ unit to create from.
     * The default unit is Teslas
     */
    constructor(value, fromUnit = MagneticFieldUnits.Teslas) {
        this.teslasLazy = null;
        this.gaussesLazy = null;
        this.nanoteslasLazy = null;
        this.microteslasLazy = null;
        this.milliteslasLazy = null;
        this.milligaussesLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of MagneticField is Teslas.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Teslas() {
        if (this.teslasLazy !== null) {
            return this.teslasLazy;
        }
        return this.teslasLazy = this.convertFromBase(MagneticFieldUnits.Teslas);
    }
    /** */
    get Gausses() {
        if (this.gaussesLazy !== null) {
            return this.gaussesLazy;
        }
        return this.gaussesLazy = this.convertFromBase(MagneticFieldUnits.Gausses);
    }
    /** */
    get Nanoteslas() {
        if (this.nanoteslasLazy !== null) {
            return this.nanoteslasLazy;
        }
        return this.nanoteslasLazy = this.convertFromBase(MagneticFieldUnits.Nanoteslas);
    }
    /** */
    get Microteslas() {
        if (this.microteslasLazy !== null) {
            return this.microteslasLazy;
        }
        return this.microteslasLazy = this.convertFromBase(MagneticFieldUnits.Microteslas);
    }
    /** */
    get Milliteslas() {
        if (this.milliteslasLazy !== null) {
            return this.milliteslasLazy;
        }
        return this.milliteslasLazy = this.convertFromBase(MagneticFieldUnits.Milliteslas);
    }
    /** */
    get Milligausses() {
        if (this.milligaussesLazy !== null) {
            return this.milligaussesLazy;
        }
        return this.milligaussesLazy = this.convertFromBase(MagneticFieldUnits.Milligausses);
    }
    /**
     * Create a new MagneticField instance from a Teslas
     *
     * @param value The unit as Teslas to create a new MagneticField from.
     * @returns The new MagneticField instance.
     */
    static FromTeslas(value) {
        return new MagneticField(value, MagneticFieldUnits.Teslas);
    }
    /**
     * Create a new MagneticField instance from a Gausses
     *
     * @param value The unit as Gausses to create a new MagneticField from.
     * @returns The new MagneticField instance.
     */
    static FromGausses(value) {
        return new MagneticField(value, MagneticFieldUnits.Gausses);
    }
    /**
     * Create a new MagneticField instance from a Nanoteslas
     *
     * @param value The unit as Nanoteslas to create a new MagneticField from.
     * @returns The new MagneticField instance.
     */
    static FromNanoteslas(value) {
        return new MagneticField(value, MagneticFieldUnits.Nanoteslas);
    }
    /**
     * Create a new MagneticField instance from a Microteslas
     *
     * @param value The unit as Microteslas to create a new MagneticField from.
     * @returns The new MagneticField instance.
     */
    static FromMicroteslas(value) {
        return new MagneticField(value, MagneticFieldUnits.Microteslas);
    }
    /**
     * Create a new MagneticField instance from a Milliteslas
     *
     * @param value The unit as Milliteslas to create a new MagneticField from.
     * @returns The new MagneticField instance.
     */
    static FromMilliteslas(value) {
        return new MagneticField(value, MagneticFieldUnits.Milliteslas);
    }
    /**
     * Create a new MagneticField instance from a Milligausses
     *
     * @param value The unit as Milligausses to create a new MagneticField from.
     * @returns The new MagneticField instance.
     */
    static FromMilligausses(value) {
        return new MagneticField(value, MagneticFieldUnits.Milligausses);
    }
    /**
     * Convert MagneticField to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case MagneticFieldUnits.Teslas: return this.Teslas;
            case MagneticFieldUnits.Gausses: return this.Gausses;
            case MagneticFieldUnits.Nanoteslas: return this.Nanoteslas;
            case MagneticFieldUnits.Microteslas: return this.Microteslas;
            case MagneticFieldUnits.Milliteslas: return this.Milliteslas;
            case MagneticFieldUnits.Milligausses: return this.Milligausses;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case MagneticFieldUnits.Teslas:
                return this.value;
            case MagneticFieldUnits.Gausses:
                return this.value * 1e4;
            case MagneticFieldUnits.Nanoteslas:
                return (this.value) / 1e-9;
            case MagneticFieldUnits.Microteslas:
                return (this.value) / 0.000001;
            case MagneticFieldUnits.Milliteslas:
                return (this.value) / 0.001;
            case MagneticFieldUnits.Milligausses:
                return (this.value * 1e4) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case MagneticFieldUnits.Teslas:
                return value;
            case MagneticFieldUnits.Gausses:
                return value / 1e4;
            case MagneticFieldUnits.Nanoteslas:
                return (value) * 1e-9;
            case MagneticFieldUnits.Microteslas:
                return (value) * 0.000001;
            case MagneticFieldUnits.Milliteslas:
                return (value) * 0.001;
            case MagneticFieldUnits.Milligausses:
                return (value / 1e4) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the MagneticField to string.
     * Note! the default format for MagneticField is Teslas.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the MagneticField.
     * @returns The string format of the MagneticField.
     */
    toString(unit = MagneticFieldUnits.Teslas) {
        switch (unit) {
            case MagneticFieldUnits.Teslas:
                return this.Teslas + ` T`;
            case MagneticFieldUnits.Gausses:
                return this.Gausses + ` G`;
            case MagneticFieldUnits.Nanoteslas:
                return this.Nanoteslas + ` nT`;
            case MagneticFieldUnits.Microteslas:
                return this.Microteslas + ` μT`;
            case MagneticFieldUnits.Milliteslas:
                return this.Milliteslas + ` mT`;
            case MagneticFieldUnits.Milligausses:
                return this.Milligausses + ` mG`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get MagneticField unit abbreviation.
     * Note! the default abbreviation for MagneticField is Teslas.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the MagneticField.
     * @returns The abbreviation string of MagneticField.
     */
    getUnitAbbreviation(unitAbbreviation = MagneticFieldUnits.Teslas) {
        switch (unitAbbreviation) {
            case MagneticFieldUnits.Teslas:
                return `T`;
            case MagneticFieldUnits.Gausses:
                return `G`;
            case MagneticFieldUnits.Nanoteslas:
                return `nT`;
            case MagneticFieldUnits.Microteslas:
                return `μT`;
            case MagneticFieldUnits.Milliteslas:
                return `mT`;
            case MagneticFieldUnits.Milligausses:
                return `mG`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given MagneticField are equals to the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns True if the given MagneticField are equal to the current MagneticField.
     */
    equals(magneticField) {
        return this.value === magneticField.BaseValue;
    }
    /**
     * Compare the given MagneticField against the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns 0 if they are equal, -1 if the current MagneticField is less then other, 1 if the current MagneticField is greater then other.
     */
    compareTo(magneticField) {
        if (this.value > magneticField.BaseValue)
            return 1;
        if (this.value < magneticField.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given MagneticField with the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns A new MagneticField instance with the results.
     */
    add(magneticField) {
        return new MagneticField(this.value + magneticField.BaseValue);
    }
    /**
     * Subtract the given MagneticField with the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns A new MagneticField instance with the results.
     */
    subtract(magneticField) {
        return new MagneticField(this.value - magneticField.BaseValue);
    }
    /**
     * Multiply the given MagneticField with the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns A new MagneticField instance with the results.
     */
    multiply(magneticField) {
        return new MagneticField(this.value * magneticField.BaseValue);
    }
    /**
     * Divide the given MagneticField with the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns A new MagneticField instance with the results.
     */
    divide(magneticField) {
        return new MagneticField(this.value / magneticField.BaseValue);
    }
    /**
     * Modulo the given MagneticField with the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns A new MagneticField instance with the results.
     */
    modulo(magneticField) {
        return new MagneticField(this.value % magneticField.BaseValue);
    }
    /**
     * Pow the given MagneticField with the current MagneticField.
     * @param magneticField The other MagneticField.
     * @returns A new MagneticField instance with the results.
     */
    pow(magneticField) {
        return new MagneticField(this.value ** magneticField.BaseValue);
    }
}
exports.MagneticField = MagneticField;
