// unitsnet-js/dist/molality.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Molality = exports.MolalityUnits = void 0;
/** MolalityUnits enumeration */
var MolalityUnits;
(function (MolalityUnits) {
    /** */
    MolalityUnits[MolalityUnits["MolesPerKilogram"] = 0] = "MolesPerKilogram";
    /** */
    MolalityUnits[MolalityUnits["MolesPerGram"] = 1] = "MolesPerGram";
})(MolalityUnits = exports.MolalityUnits || (exports.MolalityUnits = {}));
/** Molality is a measure of the amount of solute in a solution relative to a given mass of solvent. */
class Molality {
    /**
     * Create a new Molality.
     * @param value The value.
     * @param fromUnit The ‘Molality’ unit to create from.
     * The default unit is MolesPerKilogram
     */
    constructor(value, fromUnit = MolalityUnits.MolesPerKilogram) {
        this.molesperkilogramLazy = null;
        this.molespergramLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of Molality is MolesPerKilogram.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get MolesPerKilogram() {
        if (this.molesperkilogramLazy !== null) {
            return this.molesperkilogramLazy;
        }
        return this.molesperkilogramLazy = this.convertFromBase(MolalityUnits.MolesPerKilogram);
    }
    /** */
    get MolesPerGram() {
        if (this.molespergramLazy !== null) {
            return this.molespergramLazy;
        }
        return this.molespergramLazy = this.convertFromBase(MolalityUnits.MolesPerGram);
    }
    /**
     * Create a new Molality instance from a MolesPerKilogram
     *
     * @param value The unit as MolesPerKilogram to create a new Molality from.
     * @returns The new Molality instance.
     */
    static FromMolesPerKilogram(value) {
        return new Molality(value, MolalityUnits.MolesPerKilogram);
    }
    /**
     * Create a new Molality instance from a MolesPerGram
     *
     * @param value The unit as MolesPerGram to create a new Molality from.
     * @returns The new Molality instance.
     */
    static FromMolesPerGram(value) {
        return new Molality(value, MolalityUnits.MolesPerGram);
    }
    /**
     * Convert Molality to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case MolalityUnits.MolesPerKilogram: return this.MolesPerKilogram;
            case MolalityUnits.MolesPerGram: return this.MolesPerGram;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case MolalityUnits.MolesPerKilogram:
                return this.value;
            case MolalityUnits.MolesPerGram:
                return this.value * 1e-3;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case MolalityUnits.MolesPerKilogram:
                return value;
            case MolalityUnits.MolesPerGram:
                return value / 1e-3;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the Molality to string.
     * Note! the default format for Molality is MolesPerKilogram.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the Molality.
     * @returns The string format of the Molality.
     */
    toString(unit = MolalityUnits.MolesPerKilogram) {
        switch (unit) {
            case MolalityUnits.MolesPerKilogram:
                return this.MolesPerKilogram + ` mol/kg`;
            case MolalityUnits.MolesPerGram:
                return this.MolesPerGram + ` mol/g`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get Molality unit abbreviation.
     * Note! the default abbreviation for Molality is MolesPerKilogram.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the Molality.
     * @returns The abbreviation string of Molality.
     */
    getUnitAbbreviation(unitAbbreviation = MolalityUnits.MolesPerKilogram) {
        switch (unitAbbreviation) {
            case MolalityUnits.MolesPerKilogram:
                return `mol/kg`;
            case MolalityUnits.MolesPerGram:
                return `mol/g`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given Molality are equals to the current Molality.
     * @param molality The other Molality.
     * @returns True if the given Molality are equal to the current Molality.
     */
    equals(molality) {
        return this.value === molality.BaseValue;
    }
    /**
     * Compare the given Molality against the current Molality.
     * @param molality The other Molality.
     * @returns 0 if they are equal, -1 if the current Molality is less then other, 1 if the current Molality is greater then other.
     */
    compareTo(molality) {
        if (this.value > molality.BaseValue)
            return 1;
        if (this.value < molality.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given Molality with the current Molality.
     * @param molality The other Molality.
     * @returns A new Molality instance with the results.
     */
    add(molality) {
        return new Molality(this.value + molality.BaseValue);
    }
    /**
     * Subtract the given Molality with the current Molality.
     * @param molality The other Molality.
     * @returns A new Molality instance with the results.
     */
    subtract(molality) {
        return new Molality(this.value - molality.BaseValue);
    }
    /**
     * Multiply the given Molality with the current Molality.
     * @param molality The other Molality.
     * @returns A new Molality instance with the results.
     */
    multiply(molality) {
        return new Molality(this.value * molality.BaseValue);
    }
    /**
     * Divide the given Molality with the current Molality.
     * @param molality The other Molality.
     * @returns A new Molality instance with the results.
     */
    divide(molality) {
        return new Molality(this.value / molality.BaseValue);
    }
    /**
     * Modulo the given Molality with the current Molality.
     * @param molality The other Molality.
     * @returns A new Molality instance with the results.
     */
    modulo(molality) {
        return new Molality(this.value % molality.BaseValue);
    }
    /**
     * Pow the given Molality with the current Molality.
     * @param molality The other Molality.
     * @returns A new Molality instance with the results.
     */
    pow(molality) {
        return new Molality(this.value ** molality.BaseValue);
    }
}
exports.Molality = Molality;
