// unitsnet-js/dist/porousmediumpermeability.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PorousMediumPermeability = exports.PorousMediumPermeabilityUnits = void 0;
/** PorousMediumPermeabilityUnits enumeration */
var PorousMediumPermeabilityUnits;
(function (PorousMediumPermeabilityUnits) {
    /** */
    PorousMediumPermeabilityUnits[PorousMediumPermeabilityUnits["Darcys"] = 0] = "Darcys";
    /** */
    PorousMediumPermeabilityUnits[PorousMediumPermeabilityUnits["SquareMeters"] = 1] = "SquareMeters";
    /** */
    PorousMediumPermeabilityUnits[PorousMediumPermeabilityUnits["SquareCentimeters"] = 2] = "SquareCentimeters";
    /** */
    PorousMediumPermeabilityUnits[PorousMediumPermeabilityUnits["Microdarcys"] = 3] = "Microdarcys";
    /** */
    PorousMediumPermeabilityUnits[PorousMediumPermeabilityUnits["Millidarcys"] = 4] = "Millidarcys";
})(PorousMediumPermeabilityUnits = exports.PorousMediumPermeabilityUnits || (exports.PorousMediumPermeabilityUnits = {}));
/** In fluid mechanics, permeability is the measure of the ability of a porous material to allow fluids to pass through it. */
class PorousMediumPermeability {
    /**
     * Create a new PorousMediumPermeability.
     * @param value The value.
     * @param fromUnit The ‘PorousMediumPermeability’ unit to create from.
     * The default unit is SquareMeters
     */
    constructor(value, fromUnit = PorousMediumPermeabilityUnits.SquareMeters) {
        this.darcysLazy = null;
        this.squaremetersLazy = null;
        this.squarecentimetersLazy = null;
        this.microdarcysLazy = null;
        this.millidarcysLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of PorousMediumPermeability is SquareMeters.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Darcys() {
        if (this.darcysLazy !== null) {
            return this.darcysLazy;
        }
        return this.darcysLazy = this.convertFromBase(PorousMediumPermeabilityUnits.Darcys);
    }
    /** */
    get SquareMeters() {
        if (this.squaremetersLazy !== null) {
            return this.squaremetersLazy;
        }
        return this.squaremetersLazy = this.convertFromBase(PorousMediumPermeabilityUnits.SquareMeters);
    }
    /** */
    get SquareCentimeters() {
        if (this.squarecentimetersLazy !== null) {
            return this.squarecentimetersLazy;
        }
        return this.squarecentimetersLazy = this.convertFromBase(PorousMediumPermeabilityUnits.SquareCentimeters);
    }
    /** */
    get Microdarcys() {
        if (this.microdarcysLazy !== null) {
            return this.microdarcysLazy;
        }
        return this.microdarcysLazy = this.convertFromBase(PorousMediumPermeabilityUnits.Microdarcys);
    }
    /** */
    get Millidarcys() {
        if (this.millidarcysLazy !== null) {
            return this.millidarcysLazy;
        }
        return this.millidarcysLazy = this.convertFromBase(PorousMediumPermeabilityUnits.Millidarcys);
    }
    /**
     * Create a new PorousMediumPermeability instance from a Darcys
     *
     * @param value The unit as Darcys to create a new PorousMediumPermeability from.
     * @returns The new PorousMediumPermeability instance.
     */
    static FromDarcys(value) {
        return new PorousMediumPermeability(value, PorousMediumPermeabilityUnits.Darcys);
    }
    /**
     * Create a new PorousMediumPermeability instance from a SquareMeters
     *
     * @param value The unit as SquareMeters to create a new PorousMediumPermeability from.
     * @returns The new PorousMediumPermeability instance.
     */
    static FromSquareMeters(value) {
        return new PorousMediumPermeability(value, PorousMediumPermeabilityUnits.SquareMeters);
    }
    /**
     * Create a new PorousMediumPermeability instance from a SquareCentimeters
     *
     * @param value The unit as SquareCentimeters to create a new PorousMediumPermeability from.
     * @returns The new PorousMediumPermeability instance.
     */
    static FromSquareCentimeters(value) {
        return new PorousMediumPermeability(value, PorousMediumPermeabilityUnits.SquareCentimeters);
    }
    /**
     * Create a new PorousMediumPermeability instance from a Microdarcys
     *
     * @param value The unit as Microdarcys to create a new PorousMediumPermeability from.
     * @returns The new PorousMediumPermeability instance.
     */
    static FromMicrodarcys(value) {
        return new PorousMediumPermeability(value, PorousMediumPermeabilityUnits.Microdarcys);
    }
    /**
     * Create a new PorousMediumPermeability instance from a Millidarcys
     *
     * @param value The unit as Millidarcys to create a new PorousMediumPermeability from.
     * @returns The new PorousMediumPermeability instance.
     */
    static FromMillidarcys(value) {
        return new PorousMediumPermeability(value, PorousMediumPermeabilityUnits.Millidarcys);
    }
    /**
     * Convert PorousMediumPermeability to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case PorousMediumPermeabilityUnits.Darcys: return this.Darcys;
            case PorousMediumPermeabilityUnits.SquareMeters: return this.SquareMeters;
            case PorousMediumPermeabilityUnits.SquareCentimeters: return this.SquareCentimeters;
            case PorousMediumPermeabilityUnits.Microdarcys: return this.Microdarcys;
            case PorousMediumPermeabilityUnits.Millidarcys: return this.Millidarcys;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case PorousMediumPermeabilityUnits.Darcys:
                return this.value / 9.869233e-13;
            case PorousMediumPermeabilityUnits.SquareMeters:
                return this.value;
            case PorousMediumPermeabilityUnits.SquareCentimeters:
                return this.value / 1e-4;
            case PorousMediumPermeabilityUnits.Microdarcys:
                return (this.value / 9.869233e-13) / 0.000001;
            case PorousMediumPermeabilityUnits.Millidarcys:
                return (this.value / 9.869233e-13) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case PorousMediumPermeabilityUnits.Darcys:
                return value * 9.869233e-13;
            case PorousMediumPermeabilityUnits.SquareMeters:
                return value;
            case PorousMediumPermeabilityUnits.SquareCentimeters:
                return value * 1e-4;
            case PorousMediumPermeabilityUnits.Microdarcys:
                return (value * 9.869233e-13) * 0.000001;
            case PorousMediumPermeabilityUnits.Millidarcys:
                return (value * 9.869233e-13) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the PorousMediumPermeability to string.
     * Note! the default format for PorousMediumPermeability is SquareMeters.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the PorousMediumPermeability.
     * @returns The string format of the PorousMediumPermeability.
     */
    toString(unit = PorousMediumPermeabilityUnits.SquareMeters) {
        switch (unit) {
            case PorousMediumPermeabilityUnits.Darcys:
                return this.Darcys + ` D`;
            case PorousMediumPermeabilityUnits.SquareMeters:
                return this.SquareMeters + ` m²`;
            case PorousMediumPermeabilityUnits.SquareCentimeters:
                return this.SquareCentimeters + ` cm²`;
            case PorousMediumPermeabilityUnits.Microdarcys:
                return this.Microdarcys + ` μD`;
            case PorousMediumPermeabilityUnits.Millidarcys:
                return this.Millidarcys + ` mD`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get PorousMediumPermeability unit abbreviation.
     * Note! the default abbreviation for PorousMediumPermeability is SquareMeters.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the PorousMediumPermeability.
     * @returns The abbreviation string of PorousMediumPermeability.
     */
    getUnitAbbreviation(unitAbbreviation = PorousMediumPermeabilityUnits.SquareMeters) {
        switch (unitAbbreviation) {
            case PorousMediumPermeabilityUnits.Darcys:
                return `D`;
            case PorousMediumPermeabilityUnits.SquareMeters:
                return `m²`;
            case PorousMediumPermeabilityUnits.SquareCentimeters:
                return `cm²`;
            case PorousMediumPermeabilityUnits.Microdarcys:
                return `μD`;
            case PorousMediumPermeabilityUnits.Millidarcys:
                return `mD`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given PorousMediumPermeability are equals to the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns True if the given PorousMediumPermeability are equal to the current PorousMediumPermeability.
     */
    equals(porousMediumPermeability) {
        return this.value === porousMediumPermeability.BaseValue;
    }
    /**
     * Compare the given PorousMediumPermeability against the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns 0 if they are equal, -1 if the current PorousMediumPermeability is less then other, 1 if the current PorousMediumPermeability is greater then other.
     */
    compareTo(porousMediumPermeability) {
        if (this.value > porousMediumPermeability.BaseValue)
            return 1;
        if (this.value < porousMediumPermeability.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given PorousMediumPermeability with the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns A new PorousMediumPermeability instance with the results.
     */
    add(porousMediumPermeability) {
        return new PorousMediumPermeability(this.value + porousMediumPermeability.BaseValue);
    }
    /**
     * Subtract the given PorousMediumPermeability with the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns A new PorousMediumPermeability instance with the results.
     */
    subtract(porousMediumPermeability) {
        return new PorousMediumPermeability(this.value - porousMediumPermeability.BaseValue);
    }
    /**
     * Multiply the given PorousMediumPermeability with the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns A new PorousMediumPermeability instance with the results.
     */
    multiply(porousMediumPermeability) {
        return new PorousMediumPermeability(this.value * porousMediumPermeability.BaseValue);
    }
    /**
     * Divide the given PorousMediumPermeability with the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns A new PorousMediumPermeability instance with the results.
     */
    divide(porousMediumPermeability) {
        return new PorousMediumPermeability(this.value / porousMediumPermeability.BaseValue);
    }
    /**
     * Modulo the given PorousMediumPermeability with the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns A new PorousMediumPermeability instance with the results.
     */
    modulo(porousMediumPermeability) {
        return new PorousMediumPermeability(this.value % porousMediumPermeability.BaseValue);
    }
    /**
     * Pow the given PorousMediumPermeability with the current PorousMediumPermeability.
     * @param porousMediumPermeability The other PorousMediumPermeability.
     * @returns A new PorousMediumPermeability instance with the results.
     */
    pow(porousMediumPermeability) {
        return new PorousMediumPermeability(this.value ** porousMediumPermeability.BaseValue);
    }
}
exports.PorousMediumPermeability = PorousMediumPermeability;
