// unitsnet-js/dist/radiationexposure.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RadiationExposure = exports.RadiationExposureUnits = void 0;
/** RadiationExposureUnits enumeration */
var RadiationExposureUnits;
(function (RadiationExposureUnits) {
    /** */
    RadiationExposureUnits[RadiationExposureUnits["CoulombsPerKilogram"] = 0] = "CoulombsPerKilogram";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["Roentgens"] = 1] = "Roentgens";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["PicocoulombsPerKilogram"] = 2] = "PicocoulombsPerKilogram";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["NanocoulombsPerKilogram"] = 3] = "NanocoulombsPerKilogram";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["MicrocoulombsPerKilogram"] = 4] = "MicrocoulombsPerKilogram";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["MillicoulombsPerKilogram"] = 5] = "MillicoulombsPerKilogram";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["Microroentgens"] = 6] = "Microroentgens";
    /** */
    RadiationExposureUnits[RadiationExposureUnits["Milliroentgens"] = 7] = "Milliroentgens";
})(RadiationExposureUnits = exports.RadiationExposureUnits || (exports.RadiationExposureUnits = {}));
/** Radiation exposure is a measure of the ionization of air due to ionizing radiation from photons. */
class RadiationExposure {
    /**
     * Create a new RadiationExposure.
     * @param value The value.
     * @param fromUnit The ‘RadiationExposure’ unit to create from.
     * The default unit is CoulombsPerKilogram
     */
    constructor(value, fromUnit = RadiationExposureUnits.CoulombsPerKilogram) {
        this.coulombsperkilogramLazy = null;
        this.roentgensLazy = null;
        this.picocoulombsperkilogramLazy = null;
        this.nanocoulombsperkilogramLazy = null;
        this.microcoulombsperkilogramLazy = null;
        this.millicoulombsperkilogramLazy = null;
        this.microroentgensLazy = null;
        this.milliroentgensLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of RadiationExposure is CoulombsPerKilogram.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get CoulombsPerKilogram() {
        if (this.coulombsperkilogramLazy !== null) {
            return this.coulombsperkilogramLazy;
        }
        return this.coulombsperkilogramLazy = this.convertFromBase(RadiationExposureUnits.CoulombsPerKilogram);
    }
    /** */
    get Roentgens() {
        if (this.roentgensLazy !== null) {
            return this.roentgensLazy;
        }
        return this.roentgensLazy = this.convertFromBase(RadiationExposureUnits.Roentgens);
    }
    /** */
    get PicocoulombsPerKilogram() {
        if (this.picocoulombsperkilogramLazy !== null) {
            return this.picocoulombsperkilogramLazy;
        }
        return this.picocoulombsperkilogramLazy = this.convertFromBase(RadiationExposureUnits.PicocoulombsPerKilogram);
    }
    /** */
    get NanocoulombsPerKilogram() {
        if (this.nanocoulombsperkilogramLazy !== null) {
            return this.nanocoulombsperkilogramLazy;
        }
        return this.nanocoulombsperkilogramLazy = this.convertFromBase(RadiationExposureUnits.NanocoulombsPerKilogram);
    }
    /** */
    get MicrocoulombsPerKilogram() {
        if (this.microcoulombsperkilogramLazy !== null) {
            return this.microcoulombsperkilogramLazy;
        }
        return this.microcoulombsperkilogramLazy = this.convertFromBase(RadiationExposureUnits.MicrocoulombsPerKilogram);
    }
    /** */
    get MillicoulombsPerKilogram() {
        if (this.millicoulombsperkilogramLazy !== null) {
            return this.millicoulombsperkilogramLazy;
        }
        return this.millicoulombsperkilogramLazy = this.convertFromBase(RadiationExposureUnits.MillicoulombsPerKilogram);
    }
    /** */
    get Microroentgens() {
        if (this.microroentgensLazy !== null) {
            return this.microroentgensLazy;
        }
        return this.microroentgensLazy = this.convertFromBase(RadiationExposureUnits.Microroentgens);
    }
    /** */
    get Milliroentgens() {
        if (this.milliroentgensLazy !== null) {
            return this.milliroentgensLazy;
        }
        return this.milliroentgensLazy = this.convertFromBase(RadiationExposureUnits.Milliroentgens);
    }
    /**
     * Create a new RadiationExposure instance from a CoulombsPerKilogram
     *
     * @param value The unit as CoulombsPerKilogram to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromCoulombsPerKilogram(value) {
        return new RadiationExposure(value, RadiationExposureUnits.CoulombsPerKilogram);
    }
    /**
     * Create a new RadiationExposure instance from a Roentgens
     *
     * @param value The unit as Roentgens to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromRoentgens(value) {
        return new RadiationExposure(value, RadiationExposureUnits.Roentgens);
    }
    /**
     * Create a new RadiationExposure instance from a PicocoulombsPerKilogram
     *
     * @param value The unit as PicocoulombsPerKilogram to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromPicocoulombsPerKilogram(value) {
        return new RadiationExposure(value, RadiationExposureUnits.PicocoulombsPerKilogram);
    }
    /**
     * Create a new RadiationExposure instance from a NanocoulombsPerKilogram
     *
     * @param value The unit as NanocoulombsPerKilogram to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromNanocoulombsPerKilogram(value) {
        return new RadiationExposure(value, RadiationExposureUnits.NanocoulombsPerKilogram);
    }
    /**
     * Create a new RadiationExposure instance from a MicrocoulombsPerKilogram
     *
     * @param value The unit as MicrocoulombsPerKilogram to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromMicrocoulombsPerKilogram(value) {
        return new RadiationExposure(value, RadiationExposureUnits.MicrocoulombsPerKilogram);
    }
    /**
     * Create a new RadiationExposure instance from a MillicoulombsPerKilogram
     *
     * @param value The unit as MillicoulombsPerKilogram to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromMillicoulombsPerKilogram(value) {
        return new RadiationExposure(value, RadiationExposureUnits.MillicoulombsPerKilogram);
    }
    /**
     * Create a new RadiationExposure instance from a Microroentgens
     *
     * @param value The unit as Microroentgens to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromMicroroentgens(value) {
        return new RadiationExposure(value, RadiationExposureUnits.Microroentgens);
    }
    /**
     * Create a new RadiationExposure instance from a Milliroentgens
     *
     * @param value The unit as Milliroentgens to create a new RadiationExposure from.
     * @returns The new RadiationExposure instance.
     */
    static FromMilliroentgens(value) {
        return new RadiationExposure(value, RadiationExposureUnits.Milliroentgens);
    }
    /**
     * Convert RadiationExposure to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case RadiationExposureUnits.CoulombsPerKilogram: return this.CoulombsPerKilogram;
            case RadiationExposureUnits.Roentgens: return this.Roentgens;
            case RadiationExposureUnits.PicocoulombsPerKilogram: return this.PicocoulombsPerKilogram;
            case RadiationExposureUnits.NanocoulombsPerKilogram: return this.NanocoulombsPerKilogram;
            case RadiationExposureUnits.MicrocoulombsPerKilogram: return this.MicrocoulombsPerKilogram;
            case RadiationExposureUnits.MillicoulombsPerKilogram: return this.MillicoulombsPerKilogram;
            case RadiationExposureUnits.Microroentgens: return this.Microroentgens;
            case RadiationExposureUnits.Milliroentgens: return this.Milliroentgens;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case RadiationExposureUnits.CoulombsPerKilogram:
                return this.value;
            case RadiationExposureUnits.Roentgens:
                return this.value / 2.58e-4;
            case RadiationExposureUnits.PicocoulombsPerKilogram:
                return (this.value) / 1e-12;
            case RadiationExposureUnits.NanocoulombsPerKilogram:
                return (this.value) / 1e-9;
            case RadiationExposureUnits.MicrocoulombsPerKilogram:
                return (this.value) / 0.000001;
            case RadiationExposureUnits.MillicoulombsPerKilogram:
                return (this.value) / 0.001;
            case RadiationExposureUnits.Microroentgens:
                return (this.value / 2.58e-4) / 0.000001;
            case RadiationExposureUnits.Milliroentgens:
                return (this.value / 2.58e-4) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case RadiationExposureUnits.CoulombsPerKilogram:
                return value;
            case RadiationExposureUnits.Roentgens:
                return value * 2.58e-4;
            case RadiationExposureUnits.PicocoulombsPerKilogram:
                return (value) * 1e-12;
            case RadiationExposureUnits.NanocoulombsPerKilogram:
                return (value) * 1e-9;
            case RadiationExposureUnits.MicrocoulombsPerKilogram:
                return (value) * 0.000001;
            case RadiationExposureUnits.MillicoulombsPerKilogram:
                return (value) * 0.001;
            case RadiationExposureUnits.Microroentgens:
                return (value * 2.58e-4) * 0.000001;
            case RadiationExposureUnits.Milliroentgens:
                return (value * 2.58e-4) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the RadiationExposure to string.
     * Note! the default format for RadiationExposure is CoulombsPerKilogram.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the RadiationExposure.
     * @returns The string format of the RadiationExposure.
     */
    toString(unit = RadiationExposureUnits.CoulombsPerKilogram) {
        switch (unit) {
            case RadiationExposureUnits.CoulombsPerKilogram:
                return this.CoulombsPerKilogram + ` C/kg`;
            case RadiationExposureUnits.Roentgens:
                return this.Roentgens + ` R`;
            case RadiationExposureUnits.PicocoulombsPerKilogram:
                return this.PicocoulombsPerKilogram + ` pC/kg`;
            case RadiationExposureUnits.NanocoulombsPerKilogram:
                return this.NanocoulombsPerKilogram + ` nC/kg`;
            case RadiationExposureUnits.MicrocoulombsPerKilogram:
                return this.MicrocoulombsPerKilogram + ` μC/kg`;
            case RadiationExposureUnits.MillicoulombsPerKilogram:
                return this.MillicoulombsPerKilogram + ` mC/kg`;
            case RadiationExposureUnits.Microroentgens:
                return this.Microroentgens + ` μR`;
            case RadiationExposureUnits.Milliroentgens:
                return this.Milliroentgens + ` mR`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get RadiationExposure unit abbreviation.
     * Note! the default abbreviation for RadiationExposure is CoulombsPerKilogram.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the RadiationExposure.
     * @returns The abbreviation string of RadiationExposure.
     */
    getUnitAbbreviation(unitAbbreviation = RadiationExposureUnits.CoulombsPerKilogram) {
        switch (unitAbbreviation) {
            case RadiationExposureUnits.CoulombsPerKilogram:
                return `C/kg`;
            case RadiationExposureUnits.Roentgens:
                return `R`;
            case RadiationExposureUnits.PicocoulombsPerKilogram:
                return `pC/kg`;
            case RadiationExposureUnits.NanocoulombsPerKilogram:
                return `nC/kg`;
            case RadiationExposureUnits.MicrocoulombsPerKilogram:
                return `μC/kg`;
            case RadiationExposureUnits.MillicoulombsPerKilogram:
                return `mC/kg`;
            case RadiationExposureUnits.Microroentgens:
                return `μR`;
            case RadiationExposureUnits.Milliroentgens:
                return `mR`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given RadiationExposure are equals to the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns True if the given RadiationExposure are equal to the current RadiationExposure.
     */
    equals(radiationExposure) {
        return this.value === radiationExposure.BaseValue;
    }
    /**
     * Compare the given RadiationExposure against the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns 0 if they are equal, -1 if the current RadiationExposure is less then other, 1 if the current RadiationExposure is greater then other.
     */
    compareTo(radiationExposure) {
        if (this.value > radiationExposure.BaseValue)
            return 1;
        if (this.value < radiationExposure.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given RadiationExposure with the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns A new RadiationExposure instance with the results.
     */
    add(radiationExposure) {
        return new RadiationExposure(this.value + radiationExposure.BaseValue);
    }
    /**
     * Subtract the given RadiationExposure with the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns A new RadiationExposure instance with the results.
     */
    subtract(radiationExposure) {
        return new RadiationExposure(this.value - radiationExposure.BaseValue);
    }
    /**
     * Multiply the given RadiationExposure with the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns A new RadiationExposure instance with the results.
     */
    multiply(radiationExposure) {
        return new RadiationExposure(this.value * radiationExposure.BaseValue);
    }
    /**
     * Divide the given RadiationExposure with the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns A new RadiationExposure instance with the results.
     */
    divide(radiationExposure) {
        return new RadiationExposure(this.value / radiationExposure.BaseValue);
    }
    /**
     * Modulo the given RadiationExposure with the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns A new RadiationExposure instance with the results.
     */
    modulo(radiationExposure) {
        return new RadiationExposure(this.value % radiationExposure.BaseValue);
    }
    /**
     * Pow the given RadiationExposure with the current RadiationExposure.
     * @param radiationExposure The other RadiationExposure.
     * @returns A new RadiationExposure instance with the results.
     */
    pow(radiationExposure) {
        return new RadiationExposure(this.value ** radiationExposure.BaseValue);
    }
}
exports.RadiationExposure = RadiationExposure;
