// unitsnet-js/dist/radiationequivalentdose.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RadiationEquivalentDose = exports.RadiationEquivalentDoseUnits = void 0;
/** RadiationEquivalentDoseUnits enumeration */
var RadiationEquivalentDoseUnits;
(function (RadiationEquivalentDoseUnits) {
    /** The sievert is a unit in the International System of Units (SI) intended to represent the stochastic health risk of ionizing radiation, which is defined as the probability of causing radiation-induced cancer and genetic damage. */
    RadiationEquivalentDoseUnits[RadiationEquivalentDoseUnits["Sieverts"] = 0] = "Sieverts";
    /** */
    RadiationEquivalentDoseUnits[RadiationEquivalentDoseUnits["RoentgensEquivalentMan"] = 1] = "RoentgensEquivalentMan";
    /** */
    RadiationEquivalentDoseUnits[RadiationEquivalentDoseUnits["Nanosieverts"] = 2] = "Nanosieverts";
    /** */
    RadiationEquivalentDoseUnits[RadiationEquivalentDoseUnits["Microsieverts"] = 3] = "Microsieverts";
    /** */
    RadiationEquivalentDoseUnits[RadiationEquivalentDoseUnits["Millisieverts"] = 4] = "Millisieverts";
    /** */
    RadiationEquivalentDoseUnits[RadiationEquivalentDoseUnits["MilliroentgensEquivalentMan"] = 5] = "MilliroentgensEquivalentMan";
})(RadiationEquivalentDoseUnits = exports.RadiationEquivalentDoseUnits || (exports.RadiationEquivalentDoseUnits = {}));
/** Equivalent dose is a dose quantity representing the stochastic health effects of low levels of ionizing radiation on the human body which represents the probability of radiation-induced cancer and genetic damage. */
class RadiationEquivalentDose {
    /**
     * Create a new RadiationEquivalentDose.
     * @param value The value.
     * @param fromUnit The ‘RadiationEquivalentDose’ unit to create from.
     * The default unit is Sieverts
     */
    constructor(value, fromUnit = RadiationEquivalentDoseUnits.Sieverts) {
        this.sievertsLazy = null;
        this.roentgensequivalentmanLazy = null;
        this.nanosievertsLazy = null;
        this.microsievertsLazy = null;
        this.millisievertsLazy = null;
        this.milliroentgensequivalentmanLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of RadiationEquivalentDose is Sieverts.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** The sievert is a unit in the International System of Units (SI) intended to represent the stochastic health risk of ionizing radiation, which is defined as the probability of causing radiation-induced cancer and genetic damage. */
    get Sieverts() {
        if (this.sievertsLazy !== null) {
            return this.sievertsLazy;
        }
        return this.sievertsLazy = this.convertFromBase(RadiationEquivalentDoseUnits.Sieverts);
    }
    /** */
    get RoentgensEquivalentMan() {
        if (this.roentgensequivalentmanLazy !== null) {
            return this.roentgensequivalentmanLazy;
        }
        return this.roentgensequivalentmanLazy = this.convertFromBase(RadiationEquivalentDoseUnits.RoentgensEquivalentMan);
    }
    /** */
    get Nanosieverts() {
        if (this.nanosievertsLazy !== null) {
            return this.nanosievertsLazy;
        }
        return this.nanosievertsLazy = this.convertFromBase(RadiationEquivalentDoseUnits.Nanosieverts);
    }
    /** */
    get Microsieverts() {
        if (this.microsievertsLazy !== null) {
            return this.microsievertsLazy;
        }
        return this.microsievertsLazy = this.convertFromBase(RadiationEquivalentDoseUnits.Microsieverts);
    }
    /** */
    get Millisieverts() {
        if (this.millisievertsLazy !== null) {
            return this.millisievertsLazy;
        }
        return this.millisievertsLazy = this.convertFromBase(RadiationEquivalentDoseUnits.Millisieverts);
    }
    /** */
    get MilliroentgensEquivalentMan() {
        if (this.milliroentgensequivalentmanLazy !== null) {
            return this.milliroentgensequivalentmanLazy;
        }
        return this.milliroentgensequivalentmanLazy = this.convertFromBase(RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan);
    }
    /**
     * Create a new RadiationEquivalentDose instance from a Sieverts
     * The sievert is a unit in the International System of Units (SI) intended to represent the stochastic health risk of ionizing radiation, which is defined as the probability of causing radiation-induced cancer and genetic damage.
     * @param value The unit as Sieverts to create a new RadiationEquivalentDose from.
     * @returns The new RadiationEquivalentDose instance.
     */
    static FromSieverts(value) {
        return new RadiationEquivalentDose(value, RadiationEquivalentDoseUnits.Sieverts);
    }
    /**
     * Create a new RadiationEquivalentDose instance from a RoentgensEquivalentMan
     *
     * @param value The unit as RoentgensEquivalentMan to create a new RadiationEquivalentDose from.
     * @returns The new RadiationEquivalentDose instance.
     */
    static FromRoentgensEquivalentMan(value) {
        return new RadiationEquivalentDose(value, RadiationEquivalentDoseUnits.RoentgensEquivalentMan);
    }
    /**
     * Create a new RadiationEquivalentDose instance from a Nanosieverts
     *
     * @param value The unit as Nanosieverts to create a new RadiationEquivalentDose from.
     * @returns The new RadiationEquivalentDose instance.
     */
    static FromNanosieverts(value) {
        return new RadiationEquivalentDose(value, RadiationEquivalentDoseUnits.Nanosieverts);
    }
    /**
     * Create a new RadiationEquivalentDose instance from a Microsieverts
     *
     * @param value The unit as Microsieverts to create a new RadiationEquivalentDose from.
     * @returns The new RadiationEquivalentDose instance.
     */
    static FromMicrosieverts(value) {
        return new RadiationEquivalentDose(value, RadiationEquivalentDoseUnits.Microsieverts);
    }
    /**
     * Create a new RadiationEquivalentDose instance from a Millisieverts
     *
     * @param value The unit as Millisieverts to create a new RadiationEquivalentDose from.
     * @returns The new RadiationEquivalentDose instance.
     */
    static FromMillisieverts(value) {
        return new RadiationEquivalentDose(value, RadiationEquivalentDoseUnits.Millisieverts);
    }
    /**
     * Create a new RadiationEquivalentDose instance from a MilliroentgensEquivalentMan
     *
     * @param value The unit as MilliroentgensEquivalentMan to create a new RadiationEquivalentDose from.
     * @returns The new RadiationEquivalentDose instance.
     */
    static FromMilliroentgensEquivalentMan(value) {
        return new RadiationEquivalentDose(value, RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan);
    }
    /**
     * Convert RadiationEquivalentDose to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case RadiationEquivalentDoseUnits.Sieverts: return this.Sieverts;
            case RadiationEquivalentDoseUnits.RoentgensEquivalentMan: return this.RoentgensEquivalentMan;
            case RadiationEquivalentDoseUnits.Nanosieverts: return this.Nanosieverts;
            case RadiationEquivalentDoseUnits.Microsieverts: return this.Microsieverts;
            case RadiationEquivalentDoseUnits.Millisieverts: return this.Millisieverts;
            case RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan: return this.MilliroentgensEquivalentMan;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case RadiationEquivalentDoseUnits.Sieverts:
                return this.value;
            case RadiationEquivalentDoseUnits.RoentgensEquivalentMan:
                return this.value * 100;
            case RadiationEquivalentDoseUnits.Nanosieverts:
                return (this.value) / 1e-9;
            case RadiationEquivalentDoseUnits.Microsieverts:
                return (this.value) / 0.000001;
            case RadiationEquivalentDoseUnits.Millisieverts:
                return (this.value) / 0.001;
            case RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan:
                return (this.value * 100) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case RadiationEquivalentDoseUnits.Sieverts:
                return value;
            case RadiationEquivalentDoseUnits.RoentgensEquivalentMan:
                return value / 100;
            case RadiationEquivalentDoseUnits.Nanosieverts:
                return (value) * 1e-9;
            case RadiationEquivalentDoseUnits.Microsieverts:
                return (value) * 0.000001;
            case RadiationEquivalentDoseUnits.Millisieverts:
                return (value) * 0.001;
            case RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan:
                return (value / 100) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the RadiationEquivalentDose to string.
     * Note! the default format for RadiationEquivalentDose is Sieverts.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the RadiationEquivalentDose.
     * @returns The string format of the RadiationEquivalentDose.
     */
    toString(unit = RadiationEquivalentDoseUnits.Sieverts) {
        switch (unit) {
            case RadiationEquivalentDoseUnits.Sieverts:
                return this.Sieverts + ` Sv`;
            case RadiationEquivalentDoseUnits.RoentgensEquivalentMan:
                return this.RoentgensEquivalentMan + ` rem`;
            case RadiationEquivalentDoseUnits.Nanosieverts:
                return this.Nanosieverts + ` nSv`;
            case RadiationEquivalentDoseUnits.Microsieverts:
                return this.Microsieverts + ` μSv`;
            case RadiationEquivalentDoseUnits.Millisieverts:
                return this.Millisieverts + ` mSv`;
            case RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan:
                return this.MilliroentgensEquivalentMan + ` mrem`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get RadiationEquivalentDose unit abbreviation.
     * Note! the default abbreviation for RadiationEquivalentDose is Sieverts.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the RadiationEquivalentDose.
     * @returns The abbreviation string of RadiationEquivalentDose.
     */
    getUnitAbbreviation(unitAbbreviation = RadiationEquivalentDoseUnits.Sieverts) {
        switch (unitAbbreviation) {
            case RadiationEquivalentDoseUnits.Sieverts:
                return `Sv`;
            case RadiationEquivalentDoseUnits.RoentgensEquivalentMan:
                return `rem`;
            case RadiationEquivalentDoseUnits.Nanosieverts:
                return `nSv`;
            case RadiationEquivalentDoseUnits.Microsieverts:
                return `μSv`;
            case RadiationEquivalentDoseUnits.Millisieverts:
                return `mSv`;
            case RadiationEquivalentDoseUnits.MilliroentgensEquivalentMan:
                return `mrem`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given RadiationEquivalentDose are equals to the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns True if the given RadiationEquivalentDose are equal to the current RadiationEquivalentDose.
     */
    equals(radiationEquivalentDose) {
        return this.value === radiationEquivalentDose.BaseValue;
    }
    /**
     * Compare the given RadiationEquivalentDose against the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns 0 if they are equal, -1 if the current RadiationEquivalentDose is less then other, 1 if the current RadiationEquivalentDose is greater then other.
     */
    compareTo(radiationEquivalentDose) {
        if (this.value > radiationEquivalentDose.BaseValue)
            return 1;
        if (this.value < radiationEquivalentDose.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given RadiationEquivalentDose with the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns A new RadiationEquivalentDose instance with the results.
     */
    add(radiationEquivalentDose) {
        return new RadiationEquivalentDose(this.value + radiationEquivalentDose.BaseValue);
    }
    /**
     * Subtract the given RadiationEquivalentDose with the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns A new RadiationEquivalentDose instance with the results.
     */
    subtract(radiationEquivalentDose) {
        return new RadiationEquivalentDose(this.value - radiationEquivalentDose.BaseValue);
    }
    /**
     * Multiply the given RadiationEquivalentDose with the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns A new RadiationEquivalentDose instance with the results.
     */
    multiply(radiationEquivalentDose) {
        return new RadiationEquivalentDose(this.value * radiationEquivalentDose.BaseValue);
    }
    /**
     * Divide the given RadiationEquivalentDose with the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns A new RadiationEquivalentDose instance with the results.
     */
    divide(radiationEquivalentDose) {
        return new RadiationEquivalentDose(this.value / radiationEquivalentDose.BaseValue);
    }
    /**
     * Modulo the given RadiationEquivalentDose with the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns A new RadiationEquivalentDose instance with the results.
     */
    modulo(radiationEquivalentDose) {
        return new RadiationEquivalentDose(this.value % radiationEquivalentDose.BaseValue);
    }
    /**
     * Pow the given RadiationEquivalentDose with the current RadiationEquivalentDose.
     * @param radiationEquivalentDose The other RadiationEquivalentDose.
     * @returns A new RadiationEquivalentDose instance with the results.
     */
    pow(radiationEquivalentDose) {
        return new RadiationEquivalentDose(this.value ** radiationEquivalentDose.BaseValue);
    }
}
exports.RadiationEquivalentDose = RadiationEquivalentDose;
