// unitsnet-js/dist/reciprocallength.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReciprocalLength = exports.ReciprocalLengthUnits = void 0;
/** ReciprocalLengthUnits enumeration */
var ReciprocalLengthUnits;
(function (ReciprocalLengthUnits) {
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseMeters"] = 0] = "InverseMeters";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseCentimeters"] = 1] = "InverseCentimeters";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseMillimeters"] = 2] = "InverseMillimeters";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseMiles"] = 3] = "InverseMiles";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseYards"] = 4] = "InverseYards";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseFeet"] = 5] = "InverseFeet";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseUsSurveyFeet"] = 6] = "InverseUsSurveyFeet";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseInches"] = 7] = "InverseInches";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseMils"] = 8] = "InverseMils";
    /** */
    ReciprocalLengthUnits[ReciprocalLengthUnits["InverseMicroinches"] = 9] = "InverseMicroinches";
})(ReciprocalLengthUnits = exports.ReciprocalLengthUnits || (exports.ReciprocalLengthUnits = {}));
/** Reciprocal (Inverse) Length is used in various fields of science and mathematics. It is defined as the inverse value of a length unit. */
class ReciprocalLength {
    /**
     * Create a new ReciprocalLength.
     * @param value The value.
     * @param fromUnit The ‘ReciprocalLength’ unit to create from.
     * The default unit is InverseMeters
     */
    constructor(value, fromUnit = ReciprocalLengthUnits.InverseMeters) {
        this.inversemetersLazy = null;
        this.inversecentimetersLazy = null;
        this.inversemillimetersLazy = null;
        this.inversemilesLazy = null;
        this.inverseyardsLazy = null;
        this.inversefeetLazy = null;
        this.inverseussurveyfeetLazy = null;
        this.inverseinchesLazy = null;
        this.inversemilsLazy = null;
        this.inversemicroinchesLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of ReciprocalLength is InverseMeters.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get InverseMeters() {
        if (this.inversemetersLazy !== null) {
            return this.inversemetersLazy;
        }
        return this.inversemetersLazy = this.convertFromBase(ReciprocalLengthUnits.InverseMeters);
    }
    /** */
    get InverseCentimeters() {
        if (this.inversecentimetersLazy !== null) {
            return this.inversecentimetersLazy;
        }
        return this.inversecentimetersLazy = this.convertFromBase(ReciprocalLengthUnits.InverseCentimeters);
    }
    /** */
    get InverseMillimeters() {
        if (this.inversemillimetersLazy !== null) {
            return this.inversemillimetersLazy;
        }
        return this.inversemillimetersLazy = this.convertFromBase(ReciprocalLengthUnits.InverseMillimeters);
    }
    /** */
    get InverseMiles() {
        if (this.inversemilesLazy !== null) {
            return this.inversemilesLazy;
        }
        return this.inversemilesLazy = this.convertFromBase(ReciprocalLengthUnits.InverseMiles);
    }
    /** */
    get InverseYards() {
        if (this.inverseyardsLazy !== null) {
            return this.inverseyardsLazy;
        }
        return this.inverseyardsLazy = this.convertFromBase(ReciprocalLengthUnits.InverseYards);
    }
    /** */
    get InverseFeet() {
        if (this.inversefeetLazy !== null) {
            return this.inversefeetLazy;
        }
        return this.inversefeetLazy = this.convertFromBase(ReciprocalLengthUnits.InverseFeet);
    }
    /** */
    get InverseUsSurveyFeet() {
        if (this.inverseussurveyfeetLazy !== null) {
            return this.inverseussurveyfeetLazy;
        }
        return this.inverseussurveyfeetLazy = this.convertFromBase(ReciprocalLengthUnits.InverseUsSurveyFeet);
    }
    /** */
    get InverseInches() {
        if (this.inverseinchesLazy !== null) {
            return this.inverseinchesLazy;
        }
        return this.inverseinchesLazy = this.convertFromBase(ReciprocalLengthUnits.InverseInches);
    }
    /** */
    get InverseMils() {
        if (this.inversemilsLazy !== null) {
            return this.inversemilsLazy;
        }
        return this.inversemilsLazy = this.convertFromBase(ReciprocalLengthUnits.InverseMils);
    }
    /** */
    get InverseMicroinches() {
        if (this.inversemicroinchesLazy !== null) {
            return this.inversemicroinchesLazy;
        }
        return this.inversemicroinchesLazy = this.convertFromBase(ReciprocalLengthUnits.InverseMicroinches);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseMeters
     *
     * @param value The unit as InverseMeters to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseMeters(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseMeters);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseCentimeters
     *
     * @param value The unit as InverseCentimeters to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseCentimeters(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseCentimeters);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseMillimeters
     *
     * @param value The unit as InverseMillimeters to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseMillimeters(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseMillimeters);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseMiles
     *
     * @param value The unit as InverseMiles to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseMiles(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseMiles);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseYards
     *
     * @param value The unit as InverseYards to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseYards(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseYards);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseFeet
     *
     * @param value The unit as InverseFeet to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseFeet(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseFeet);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseUsSurveyFeet
     *
     * @param value The unit as InverseUsSurveyFeet to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseUsSurveyFeet(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseUsSurveyFeet);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseInches
     *
     * @param value The unit as InverseInches to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseInches(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseInches);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseMils
     *
     * @param value The unit as InverseMils to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseMils(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseMils);
    }
    /**
     * Create a new ReciprocalLength instance from a InverseMicroinches
     *
     * @param value The unit as InverseMicroinches to create a new ReciprocalLength from.
     * @returns The new ReciprocalLength instance.
     */
    static FromInverseMicroinches(value) {
        return new ReciprocalLength(value, ReciprocalLengthUnits.InverseMicroinches);
    }
    /**
     * Convert ReciprocalLength to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case ReciprocalLengthUnits.InverseMeters: return this.InverseMeters;
            case ReciprocalLengthUnits.InverseCentimeters: return this.InverseCentimeters;
            case ReciprocalLengthUnits.InverseMillimeters: return this.InverseMillimeters;
            case ReciprocalLengthUnits.InverseMiles: return this.InverseMiles;
            case ReciprocalLengthUnits.InverseYards: return this.InverseYards;
            case ReciprocalLengthUnits.InverseFeet: return this.InverseFeet;
            case ReciprocalLengthUnits.InverseUsSurveyFeet: return this.InverseUsSurveyFeet;
            case ReciprocalLengthUnits.InverseInches: return this.InverseInches;
            case ReciprocalLengthUnits.InverseMils: return this.InverseMils;
            case ReciprocalLengthUnits.InverseMicroinches: return this.InverseMicroinches;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case ReciprocalLengthUnits.InverseMeters:
                return this.value;
            case ReciprocalLengthUnits.InverseCentimeters:
                return this.value / 1e2;
            case ReciprocalLengthUnits.InverseMillimeters:
                return this.value / 1e3;
            case ReciprocalLengthUnits.InverseMiles:
                return this.value * 1609.344;
            case ReciprocalLengthUnits.InverseYards:
                return this.value * 0.9144;
            case ReciprocalLengthUnits.InverseFeet:
                return this.value * 0.3048;
            case ReciprocalLengthUnits.InverseUsSurveyFeet:
                return this.value * 1200 / 3937;
            case ReciprocalLengthUnits.InverseInches:
                return this.value * 2.54e-2;
            case ReciprocalLengthUnits.InverseMils:
                return this.value * 2.54e-5;
            case ReciprocalLengthUnits.InverseMicroinches:
                return this.value * 2.54e-8;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case ReciprocalLengthUnits.InverseMeters:
                return value;
            case ReciprocalLengthUnits.InverseCentimeters:
                return value * 1e2;
            case ReciprocalLengthUnits.InverseMillimeters:
                return value * 1e3;
            case ReciprocalLengthUnits.InverseMiles:
                return value / 1609.344;
            case ReciprocalLengthUnits.InverseYards:
                return value / 0.9144;
            case ReciprocalLengthUnits.InverseFeet:
                return value / 0.3048;
            case ReciprocalLengthUnits.InverseUsSurveyFeet:
                return value * 3937 / 1200;
            case ReciprocalLengthUnits.InverseInches:
                return value / 2.54e-2;
            case ReciprocalLengthUnits.InverseMils:
                return value / 2.54e-5;
            case ReciprocalLengthUnits.InverseMicroinches:
                return value / 2.54e-8;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the ReciprocalLength to string.
     * Note! the default format for ReciprocalLength is InverseMeters.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the ReciprocalLength.
     * @returns The string format of the ReciprocalLength.
     */
    toString(unit = ReciprocalLengthUnits.InverseMeters) {
        switch (unit) {
            case ReciprocalLengthUnits.InverseMeters:
                return this.InverseMeters + ` m⁻¹`;
            case ReciprocalLengthUnits.InverseCentimeters:
                return this.InverseCentimeters + ` cm⁻¹`;
            case ReciprocalLengthUnits.InverseMillimeters:
                return this.InverseMillimeters + ` mm⁻¹`;
            case ReciprocalLengthUnits.InverseMiles:
                return this.InverseMiles + ` mi⁻¹`;
            case ReciprocalLengthUnits.InverseYards:
                return this.InverseYards + ` yd⁻¹`;
            case ReciprocalLengthUnits.InverseFeet:
                return this.InverseFeet + ` ft⁻¹`;
            case ReciprocalLengthUnits.InverseUsSurveyFeet:
                return this.InverseUsSurveyFeet + ` ftUS⁻¹`;
            case ReciprocalLengthUnits.InverseInches:
                return this.InverseInches + ` in⁻¹`;
            case ReciprocalLengthUnits.InverseMils:
                return this.InverseMils + ` mil⁻¹`;
            case ReciprocalLengthUnits.InverseMicroinches:
                return this.InverseMicroinches + ` µin⁻¹`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get ReciprocalLength unit abbreviation.
     * Note! the default abbreviation for ReciprocalLength is InverseMeters.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the ReciprocalLength.
     * @returns The abbreviation string of ReciprocalLength.
     */
    getUnitAbbreviation(unitAbbreviation = ReciprocalLengthUnits.InverseMeters) {
        switch (unitAbbreviation) {
            case ReciprocalLengthUnits.InverseMeters:
                return `m⁻¹`;
            case ReciprocalLengthUnits.InverseCentimeters:
                return `cm⁻¹`;
            case ReciprocalLengthUnits.InverseMillimeters:
                return `mm⁻¹`;
            case ReciprocalLengthUnits.InverseMiles:
                return `mi⁻¹`;
            case ReciprocalLengthUnits.InverseYards:
                return `yd⁻¹`;
            case ReciprocalLengthUnits.InverseFeet:
                return `ft⁻¹`;
            case ReciprocalLengthUnits.InverseUsSurveyFeet:
                return `ftUS⁻¹`;
            case ReciprocalLengthUnits.InverseInches:
                return `in⁻¹`;
            case ReciprocalLengthUnits.InverseMils:
                return `mil⁻¹`;
            case ReciprocalLengthUnits.InverseMicroinches:
                return `µin⁻¹`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given ReciprocalLength are equals to the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns True if the given ReciprocalLength are equal to the current ReciprocalLength.
     */
    equals(reciprocalLength) {
        return this.value === reciprocalLength.BaseValue;
    }
    /**
     * Compare the given ReciprocalLength against the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns 0 if they are equal, -1 if the current ReciprocalLength is less then other, 1 if the current ReciprocalLength is greater then other.
     */
    compareTo(reciprocalLength) {
        if (this.value > reciprocalLength.BaseValue)
            return 1;
        if (this.value < reciprocalLength.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given ReciprocalLength with the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns A new ReciprocalLength instance with the results.
     */
    add(reciprocalLength) {
        return new ReciprocalLength(this.value + reciprocalLength.BaseValue);
    }
    /**
     * Subtract the given ReciprocalLength with the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns A new ReciprocalLength instance with the results.
     */
    subtract(reciprocalLength) {
        return new ReciprocalLength(this.value - reciprocalLength.BaseValue);
    }
    /**
     * Multiply the given ReciprocalLength with the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns A new ReciprocalLength instance with the results.
     */
    multiply(reciprocalLength) {
        return new ReciprocalLength(this.value * reciprocalLength.BaseValue);
    }
    /**
     * Divide the given ReciprocalLength with the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns A new ReciprocalLength instance with the results.
     */
    divide(reciprocalLength) {
        return new ReciprocalLength(this.value / reciprocalLength.BaseValue);
    }
    /**
     * Modulo the given ReciprocalLength with the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns A new ReciprocalLength instance with the results.
     */
    modulo(reciprocalLength) {
        return new ReciprocalLength(this.value % reciprocalLength.BaseValue);
    }
    /**
     * Pow the given ReciprocalLength with the current ReciprocalLength.
     * @param reciprocalLength The other ReciprocalLength.
     * @returns A new ReciprocalLength instance with the results.
     */
    pow(reciprocalLength) {
        return new ReciprocalLength(this.value ** reciprocalLength.BaseValue);
    }
}
exports.ReciprocalLength = ReciprocalLength;
