// unitsnet-js/dist/relativehumidity.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelativeHumidity = exports.RelativeHumidityUnits = void 0;
/** RelativeHumidityUnits enumeration */
var RelativeHumidityUnits;
(function (RelativeHumidityUnits) {
    /** */
    RelativeHumidityUnits[RelativeHumidityUnits["Percent"] = 0] = "Percent";
})(RelativeHumidityUnits = exports.RelativeHumidityUnits || (exports.RelativeHumidityUnits = {}));
/** Relative humidity is a ratio of the actual water vapor present in the air to the maximum water vapor in the air at the given temperature. */
class RelativeHumidity {
    /**
     * Create a new RelativeHumidity.
     * @param value The value.
     * @param fromUnit The ‘RelativeHumidity’ unit to create from.
     * The default unit is Percent
     */
    constructor(value, fromUnit = RelativeHumidityUnits.Percent) {
        this.percentLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of RelativeHumidity is Percent.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Percent() {
        if (this.percentLazy !== null) {
            return this.percentLazy;
        }
        return this.percentLazy = this.convertFromBase(RelativeHumidityUnits.Percent);
    }
    /**
     * Create a new RelativeHumidity instance from a Percent
     *
     * @param value The unit as Percent to create a new RelativeHumidity from.
     * @returns The new RelativeHumidity instance.
     */
    static FromPercent(value) {
        return new RelativeHumidity(value, RelativeHumidityUnits.Percent);
    }
    /**
     * Convert RelativeHumidity to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case RelativeHumidityUnits.Percent: return this.Percent;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case RelativeHumidityUnits.Percent:
                return this.value;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case RelativeHumidityUnits.Percent:
                return value;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the RelativeHumidity to string.
     * Note! the default format for RelativeHumidity is Percent.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the RelativeHumidity.
     * @returns The string format of the RelativeHumidity.
     */
    toString(unit = RelativeHumidityUnits.Percent) {
        switch (unit) {
            case RelativeHumidityUnits.Percent:
                return this.Percent + ` %RH`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get RelativeHumidity unit abbreviation.
     * Note! the default abbreviation for RelativeHumidity is Percent.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the RelativeHumidity.
     * @returns The abbreviation string of RelativeHumidity.
     */
    getUnitAbbreviation(unitAbbreviation = RelativeHumidityUnits.Percent) {
        switch (unitAbbreviation) {
            case RelativeHumidityUnits.Percent:
                return `%RH`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given RelativeHumidity are equals to the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns True if the given RelativeHumidity are equal to the current RelativeHumidity.
     */
    equals(relativeHumidity) {
        return this.value === relativeHumidity.BaseValue;
    }
    /**
     * Compare the given RelativeHumidity against the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns 0 if they are equal, -1 if the current RelativeHumidity is less then other, 1 if the current RelativeHumidity is greater then other.
     */
    compareTo(relativeHumidity) {
        if (this.value > relativeHumidity.BaseValue)
            return 1;
        if (this.value < relativeHumidity.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given RelativeHumidity with the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns A new RelativeHumidity instance with the results.
     */
    add(relativeHumidity) {
        return new RelativeHumidity(this.value + relativeHumidity.BaseValue);
    }
    /**
     * Subtract the given RelativeHumidity with the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns A new RelativeHumidity instance with the results.
     */
    subtract(relativeHumidity) {
        return new RelativeHumidity(this.value - relativeHumidity.BaseValue);
    }
    /**
     * Multiply the given RelativeHumidity with the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns A new RelativeHumidity instance with the results.
     */
    multiply(relativeHumidity) {
        return new RelativeHumidity(this.value * relativeHumidity.BaseValue);
    }
    /**
     * Divide the given RelativeHumidity with the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns A new RelativeHumidity instance with the results.
     */
    divide(relativeHumidity) {
        return new RelativeHumidity(this.value / relativeHumidity.BaseValue);
    }
    /**
     * Modulo the given RelativeHumidity with the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns A new RelativeHumidity instance with the results.
     */
    modulo(relativeHumidity) {
        return new RelativeHumidity(this.value % relativeHumidity.BaseValue);
    }
    /**
     * Pow the given RelativeHumidity with the current RelativeHumidity.
     * @param relativeHumidity The other RelativeHumidity.
     * @returns A new RelativeHumidity instance with the results.
     */
    pow(relativeHumidity) {
        return new RelativeHumidity(this.value ** relativeHumidity.BaseValue);
    }
}
exports.RelativeHumidity = RelativeHumidity;
