// unitsnet-js/dist/rotationalstiffness.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RotationalStiffness = exports.RotationalStiffnessUnits = void 0;
/** RotationalStiffnessUnits enumeration */
var RotationalStiffnessUnits;
(function (RotationalStiffnessUnits) {
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NewtonMetersPerRadian"] = 0] = "NewtonMetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["PoundForceFeetPerDegrees"] = 1] = "PoundForceFeetPerDegrees";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["KilopoundForceFeetPerDegrees"] = 2] = "KilopoundForceFeetPerDegrees";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NewtonMillimetersPerDegree"] = 3] = "NewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NewtonMetersPerDegree"] = 4] = "NewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NewtonMillimetersPerRadian"] = 5] = "NewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["PoundForceFeetPerRadian"] = 6] = "PoundForceFeetPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["KilonewtonMetersPerRadian"] = 7] = "KilonewtonMetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MeganewtonMetersPerRadian"] = 8] = "MeganewtonMetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NanonewtonMillimetersPerDegree"] = 9] = "NanonewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MicronewtonMillimetersPerDegree"] = 10] = "MicronewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MillinewtonMillimetersPerDegree"] = 11] = "MillinewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["CentinewtonMillimetersPerDegree"] = 12] = "CentinewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["DecinewtonMillimetersPerDegree"] = 13] = "DecinewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["DecanewtonMillimetersPerDegree"] = 14] = "DecanewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["KilonewtonMillimetersPerDegree"] = 15] = "KilonewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MeganewtonMillimetersPerDegree"] = 16] = "MeganewtonMillimetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NanonewtonMetersPerDegree"] = 17] = "NanonewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MicronewtonMetersPerDegree"] = 18] = "MicronewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MillinewtonMetersPerDegree"] = 19] = "MillinewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["CentinewtonMetersPerDegree"] = 20] = "CentinewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["DecinewtonMetersPerDegree"] = 21] = "DecinewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["DecanewtonMetersPerDegree"] = 22] = "DecanewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["KilonewtonMetersPerDegree"] = 23] = "KilonewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MeganewtonMetersPerDegree"] = 24] = "MeganewtonMetersPerDegree";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["NanonewtonMillimetersPerRadian"] = 25] = "NanonewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MicronewtonMillimetersPerRadian"] = 26] = "MicronewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MillinewtonMillimetersPerRadian"] = 27] = "MillinewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["CentinewtonMillimetersPerRadian"] = 28] = "CentinewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["DecinewtonMillimetersPerRadian"] = 29] = "DecinewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["DecanewtonMillimetersPerRadian"] = 30] = "DecanewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["KilonewtonMillimetersPerRadian"] = 31] = "KilonewtonMillimetersPerRadian";
    /** */
    RotationalStiffnessUnits[RotationalStiffnessUnits["MeganewtonMillimetersPerRadian"] = 32] = "MeganewtonMillimetersPerRadian";
})(RotationalStiffnessUnits = exports.RotationalStiffnessUnits || (exports.RotationalStiffnessUnits = {}));
/** https://en.wikipedia.org/wiki/Stiffness#Rotational_stiffness */
class RotationalStiffness {
    /**
     * Create a new RotationalStiffness.
     * @param value The value.
     * @param fromUnit The ‘RotationalStiffness’ unit to create from.
     * The default unit is NewtonMetersPerRadian
     */
    constructor(value, fromUnit = RotationalStiffnessUnits.NewtonMetersPerRadian) {
        this.newtonmetersperradianLazy = null;
        this.poundforcefeetperdegreesLazy = null;
        this.kilopoundforcefeetperdegreesLazy = null;
        this.newtonmillimetersperdegreeLazy = null;
        this.newtonmetersperdegreeLazy = null;
        this.newtonmillimetersperradianLazy = null;
        this.poundforcefeetperradianLazy = null;
        this.kilonewtonmetersperradianLazy = null;
        this.meganewtonmetersperradianLazy = null;
        this.nanonewtonmillimetersperdegreeLazy = null;
        this.micronewtonmillimetersperdegreeLazy = null;
        this.millinewtonmillimetersperdegreeLazy = null;
        this.centinewtonmillimetersperdegreeLazy = null;
        this.decinewtonmillimetersperdegreeLazy = null;
        this.decanewtonmillimetersperdegreeLazy = null;
        this.kilonewtonmillimetersperdegreeLazy = null;
        this.meganewtonmillimetersperdegreeLazy = null;
        this.nanonewtonmetersperdegreeLazy = null;
        this.micronewtonmetersperdegreeLazy = null;
        this.millinewtonmetersperdegreeLazy = null;
        this.centinewtonmetersperdegreeLazy = null;
        this.decinewtonmetersperdegreeLazy = null;
        this.decanewtonmetersperdegreeLazy = null;
        this.kilonewtonmetersperdegreeLazy = null;
        this.meganewtonmetersperdegreeLazy = null;
        this.nanonewtonmillimetersperradianLazy = null;
        this.micronewtonmillimetersperradianLazy = null;
        this.millinewtonmillimetersperradianLazy = null;
        this.centinewtonmillimetersperradianLazy = null;
        this.decinewtonmillimetersperradianLazy = null;
        this.decanewtonmillimetersperradianLazy = null;
        this.kilonewtonmillimetersperradianLazy = null;
        this.meganewtonmillimetersperradianLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of RotationalStiffness is NewtonMetersPerRadian.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get NewtonMetersPerRadian() {
        if (this.newtonmetersperradianLazy !== null) {
            return this.newtonmetersperradianLazy;
        }
        return this.newtonmetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.NewtonMetersPerRadian);
    }
    /** */
    get PoundForceFeetPerDegrees() {
        if (this.poundforcefeetperdegreesLazy !== null) {
            return this.poundforcefeetperdegreesLazy;
        }
        return this.poundforcefeetperdegreesLazy = this.convertFromBase(RotationalStiffnessUnits.PoundForceFeetPerDegrees);
    }
    /** */
    get KilopoundForceFeetPerDegrees() {
        if (this.kilopoundforcefeetperdegreesLazy !== null) {
            return this.kilopoundforcefeetperdegreesLazy;
        }
        return this.kilopoundforcefeetperdegreesLazy = this.convertFromBase(RotationalStiffnessUnits.KilopoundForceFeetPerDegrees);
    }
    /** */
    get NewtonMillimetersPerDegree() {
        if (this.newtonmillimetersperdegreeLazy !== null) {
            return this.newtonmillimetersperdegreeLazy;
        }
        return this.newtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.NewtonMillimetersPerDegree);
    }
    /** */
    get NewtonMetersPerDegree() {
        if (this.newtonmetersperdegreeLazy !== null) {
            return this.newtonmetersperdegreeLazy;
        }
        return this.newtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.NewtonMetersPerDegree);
    }
    /** */
    get NewtonMillimetersPerRadian() {
        if (this.newtonmillimetersperradianLazy !== null) {
            return this.newtonmillimetersperradianLazy;
        }
        return this.newtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.NewtonMillimetersPerRadian);
    }
    /** */
    get PoundForceFeetPerRadian() {
        if (this.poundforcefeetperradianLazy !== null) {
            return this.poundforcefeetperradianLazy;
        }
        return this.poundforcefeetperradianLazy = this.convertFromBase(RotationalStiffnessUnits.PoundForceFeetPerRadian);
    }
    /** */
    get KilonewtonMetersPerRadian() {
        if (this.kilonewtonmetersperradianLazy !== null) {
            return this.kilonewtonmetersperradianLazy;
        }
        return this.kilonewtonmetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.KilonewtonMetersPerRadian);
    }
    /** */
    get MeganewtonMetersPerRadian() {
        if (this.meganewtonmetersperradianLazy !== null) {
            return this.meganewtonmetersperradianLazy;
        }
        return this.meganewtonmetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.MeganewtonMetersPerRadian);
    }
    /** */
    get NanonewtonMillimetersPerDegree() {
        if (this.nanonewtonmillimetersperdegreeLazy !== null) {
            return this.nanonewtonmillimetersperdegreeLazy;
        }
        return this.nanonewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.NanonewtonMillimetersPerDegree);
    }
    /** */
    get MicronewtonMillimetersPerDegree() {
        if (this.micronewtonmillimetersperdegreeLazy !== null) {
            return this.micronewtonmillimetersperdegreeLazy;
        }
        return this.micronewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.MicronewtonMillimetersPerDegree);
    }
    /** */
    get MillinewtonMillimetersPerDegree() {
        if (this.millinewtonmillimetersperdegreeLazy !== null) {
            return this.millinewtonmillimetersperdegreeLazy;
        }
        return this.millinewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.MillinewtonMillimetersPerDegree);
    }
    /** */
    get CentinewtonMillimetersPerDegree() {
        if (this.centinewtonmillimetersperdegreeLazy !== null) {
            return this.centinewtonmillimetersperdegreeLazy;
        }
        return this.centinewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.CentinewtonMillimetersPerDegree);
    }
    /** */
    get DecinewtonMillimetersPerDegree() {
        if (this.decinewtonmillimetersperdegreeLazy !== null) {
            return this.decinewtonmillimetersperdegreeLazy;
        }
        return this.decinewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.DecinewtonMillimetersPerDegree);
    }
    /** */
    get DecanewtonMillimetersPerDegree() {
        if (this.decanewtonmillimetersperdegreeLazy !== null) {
            return this.decanewtonmillimetersperdegreeLazy;
        }
        return this.decanewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.DecanewtonMillimetersPerDegree);
    }
    /** */
    get KilonewtonMillimetersPerDegree() {
        if (this.kilonewtonmillimetersperdegreeLazy !== null) {
            return this.kilonewtonmillimetersperdegreeLazy;
        }
        return this.kilonewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.KilonewtonMillimetersPerDegree);
    }
    /** */
    get MeganewtonMillimetersPerDegree() {
        if (this.meganewtonmillimetersperdegreeLazy !== null) {
            return this.meganewtonmillimetersperdegreeLazy;
        }
        return this.meganewtonmillimetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.MeganewtonMillimetersPerDegree);
    }
    /** */
    get NanonewtonMetersPerDegree() {
        if (this.nanonewtonmetersperdegreeLazy !== null) {
            return this.nanonewtonmetersperdegreeLazy;
        }
        return this.nanonewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.NanonewtonMetersPerDegree);
    }
    /** */
    get MicronewtonMetersPerDegree() {
        if (this.micronewtonmetersperdegreeLazy !== null) {
            return this.micronewtonmetersperdegreeLazy;
        }
        return this.micronewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.MicronewtonMetersPerDegree);
    }
    /** */
    get MillinewtonMetersPerDegree() {
        if (this.millinewtonmetersperdegreeLazy !== null) {
            return this.millinewtonmetersperdegreeLazy;
        }
        return this.millinewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.MillinewtonMetersPerDegree);
    }
    /** */
    get CentinewtonMetersPerDegree() {
        if (this.centinewtonmetersperdegreeLazy !== null) {
            return this.centinewtonmetersperdegreeLazy;
        }
        return this.centinewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.CentinewtonMetersPerDegree);
    }
    /** */
    get DecinewtonMetersPerDegree() {
        if (this.decinewtonmetersperdegreeLazy !== null) {
            return this.decinewtonmetersperdegreeLazy;
        }
        return this.decinewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.DecinewtonMetersPerDegree);
    }
    /** */
    get DecanewtonMetersPerDegree() {
        if (this.decanewtonmetersperdegreeLazy !== null) {
            return this.decanewtonmetersperdegreeLazy;
        }
        return this.decanewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.DecanewtonMetersPerDegree);
    }
    /** */
    get KilonewtonMetersPerDegree() {
        if (this.kilonewtonmetersperdegreeLazy !== null) {
            return this.kilonewtonmetersperdegreeLazy;
        }
        return this.kilonewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.KilonewtonMetersPerDegree);
    }
    /** */
    get MeganewtonMetersPerDegree() {
        if (this.meganewtonmetersperdegreeLazy !== null) {
            return this.meganewtonmetersperdegreeLazy;
        }
        return this.meganewtonmetersperdegreeLazy = this.convertFromBase(RotationalStiffnessUnits.MeganewtonMetersPerDegree);
    }
    /** */
    get NanonewtonMillimetersPerRadian() {
        if (this.nanonewtonmillimetersperradianLazy !== null) {
            return this.nanonewtonmillimetersperradianLazy;
        }
        return this.nanonewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.NanonewtonMillimetersPerRadian);
    }
    /** */
    get MicronewtonMillimetersPerRadian() {
        if (this.micronewtonmillimetersperradianLazy !== null) {
            return this.micronewtonmillimetersperradianLazy;
        }
        return this.micronewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.MicronewtonMillimetersPerRadian);
    }
    /** */
    get MillinewtonMillimetersPerRadian() {
        if (this.millinewtonmillimetersperradianLazy !== null) {
            return this.millinewtonmillimetersperradianLazy;
        }
        return this.millinewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.MillinewtonMillimetersPerRadian);
    }
    /** */
    get CentinewtonMillimetersPerRadian() {
        if (this.centinewtonmillimetersperradianLazy !== null) {
            return this.centinewtonmillimetersperradianLazy;
        }
        return this.centinewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.CentinewtonMillimetersPerRadian);
    }
    /** */
    get DecinewtonMillimetersPerRadian() {
        if (this.decinewtonmillimetersperradianLazy !== null) {
            return this.decinewtonmillimetersperradianLazy;
        }
        return this.decinewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.DecinewtonMillimetersPerRadian);
    }
    /** */
    get DecanewtonMillimetersPerRadian() {
        if (this.decanewtonmillimetersperradianLazy !== null) {
            return this.decanewtonmillimetersperradianLazy;
        }
        return this.decanewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.DecanewtonMillimetersPerRadian);
    }
    /** */
    get KilonewtonMillimetersPerRadian() {
        if (this.kilonewtonmillimetersperradianLazy !== null) {
            return this.kilonewtonmillimetersperradianLazy;
        }
        return this.kilonewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.KilonewtonMillimetersPerRadian);
    }
    /** */
    get MeganewtonMillimetersPerRadian() {
        if (this.meganewtonmillimetersperradianLazy !== null) {
            return this.meganewtonmillimetersperradianLazy;
        }
        return this.meganewtonmillimetersperradianLazy = this.convertFromBase(RotationalStiffnessUnits.MeganewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a NewtonMetersPerRadian
     *
     * @param value The unit as NewtonMetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNewtonMetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NewtonMetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a PoundForceFeetPerDegrees
     *
     * @param value The unit as PoundForceFeetPerDegrees to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromPoundForceFeetPerDegrees(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.PoundForceFeetPerDegrees);
    }
    /**
     * Create a new RotationalStiffness instance from a KilopoundForceFeetPerDegrees
     *
     * @param value The unit as KilopoundForceFeetPerDegrees to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromKilopoundForceFeetPerDegrees(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.KilopoundForceFeetPerDegrees);
    }
    /**
     * Create a new RotationalStiffness instance from a NewtonMillimetersPerDegree
     *
     * @param value The unit as NewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a NewtonMetersPerDegree
     *
     * @param value The unit as NewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a NewtonMillimetersPerRadian
     *
     * @param value The unit as NewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a PoundForceFeetPerRadian
     *
     * @param value The unit as PoundForceFeetPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromPoundForceFeetPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.PoundForceFeetPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a KilonewtonMetersPerRadian
     *
     * @param value The unit as KilonewtonMetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromKilonewtonMetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.KilonewtonMetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a MeganewtonMetersPerRadian
     *
     * @param value The unit as MeganewtonMetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMeganewtonMetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MeganewtonMetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a NanonewtonMillimetersPerDegree
     *
     * @param value The unit as NanonewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNanonewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NanonewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a MicronewtonMillimetersPerDegree
     *
     * @param value The unit as MicronewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMicronewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MicronewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a MillinewtonMillimetersPerDegree
     *
     * @param value The unit as MillinewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMillinewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MillinewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a CentinewtonMillimetersPerDegree
     *
     * @param value The unit as CentinewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromCentinewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.CentinewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a DecinewtonMillimetersPerDegree
     *
     * @param value The unit as DecinewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromDecinewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.DecinewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a DecanewtonMillimetersPerDegree
     *
     * @param value The unit as DecanewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromDecanewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.DecanewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a KilonewtonMillimetersPerDegree
     *
     * @param value The unit as KilonewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromKilonewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.KilonewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a MeganewtonMillimetersPerDegree
     *
     * @param value The unit as MeganewtonMillimetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMeganewtonMillimetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MeganewtonMillimetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a NanonewtonMetersPerDegree
     *
     * @param value The unit as NanonewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNanonewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NanonewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a MicronewtonMetersPerDegree
     *
     * @param value The unit as MicronewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMicronewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MicronewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a MillinewtonMetersPerDegree
     *
     * @param value The unit as MillinewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMillinewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MillinewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a CentinewtonMetersPerDegree
     *
     * @param value The unit as CentinewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromCentinewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.CentinewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a DecinewtonMetersPerDegree
     *
     * @param value The unit as DecinewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromDecinewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.DecinewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a DecanewtonMetersPerDegree
     *
     * @param value The unit as DecanewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromDecanewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.DecanewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a KilonewtonMetersPerDegree
     *
     * @param value The unit as KilonewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromKilonewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.KilonewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a MeganewtonMetersPerDegree
     *
     * @param value The unit as MeganewtonMetersPerDegree to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMeganewtonMetersPerDegree(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MeganewtonMetersPerDegree);
    }
    /**
     * Create a new RotationalStiffness instance from a NanonewtonMillimetersPerRadian
     *
     * @param value The unit as NanonewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromNanonewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.NanonewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a MicronewtonMillimetersPerRadian
     *
     * @param value The unit as MicronewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMicronewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MicronewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a MillinewtonMillimetersPerRadian
     *
     * @param value The unit as MillinewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMillinewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MillinewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a CentinewtonMillimetersPerRadian
     *
     * @param value The unit as CentinewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromCentinewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.CentinewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a DecinewtonMillimetersPerRadian
     *
     * @param value The unit as DecinewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromDecinewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.DecinewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a DecanewtonMillimetersPerRadian
     *
     * @param value The unit as DecanewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromDecanewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.DecanewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a KilonewtonMillimetersPerRadian
     *
     * @param value The unit as KilonewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromKilonewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.KilonewtonMillimetersPerRadian);
    }
    /**
     * Create a new RotationalStiffness instance from a MeganewtonMillimetersPerRadian
     *
     * @param value The unit as MeganewtonMillimetersPerRadian to create a new RotationalStiffness from.
     * @returns The new RotationalStiffness instance.
     */
    static FromMeganewtonMillimetersPerRadian(value) {
        return new RotationalStiffness(value, RotationalStiffnessUnits.MeganewtonMillimetersPerRadian);
    }
    /**
     * Convert RotationalStiffness to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case RotationalStiffnessUnits.NewtonMetersPerRadian: return this.NewtonMetersPerRadian;
            case RotationalStiffnessUnits.PoundForceFeetPerDegrees: return this.PoundForceFeetPerDegrees;
            case RotationalStiffnessUnits.KilopoundForceFeetPerDegrees: return this.KilopoundForceFeetPerDegrees;
            case RotationalStiffnessUnits.NewtonMillimetersPerDegree: return this.NewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.NewtonMetersPerDegree: return this.NewtonMetersPerDegree;
            case RotationalStiffnessUnits.NewtonMillimetersPerRadian: return this.NewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.PoundForceFeetPerRadian: return this.PoundForceFeetPerRadian;
            case RotationalStiffnessUnits.KilonewtonMetersPerRadian: return this.KilonewtonMetersPerRadian;
            case RotationalStiffnessUnits.MeganewtonMetersPerRadian: return this.MeganewtonMetersPerRadian;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerDegree: return this.NanonewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerDegree: return this.MicronewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerDegree: return this.MillinewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerDegree: return this.CentinewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerDegree: return this.DecinewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerDegree: return this.DecanewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerDegree: return this.KilonewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerDegree: return this.MeganewtonMillimetersPerDegree;
            case RotationalStiffnessUnits.NanonewtonMetersPerDegree: return this.NanonewtonMetersPerDegree;
            case RotationalStiffnessUnits.MicronewtonMetersPerDegree: return this.MicronewtonMetersPerDegree;
            case RotationalStiffnessUnits.MillinewtonMetersPerDegree: return this.MillinewtonMetersPerDegree;
            case RotationalStiffnessUnits.CentinewtonMetersPerDegree: return this.CentinewtonMetersPerDegree;
            case RotationalStiffnessUnits.DecinewtonMetersPerDegree: return this.DecinewtonMetersPerDegree;
            case RotationalStiffnessUnits.DecanewtonMetersPerDegree: return this.DecanewtonMetersPerDegree;
            case RotationalStiffnessUnits.KilonewtonMetersPerDegree: return this.KilonewtonMetersPerDegree;
            case RotationalStiffnessUnits.MeganewtonMetersPerDegree: return this.MeganewtonMetersPerDegree;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerRadian: return this.NanonewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerRadian: return this.MicronewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerRadian: return this.MillinewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerRadian: return this.CentinewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerRadian: return this.DecinewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerRadian: return this.DecanewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerRadian: return this.KilonewtonMillimetersPerRadian;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerRadian: return this.MeganewtonMillimetersPerRadian;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case RotationalStiffnessUnits.NewtonMetersPerRadian:
                return this.value;
            case RotationalStiffnessUnits.PoundForceFeetPerDegrees:
                return this.value / 77.6826;
            case RotationalStiffnessUnits.KilopoundForceFeetPerDegrees:
                return this.value / 77682.6;
            case RotationalStiffnessUnits.NewtonMillimetersPerDegree:
                return this.value / 180 * Math.PI * 1000;
            case RotationalStiffnessUnits.NewtonMetersPerDegree:
                return this.value / (180 / Math.PI);
            case RotationalStiffnessUnits.NewtonMillimetersPerRadian:
                return this.value * 1000;
            case RotationalStiffnessUnits.PoundForceFeetPerRadian:
                return this.value / 1.3558179483314;
            case RotationalStiffnessUnits.KilonewtonMetersPerRadian:
                return (this.value) / 1000;
            case RotationalStiffnessUnits.MeganewtonMetersPerRadian:
                return (this.value) / 1000000;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 1e-9;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 0.000001;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 0.001;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 0.01;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 0.1;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 10;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 1000;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerDegree:
                return (this.value / 180 * Math.PI * 1000) / 1000000;
            case RotationalStiffnessUnits.NanonewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 1e-9;
            case RotationalStiffnessUnits.MicronewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 0.000001;
            case RotationalStiffnessUnits.MillinewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 0.001;
            case RotationalStiffnessUnits.CentinewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 0.01;
            case RotationalStiffnessUnits.DecinewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 0.1;
            case RotationalStiffnessUnits.DecanewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 10;
            case RotationalStiffnessUnits.KilonewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 1000;
            case RotationalStiffnessUnits.MeganewtonMetersPerDegree:
                return (this.value / (180 / Math.PI)) / 1000000;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerRadian:
                return (this.value * 1000) / 1e-9;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerRadian:
                return (this.value * 1000) / 0.000001;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerRadian:
                return (this.value * 1000) / 0.001;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerRadian:
                return (this.value * 1000) / 0.01;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerRadian:
                return (this.value * 1000) / 0.1;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerRadian:
                return (this.value * 1000) / 10;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerRadian:
                return (this.value * 1000) / 1000;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerRadian:
                return (this.value * 1000) / 1000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case RotationalStiffnessUnits.NewtonMetersPerRadian:
                return value;
            case RotationalStiffnessUnits.PoundForceFeetPerDegrees:
                return value * 77.6826;
            case RotationalStiffnessUnits.KilopoundForceFeetPerDegrees:
                return value * 77682.6;
            case RotationalStiffnessUnits.NewtonMillimetersPerDegree:
                return value * 180 / Math.PI * 0.001;
            case RotationalStiffnessUnits.NewtonMetersPerDegree:
                return value * (180 / Math.PI);
            case RotationalStiffnessUnits.NewtonMillimetersPerRadian:
                return value * 0.001;
            case RotationalStiffnessUnits.PoundForceFeetPerRadian:
                return value * 1.3558179483314;
            case RotationalStiffnessUnits.KilonewtonMetersPerRadian:
                return (value) * 1000;
            case RotationalStiffnessUnits.MeganewtonMetersPerRadian:
                return (value) * 1000000;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 1e-9;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 0.000001;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 0.001;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 0.01;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 0.1;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 10;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 1000;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerDegree:
                return (value * 180 / Math.PI * 0.001) * 1000000;
            case RotationalStiffnessUnits.NanonewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 1e-9;
            case RotationalStiffnessUnits.MicronewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 0.000001;
            case RotationalStiffnessUnits.MillinewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 0.001;
            case RotationalStiffnessUnits.CentinewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 0.01;
            case RotationalStiffnessUnits.DecinewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 0.1;
            case RotationalStiffnessUnits.DecanewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 10;
            case RotationalStiffnessUnits.KilonewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 1000;
            case RotationalStiffnessUnits.MeganewtonMetersPerDegree:
                return (value * (180 / Math.PI)) * 1000000;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerRadian:
                return (value * 0.001) * 1e-9;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerRadian:
                return (value * 0.001) * 0.000001;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerRadian:
                return (value * 0.001) * 0.001;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerRadian:
                return (value * 0.001) * 0.01;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerRadian:
                return (value * 0.001) * 0.1;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerRadian:
                return (value * 0.001) * 10;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerRadian:
                return (value * 0.001) * 1000;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerRadian:
                return (value * 0.001) * 1000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the RotationalStiffness to string.
     * Note! the default format for RotationalStiffness is NewtonMetersPerRadian.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the RotationalStiffness.
     * @returns The string format of the RotationalStiffness.
     */
    toString(unit = RotationalStiffnessUnits.NewtonMetersPerRadian) {
        switch (unit) {
            case RotationalStiffnessUnits.NewtonMetersPerRadian:
                return this.NewtonMetersPerRadian + ` N·m/rad`;
            case RotationalStiffnessUnits.PoundForceFeetPerDegrees:
                return this.PoundForceFeetPerDegrees + ` lbf·ft/deg`;
            case RotationalStiffnessUnits.KilopoundForceFeetPerDegrees:
                return this.KilopoundForceFeetPerDegrees + ` kipf·ft/°`;
            case RotationalStiffnessUnits.NewtonMillimetersPerDegree:
                return this.NewtonMillimetersPerDegree + ` N·mm/deg`;
            case RotationalStiffnessUnits.NewtonMetersPerDegree:
                return this.NewtonMetersPerDegree + ` N·m/deg`;
            case RotationalStiffnessUnits.NewtonMillimetersPerRadian:
                return this.NewtonMillimetersPerRadian + ` N·mm/rad`;
            case RotationalStiffnessUnits.PoundForceFeetPerRadian:
                return this.PoundForceFeetPerRadian + ` lbf·ft/rad`;
            case RotationalStiffnessUnits.KilonewtonMetersPerRadian:
                return this.KilonewtonMetersPerRadian + ` kN·m/rad`;
            case RotationalStiffnessUnits.MeganewtonMetersPerRadian:
                return this.MeganewtonMetersPerRadian + ` MN·m/rad`;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerDegree:
                return this.NanonewtonMillimetersPerDegree + ` nN·mm/deg`;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerDegree:
                return this.MicronewtonMillimetersPerDegree + ` μN·mm/deg`;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerDegree:
                return this.MillinewtonMillimetersPerDegree + ` mN·mm/deg`;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerDegree:
                return this.CentinewtonMillimetersPerDegree + ` cN·mm/deg`;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerDegree:
                return this.DecinewtonMillimetersPerDegree + ` dN·mm/deg`;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerDegree:
                return this.DecanewtonMillimetersPerDegree + ` daN·mm/deg`;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerDegree:
                return this.KilonewtonMillimetersPerDegree + ` kN·mm/deg`;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerDegree:
                return this.MeganewtonMillimetersPerDegree + ` MN·mm/deg`;
            case RotationalStiffnessUnits.NanonewtonMetersPerDegree:
                return this.NanonewtonMetersPerDegree + ` nN·m/deg`;
            case RotationalStiffnessUnits.MicronewtonMetersPerDegree:
                return this.MicronewtonMetersPerDegree + ` μN·m/deg`;
            case RotationalStiffnessUnits.MillinewtonMetersPerDegree:
                return this.MillinewtonMetersPerDegree + ` mN·m/deg`;
            case RotationalStiffnessUnits.CentinewtonMetersPerDegree:
                return this.CentinewtonMetersPerDegree + ` cN·m/deg`;
            case RotationalStiffnessUnits.DecinewtonMetersPerDegree:
                return this.DecinewtonMetersPerDegree + ` dN·m/deg`;
            case RotationalStiffnessUnits.DecanewtonMetersPerDegree:
                return this.DecanewtonMetersPerDegree + ` daN·m/deg`;
            case RotationalStiffnessUnits.KilonewtonMetersPerDegree:
                return this.KilonewtonMetersPerDegree + ` kN·m/deg`;
            case RotationalStiffnessUnits.MeganewtonMetersPerDegree:
                return this.MeganewtonMetersPerDegree + ` MN·m/deg`;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerRadian:
                return this.NanonewtonMillimetersPerRadian + ` nN·mm/rad`;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerRadian:
                return this.MicronewtonMillimetersPerRadian + ` μN·mm/rad`;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerRadian:
                return this.MillinewtonMillimetersPerRadian + ` mN·mm/rad`;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerRadian:
                return this.CentinewtonMillimetersPerRadian + ` cN·mm/rad`;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerRadian:
                return this.DecinewtonMillimetersPerRadian + ` dN·mm/rad`;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerRadian:
                return this.DecanewtonMillimetersPerRadian + ` daN·mm/rad`;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerRadian:
                return this.KilonewtonMillimetersPerRadian + ` kN·mm/rad`;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerRadian:
                return this.MeganewtonMillimetersPerRadian + ` MN·mm/rad`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get RotationalStiffness unit abbreviation.
     * Note! the default abbreviation for RotationalStiffness is NewtonMetersPerRadian.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the RotationalStiffness.
     * @returns The abbreviation string of RotationalStiffness.
     */
    getUnitAbbreviation(unitAbbreviation = RotationalStiffnessUnits.NewtonMetersPerRadian) {
        switch (unitAbbreviation) {
            case RotationalStiffnessUnits.NewtonMetersPerRadian:
                return `N·m/rad`;
            case RotationalStiffnessUnits.PoundForceFeetPerDegrees:
                return `lbf·ft/deg`;
            case RotationalStiffnessUnits.KilopoundForceFeetPerDegrees:
                return `kipf·ft/°`;
            case RotationalStiffnessUnits.NewtonMillimetersPerDegree:
                return `N·mm/deg`;
            case RotationalStiffnessUnits.NewtonMetersPerDegree:
                return `N·m/deg`;
            case RotationalStiffnessUnits.NewtonMillimetersPerRadian:
                return `N·mm/rad`;
            case RotationalStiffnessUnits.PoundForceFeetPerRadian:
                return `lbf·ft/rad`;
            case RotationalStiffnessUnits.KilonewtonMetersPerRadian:
                return `kN·m/rad`;
            case RotationalStiffnessUnits.MeganewtonMetersPerRadian:
                return `MN·m/rad`;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerDegree:
                return `nN·mm/deg`;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerDegree:
                return `μN·mm/deg`;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerDegree:
                return `mN·mm/deg`;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerDegree:
                return `cN·mm/deg`;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerDegree:
                return `dN·mm/deg`;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerDegree:
                return `daN·mm/deg`;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerDegree:
                return `kN·mm/deg`;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerDegree:
                return `MN·mm/deg`;
            case RotationalStiffnessUnits.NanonewtonMetersPerDegree:
                return `nN·m/deg`;
            case RotationalStiffnessUnits.MicronewtonMetersPerDegree:
                return `μN·m/deg`;
            case RotationalStiffnessUnits.MillinewtonMetersPerDegree:
                return `mN·m/deg`;
            case RotationalStiffnessUnits.CentinewtonMetersPerDegree:
                return `cN·m/deg`;
            case RotationalStiffnessUnits.DecinewtonMetersPerDegree:
                return `dN·m/deg`;
            case RotationalStiffnessUnits.DecanewtonMetersPerDegree:
                return `daN·m/deg`;
            case RotationalStiffnessUnits.KilonewtonMetersPerDegree:
                return `kN·m/deg`;
            case RotationalStiffnessUnits.MeganewtonMetersPerDegree:
                return `MN·m/deg`;
            case RotationalStiffnessUnits.NanonewtonMillimetersPerRadian:
                return `nN·mm/rad`;
            case RotationalStiffnessUnits.MicronewtonMillimetersPerRadian:
                return `μN·mm/rad`;
            case RotationalStiffnessUnits.MillinewtonMillimetersPerRadian:
                return `mN·mm/rad`;
            case RotationalStiffnessUnits.CentinewtonMillimetersPerRadian:
                return `cN·mm/rad`;
            case RotationalStiffnessUnits.DecinewtonMillimetersPerRadian:
                return `dN·mm/rad`;
            case RotationalStiffnessUnits.DecanewtonMillimetersPerRadian:
                return `daN·mm/rad`;
            case RotationalStiffnessUnits.KilonewtonMillimetersPerRadian:
                return `kN·mm/rad`;
            case RotationalStiffnessUnits.MeganewtonMillimetersPerRadian:
                return `MN·mm/rad`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given RotationalStiffness are equals to the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns True if the given RotationalStiffness are equal to the current RotationalStiffness.
     */
    equals(rotationalStiffness) {
        return this.value === rotationalStiffness.BaseValue;
    }
    /**
     * Compare the given RotationalStiffness against the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns 0 if they are equal, -1 if the current RotationalStiffness is less then other, 1 if the current RotationalStiffness is greater then other.
     */
    compareTo(rotationalStiffness) {
        if (this.value > rotationalStiffness.BaseValue)
            return 1;
        if (this.value < rotationalStiffness.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given RotationalStiffness with the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns A new RotationalStiffness instance with the results.
     */
    add(rotationalStiffness) {
        return new RotationalStiffness(this.value + rotationalStiffness.BaseValue);
    }
    /**
     * Subtract the given RotationalStiffness with the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns A new RotationalStiffness instance with the results.
     */
    subtract(rotationalStiffness) {
        return new RotationalStiffness(this.value - rotationalStiffness.BaseValue);
    }
    /**
     * Multiply the given RotationalStiffness with the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns A new RotationalStiffness instance with the results.
     */
    multiply(rotationalStiffness) {
        return new RotationalStiffness(this.value * rotationalStiffness.BaseValue);
    }
    /**
     * Divide the given RotationalStiffness with the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns A new RotationalStiffness instance with the results.
     */
    divide(rotationalStiffness) {
        return new RotationalStiffness(this.value / rotationalStiffness.BaseValue);
    }
    /**
     * Modulo the given RotationalStiffness with the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns A new RotationalStiffness instance with the results.
     */
    modulo(rotationalStiffness) {
        return new RotationalStiffness(this.value % rotationalStiffness.BaseValue);
    }
    /**
     * Pow the given RotationalStiffness with the current RotationalStiffness.
     * @param rotationalStiffness The other RotationalStiffness.
     * @returns A new RotationalStiffness instance with the results.
     */
    pow(rotationalStiffness) {
        return new RotationalStiffness(this.value ** rotationalStiffness.BaseValue);
    }
}
exports.RotationalStiffness = RotationalStiffness;
