// unitsnet-js/dist/specificenergy.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpecificEnergy = exports.SpecificEnergyUnits = void 0;
/** SpecificEnergyUnits enumeration */
var SpecificEnergyUnits;
(function (SpecificEnergyUnits) {
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["JoulesPerKilogram"] = 0] = "JoulesPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegaJoulesPerTonne"] = 1] = "MegaJoulesPerTonne";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["CaloriesPerGram"] = 2] = "CaloriesPerGram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["WattHoursPerKilogram"] = 3] = "WattHoursPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["WattDaysPerKilogram"] = 4] = "WattDaysPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["WattDaysPerTonne"] = 5] = "WattDaysPerTonne";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["WattDaysPerShortTon"] = 6] = "WattDaysPerShortTon";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["WattHoursPerPound"] = 7] = "WattHoursPerPound";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["BtuPerPound"] = 8] = "BtuPerPound";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilojoulesPerKilogram"] = 9] = "KilojoulesPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegajoulesPerKilogram"] = 10] = "MegajoulesPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilocaloriesPerGram"] = 11] = "KilocaloriesPerGram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilowattHoursPerKilogram"] = 12] = "KilowattHoursPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegawattHoursPerKilogram"] = 13] = "MegawattHoursPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["GigawattHoursPerKilogram"] = 14] = "GigawattHoursPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilowattDaysPerKilogram"] = 15] = "KilowattDaysPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegawattDaysPerKilogram"] = 16] = "MegawattDaysPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["GigawattDaysPerKilogram"] = 17] = "GigawattDaysPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["TerawattDaysPerKilogram"] = 18] = "TerawattDaysPerKilogram";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilowattDaysPerTonne"] = 19] = "KilowattDaysPerTonne";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegawattDaysPerTonne"] = 20] = "MegawattDaysPerTonne";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["GigawattDaysPerTonne"] = 21] = "GigawattDaysPerTonne";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["TerawattDaysPerTonne"] = 22] = "TerawattDaysPerTonne";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilowattDaysPerShortTon"] = 23] = "KilowattDaysPerShortTon";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegawattDaysPerShortTon"] = 24] = "MegawattDaysPerShortTon";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["GigawattDaysPerShortTon"] = 25] = "GigawattDaysPerShortTon";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["TerawattDaysPerShortTon"] = 26] = "TerawattDaysPerShortTon";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["KilowattHoursPerPound"] = 27] = "KilowattHoursPerPound";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["MegawattHoursPerPound"] = 28] = "MegawattHoursPerPound";
    /** */
    SpecificEnergyUnits[SpecificEnergyUnits["GigawattHoursPerPound"] = 29] = "GigawattHoursPerPound";
})(SpecificEnergyUnits = exports.SpecificEnergyUnits || (exports.SpecificEnergyUnits = {}));
/** The SpecificEnergy */
class SpecificEnergy {
    /**
     * Create a new SpecificEnergy.
     * @param value The value.
     * @param fromUnit The ‘SpecificEnergy’ unit to create from.
     * The default unit is JoulesPerKilogram
     */
    constructor(value, fromUnit = SpecificEnergyUnits.JoulesPerKilogram) {
        this.joulesperkilogramLazy = null;
        this.megajoulespertonneLazy = null;
        this.caloriespergramLazy = null;
        this.watthoursperkilogramLazy = null;
        this.wattdaysperkilogramLazy = null;
        this.wattdayspertonneLazy = null;
        this.wattdayspershorttonLazy = null;
        this.watthoursperpoundLazy = null;
        this.btuperpoundLazy = null;
        this.kilojoulesperkilogramLazy = null;
        this.megajoulesperkilogramLazy = null;
        this.kilocaloriespergramLazy = null;
        this.kilowatthoursperkilogramLazy = null;
        this.megawatthoursperkilogramLazy = null;
        this.gigawatthoursperkilogramLazy = null;
        this.kilowattdaysperkilogramLazy = null;
        this.megawattdaysperkilogramLazy = null;
        this.gigawattdaysperkilogramLazy = null;
        this.terawattdaysperkilogramLazy = null;
        this.kilowattdayspertonneLazy = null;
        this.megawattdayspertonneLazy = null;
        this.gigawattdayspertonneLazy = null;
        this.terawattdayspertonneLazy = null;
        this.kilowattdayspershorttonLazy = null;
        this.megawattdayspershorttonLazy = null;
        this.gigawattdayspershorttonLazy = null;
        this.terawattdayspershorttonLazy = null;
        this.kilowatthoursperpoundLazy = null;
        this.megawatthoursperpoundLazy = null;
        this.gigawatthoursperpoundLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of SpecificEnergy is JoulesPerKilogram.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get JoulesPerKilogram() {
        if (this.joulesperkilogramLazy !== null) {
            return this.joulesperkilogramLazy;
        }
        return this.joulesperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.JoulesPerKilogram);
    }
    /** */
    get MegaJoulesPerTonne() {
        if (this.megajoulespertonneLazy !== null) {
            return this.megajoulespertonneLazy;
        }
        return this.megajoulespertonneLazy = this.convertFromBase(SpecificEnergyUnits.MegaJoulesPerTonne);
    }
    /** */
    get CaloriesPerGram() {
        if (this.caloriespergramLazy !== null) {
            return this.caloriespergramLazy;
        }
        return this.caloriespergramLazy = this.convertFromBase(SpecificEnergyUnits.CaloriesPerGram);
    }
    /** */
    get WattHoursPerKilogram() {
        if (this.watthoursperkilogramLazy !== null) {
            return this.watthoursperkilogramLazy;
        }
        return this.watthoursperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.WattHoursPerKilogram);
    }
    /** */
    get WattDaysPerKilogram() {
        if (this.wattdaysperkilogramLazy !== null) {
            return this.wattdaysperkilogramLazy;
        }
        return this.wattdaysperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.WattDaysPerKilogram);
    }
    /** */
    get WattDaysPerTonne() {
        if (this.wattdayspertonneLazy !== null) {
            return this.wattdayspertonneLazy;
        }
        return this.wattdayspertonneLazy = this.convertFromBase(SpecificEnergyUnits.WattDaysPerTonne);
    }
    /** */
    get WattDaysPerShortTon() {
        if (this.wattdayspershorttonLazy !== null) {
            return this.wattdayspershorttonLazy;
        }
        return this.wattdayspershorttonLazy = this.convertFromBase(SpecificEnergyUnits.WattDaysPerShortTon);
    }
    /** */
    get WattHoursPerPound() {
        if (this.watthoursperpoundLazy !== null) {
            return this.watthoursperpoundLazy;
        }
        return this.watthoursperpoundLazy = this.convertFromBase(SpecificEnergyUnits.WattHoursPerPound);
    }
    /** */
    get BtuPerPound() {
        if (this.btuperpoundLazy !== null) {
            return this.btuperpoundLazy;
        }
        return this.btuperpoundLazy = this.convertFromBase(SpecificEnergyUnits.BtuPerPound);
    }
    /** */
    get KilojoulesPerKilogram() {
        if (this.kilojoulesperkilogramLazy !== null) {
            return this.kilojoulesperkilogramLazy;
        }
        return this.kilojoulesperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.KilojoulesPerKilogram);
    }
    /** */
    get MegajoulesPerKilogram() {
        if (this.megajoulesperkilogramLazy !== null) {
            return this.megajoulesperkilogramLazy;
        }
        return this.megajoulesperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.MegajoulesPerKilogram);
    }
    /** */
    get KilocaloriesPerGram() {
        if (this.kilocaloriespergramLazy !== null) {
            return this.kilocaloriespergramLazy;
        }
        return this.kilocaloriespergramLazy = this.convertFromBase(SpecificEnergyUnits.KilocaloriesPerGram);
    }
    /** */
    get KilowattHoursPerKilogram() {
        if (this.kilowatthoursperkilogramLazy !== null) {
            return this.kilowatthoursperkilogramLazy;
        }
        return this.kilowatthoursperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.KilowattHoursPerKilogram);
    }
    /** */
    get MegawattHoursPerKilogram() {
        if (this.megawatthoursperkilogramLazy !== null) {
            return this.megawatthoursperkilogramLazy;
        }
        return this.megawatthoursperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.MegawattHoursPerKilogram);
    }
    /** */
    get GigawattHoursPerKilogram() {
        if (this.gigawatthoursperkilogramLazy !== null) {
            return this.gigawatthoursperkilogramLazy;
        }
        return this.gigawatthoursperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.GigawattHoursPerKilogram);
    }
    /** */
    get KilowattDaysPerKilogram() {
        if (this.kilowattdaysperkilogramLazy !== null) {
            return this.kilowattdaysperkilogramLazy;
        }
        return this.kilowattdaysperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.KilowattDaysPerKilogram);
    }
    /** */
    get MegawattDaysPerKilogram() {
        if (this.megawattdaysperkilogramLazy !== null) {
            return this.megawattdaysperkilogramLazy;
        }
        return this.megawattdaysperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.MegawattDaysPerKilogram);
    }
    /** */
    get GigawattDaysPerKilogram() {
        if (this.gigawattdaysperkilogramLazy !== null) {
            return this.gigawattdaysperkilogramLazy;
        }
        return this.gigawattdaysperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.GigawattDaysPerKilogram);
    }
    /** */
    get TerawattDaysPerKilogram() {
        if (this.terawattdaysperkilogramLazy !== null) {
            return this.terawattdaysperkilogramLazy;
        }
        return this.terawattdaysperkilogramLazy = this.convertFromBase(SpecificEnergyUnits.TerawattDaysPerKilogram);
    }
    /** */
    get KilowattDaysPerTonne() {
        if (this.kilowattdayspertonneLazy !== null) {
            return this.kilowattdayspertonneLazy;
        }
        return this.kilowattdayspertonneLazy = this.convertFromBase(SpecificEnergyUnits.KilowattDaysPerTonne);
    }
    /** */
    get MegawattDaysPerTonne() {
        if (this.megawattdayspertonneLazy !== null) {
            return this.megawattdayspertonneLazy;
        }
        return this.megawattdayspertonneLazy = this.convertFromBase(SpecificEnergyUnits.MegawattDaysPerTonne);
    }
    /** */
    get GigawattDaysPerTonne() {
        if (this.gigawattdayspertonneLazy !== null) {
            return this.gigawattdayspertonneLazy;
        }
        return this.gigawattdayspertonneLazy = this.convertFromBase(SpecificEnergyUnits.GigawattDaysPerTonne);
    }
    /** */
    get TerawattDaysPerTonne() {
        if (this.terawattdayspertonneLazy !== null) {
            return this.terawattdayspertonneLazy;
        }
        return this.terawattdayspertonneLazy = this.convertFromBase(SpecificEnergyUnits.TerawattDaysPerTonne);
    }
    /** */
    get KilowattDaysPerShortTon() {
        if (this.kilowattdayspershorttonLazy !== null) {
            return this.kilowattdayspershorttonLazy;
        }
        return this.kilowattdayspershorttonLazy = this.convertFromBase(SpecificEnergyUnits.KilowattDaysPerShortTon);
    }
    /** */
    get MegawattDaysPerShortTon() {
        if (this.megawattdayspershorttonLazy !== null) {
            return this.megawattdayspershorttonLazy;
        }
        return this.megawattdayspershorttonLazy = this.convertFromBase(SpecificEnergyUnits.MegawattDaysPerShortTon);
    }
    /** */
    get GigawattDaysPerShortTon() {
        if (this.gigawattdayspershorttonLazy !== null) {
            return this.gigawattdayspershorttonLazy;
        }
        return this.gigawattdayspershorttonLazy = this.convertFromBase(SpecificEnergyUnits.GigawattDaysPerShortTon);
    }
    /** */
    get TerawattDaysPerShortTon() {
        if (this.terawattdayspershorttonLazy !== null) {
            return this.terawattdayspershorttonLazy;
        }
        return this.terawattdayspershorttonLazy = this.convertFromBase(SpecificEnergyUnits.TerawattDaysPerShortTon);
    }
    /** */
    get KilowattHoursPerPound() {
        if (this.kilowatthoursperpoundLazy !== null) {
            return this.kilowatthoursperpoundLazy;
        }
        return this.kilowatthoursperpoundLazy = this.convertFromBase(SpecificEnergyUnits.KilowattHoursPerPound);
    }
    /** */
    get MegawattHoursPerPound() {
        if (this.megawatthoursperpoundLazy !== null) {
            return this.megawatthoursperpoundLazy;
        }
        return this.megawatthoursperpoundLazy = this.convertFromBase(SpecificEnergyUnits.MegawattHoursPerPound);
    }
    /** */
    get GigawattHoursPerPound() {
        if (this.gigawatthoursperpoundLazy !== null) {
            return this.gigawatthoursperpoundLazy;
        }
        return this.gigawatthoursperpoundLazy = this.convertFromBase(SpecificEnergyUnits.GigawattHoursPerPound);
    }
    /**
     * Create a new SpecificEnergy instance from a JoulesPerKilogram
     *
     * @param value The unit as JoulesPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromJoulesPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.JoulesPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a MegaJoulesPerTonne
     *
     * @param value The unit as MegaJoulesPerTonne to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegaJoulesPerTonne(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegaJoulesPerTonne);
    }
    /**
     * Create a new SpecificEnergy instance from a CaloriesPerGram
     *
     * @param value The unit as CaloriesPerGram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromCaloriesPerGram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.CaloriesPerGram);
    }
    /**
     * Create a new SpecificEnergy instance from a WattHoursPerKilogram
     *
     * @param value The unit as WattHoursPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromWattHoursPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.WattHoursPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a WattDaysPerKilogram
     *
     * @param value The unit as WattDaysPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromWattDaysPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.WattDaysPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a WattDaysPerTonne
     *
     * @param value The unit as WattDaysPerTonne to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromWattDaysPerTonne(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.WattDaysPerTonne);
    }
    /**
     * Create a new SpecificEnergy instance from a WattDaysPerShortTon
     *
     * @param value The unit as WattDaysPerShortTon to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromWattDaysPerShortTon(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.WattDaysPerShortTon);
    }
    /**
     * Create a new SpecificEnergy instance from a WattHoursPerPound
     *
     * @param value The unit as WattHoursPerPound to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromWattHoursPerPound(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.WattHoursPerPound);
    }
    /**
     * Create a new SpecificEnergy instance from a BtuPerPound
     *
     * @param value The unit as BtuPerPound to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromBtuPerPound(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.BtuPerPound);
    }
    /**
     * Create a new SpecificEnergy instance from a KilojoulesPerKilogram
     *
     * @param value The unit as KilojoulesPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilojoulesPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilojoulesPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a MegajoulesPerKilogram
     *
     * @param value The unit as MegajoulesPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegajoulesPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegajoulesPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a KilocaloriesPerGram
     *
     * @param value The unit as KilocaloriesPerGram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilocaloriesPerGram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilocaloriesPerGram);
    }
    /**
     * Create a new SpecificEnergy instance from a KilowattHoursPerKilogram
     *
     * @param value The unit as KilowattHoursPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilowattHoursPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilowattHoursPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a MegawattHoursPerKilogram
     *
     * @param value The unit as MegawattHoursPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegawattHoursPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegawattHoursPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a GigawattHoursPerKilogram
     *
     * @param value The unit as GigawattHoursPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromGigawattHoursPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.GigawattHoursPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a KilowattDaysPerKilogram
     *
     * @param value The unit as KilowattDaysPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilowattDaysPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilowattDaysPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a MegawattDaysPerKilogram
     *
     * @param value The unit as MegawattDaysPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegawattDaysPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegawattDaysPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a GigawattDaysPerKilogram
     *
     * @param value The unit as GigawattDaysPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromGigawattDaysPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.GigawattDaysPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a TerawattDaysPerKilogram
     *
     * @param value The unit as TerawattDaysPerKilogram to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromTerawattDaysPerKilogram(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.TerawattDaysPerKilogram);
    }
    /**
     * Create a new SpecificEnergy instance from a KilowattDaysPerTonne
     *
     * @param value The unit as KilowattDaysPerTonne to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilowattDaysPerTonne(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilowattDaysPerTonne);
    }
    /**
     * Create a new SpecificEnergy instance from a MegawattDaysPerTonne
     *
     * @param value The unit as MegawattDaysPerTonne to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegawattDaysPerTonne(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegawattDaysPerTonne);
    }
    /**
     * Create a new SpecificEnergy instance from a GigawattDaysPerTonne
     *
     * @param value The unit as GigawattDaysPerTonne to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromGigawattDaysPerTonne(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.GigawattDaysPerTonne);
    }
    /**
     * Create a new SpecificEnergy instance from a TerawattDaysPerTonne
     *
     * @param value The unit as TerawattDaysPerTonne to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromTerawattDaysPerTonne(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.TerawattDaysPerTonne);
    }
    /**
     * Create a new SpecificEnergy instance from a KilowattDaysPerShortTon
     *
     * @param value The unit as KilowattDaysPerShortTon to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilowattDaysPerShortTon(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilowattDaysPerShortTon);
    }
    /**
     * Create a new SpecificEnergy instance from a MegawattDaysPerShortTon
     *
     * @param value The unit as MegawattDaysPerShortTon to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegawattDaysPerShortTon(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegawattDaysPerShortTon);
    }
    /**
     * Create a new SpecificEnergy instance from a GigawattDaysPerShortTon
     *
     * @param value The unit as GigawattDaysPerShortTon to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromGigawattDaysPerShortTon(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.GigawattDaysPerShortTon);
    }
    /**
     * Create a new SpecificEnergy instance from a TerawattDaysPerShortTon
     *
     * @param value The unit as TerawattDaysPerShortTon to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromTerawattDaysPerShortTon(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.TerawattDaysPerShortTon);
    }
    /**
     * Create a new SpecificEnergy instance from a KilowattHoursPerPound
     *
     * @param value The unit as KilowattHoursPerPound to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromKilowattHoursPerPound(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.KilowattHoursPerPound);
    }
    /**
     * Create a new SpecificEnergy instance from a MegawattHoursPerPound
     *
     * @param value The unit as MegawattHoursPerPound to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromMegawattHoursPerPound(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.MegawattHoursPerPound);
    }
    /**
     * Create a new SpecificEnergy instance from a GigawattHoursPerPound
     *
     * @param value The unit as GigawattHoursPerPound to create a new SpecificEnergy from.
     * @returns The new SpecificEnergy instance.
     */
    static FromGigawattHoursPerPound(value) {
        return new SpecificEnergy(value, SpecificEnergyUnits.GigawattHoursPerPound);
    }
    /**
     * Convert SpecificEnergy to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case SpecificEnergyUnits.JoulesPerKilogram: return this.JoulesPerKilogram;
            case SpecificEnergyUnits.MegaJoulesPerTonne: return this.MegaJoulesPerTonne;
            case SpecificEnergyUnits.CaloriesPerGram: return this.CaloriesPerGram;
            case SpecificEnergyUnits.WattHoursPerKilogram: return this.WattHoursPerKilogram;
            case SpecificEnergyUnits.WattDaysPerKilogram: return this.WattDaysPerKilogram;
            case SpecificEnergyUnits.WattDaysPerTonne: return this.WattDaysPerTonne;
            case SpecificEnergyUnits.WattDaysPerShortTon: return this.WattDaysPerShortTon;
            case SpecificEnergyUnits.WattHoursPerPound: return this.WattHoursPerPound;
            case SpecificEnergyUnits.BtuPerPound: return this.BtuPerPound;
            case SpecificEnergyUnits.KilojoulesPerKilogram: return this.KilojoulesPerKilogram;
            case SpecificEnergyUnits.MegajoulesPerKilogram: return this.MegajoulesPerKilogram;
            case SpecificEnergyUnits.KilocaloriesPerGram: return this.KilocaloriesPerGram;
            case SpecificEnergyUnits.KilowattHoursPerKilogram: return this.KilowattHoursPerKilogram;
            case SpecificEnergyUnits.MegawattHoursPerKilogram: return this.MegawattHoursPerKilogram;
            case SpecificEnergyUnits.GigawattHoursPerKilogram: return this.GigawattHoursPerKilogram;
            case SpecificEnergyUnits.KilowattDaysPerKilogram: return this.KilowattDaysPerKilogram;
            case SpecificEnergyUnits.MegawattDaysPerKilogram: return this.MegawattDaysPerKilogram;
            case SpecificEnergyUnits.GigawattDaysPerKilogram: return this.GigawattDaysPerKilogram;
            case SpecificEnergyUnits.TerawattDaysPerKilogram: return this.TerawattDaysPerKilogram;
            case SpecificEnergyUnits.KilowattDaysPerTonne: return this.KilowattDaysPerTonne;
            case SpecificEnergyUnits.MegawattDaysPerTonne: return this.MegawattDaysPerTonne;
            case SpecificEnergyUnits.GigawattDaysPerTonne: return this.GigawattDaysPerTonne;
            case SpecificEnergyUnits.TerawattDaysPerTonne: return this.TerawattDaysPerTonne;
            case SpecificEnergyUnits.KilowattDaysPerShortTon: return this.KilowattDaysPerShortTon;
            case SpecificEnergyUnits.MegawattDaysPerShortTon: return this.MegawattDaysPerShortTon;
            case SpecificEnergyUnits.GigawattDaysPerShortTon: return this.GigawattDaysPerShortTon;
            case SpecificEnergyUnits.TerawattDaysPerShortTon: return this.TerawattDaysPerShortTon;
            case SpecificEnergyUnits.KilowattHoursPerPound: return this.KilowattHoursPerPound;
            case SpecificEnergyUnits.MegawattHoursPerPound: return this.MegawattHoursPerPound;
            case SpecificEnergyUnits.GigawattHoursPerPound: return this.GigawattHoursPerPound;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case SpecificEnergyUnits.JoulesPerKilogram:
                return this.value;
            case SpecificEnergyUnits.MegaJoulesPerTonne:
                return this.value / 1e3;
            case SpecificEnergyUnits.CaloriesPerGram:
                return this.value / 4.184e3;
            case SpecificEnergyUnits.WattHoursPerKilogram:
                return this.value / 3.6e3;
            case SpecificEnergyUnits.WattDaysPerKilogram:
                return this.value / (24 * 3.6e3);
            case SpecificEnergyUnits.WattDaysPerTonne:
                return this.value / ((24 * 3.6e3) / 1e3);
            case SpecificEnergyUnits.WattDaysPerShortTon:
                return this.value / ((24 * 3.6e3) / 9.0718474e2);
            case SpecificEnergyUnits.WattHoursPerPound:
                return this.value / 7.93664e3;
            case SpecificEnergyUnits.BtuPerPound:
                return this.value / 2326.000075362;
            case SpecificEnergyUnits.KilojoulesPerKilogram:
                return (this.value) / 1000;
            case SpecificEnergyUnits.MegajoulesPerKilogram:
                return (this.value) / 1000000;
            case SpecificEnergyUnits.KilocaloriesPerGram:
                return (this.value / 4.184e3) / 1000;
            case SpecificEnergyUnits.KilowattHoursPerKilogram:
                return (this.value / 3.6e3) / 1000;
            case SpecificEnergyUnits.MegawattHoursPerKilogram:
                return (this.value / 3.6e3) / 1000000;
            case SpecificEnergyUnits.GigawattHoursPerKilogram:
                return (this.value / 3.6e3) / 1000000000;
            case SpecificEnergyUnits.KilowattDaysPerKilogram:
                return (this.value / (24 * 3.6e3)) / 1000;
            case SpecificEnergyUnits.MegawattDaysPerKilogram:
                return (this.value / (24 * 3.6e3)) / 1000000;
            case SpecificEnergyUnits.GigawattDaysPerKilogram:
                return (this.value / (24 * 3.6e3)) / 1000000000;
            case SpecificEnergyUnits.TerawattDaysPerKilogram:
                return (this.value / (24 * 3.6e3)) / 1000000000000;
            case SpecificEnergyUnits.KilowattDaysPerTonne:
                return (this.value / ((24 * 3.6e3) / 1e3)) / 1000;
            case SpecificEnergyUnits.MegawattDaysPerTonne:
                return (this.value / ((24 * 3.6e3) / 1e3)) / 1000000;
            case SpecificEnergyUnits.GigawattDaysPerTonne:
                return (this.value / ((24 * 3.6e3) / 1e3)) / 1000000000;
            case SpecificEnergyUnits.TerawattDaysPerTonne:
                return (this.value / ((24 * 3.6e3) / 1e3)) / 1000000000000;
            case SpecificEnergyUnits.KilowattDaysPerShortTon:
                return (this.value / ((24 * 3.6e3) / 9.0718474e2)) / 1000;
            case SpecificEnergyUnits.MegawattDaysPerShortTon:
                return (this.value / ((24 * 3.6e3) / 9.0718474e2)) / 1000000;
            case SpecificEnergyUnits.GigawattDaysPerShortTon:
                return (this.value / ((24 * 3.6e3) / 9.0718474e2)) / 1000000000;
            case SpecificEnergyUnits.TerawattDaysPerShortTon:
                return (this.value / ((24 * 3.6e3) / 9.0718474e2)) / 1000000000000;
            case SpecificEnergyUnits.KilowattHoursPerPound:
                return (this.value / 7.93664e3) / 1000;
            case SpecificEnergyUnits.MegawattHoursPerPound:
                return (this.value / 7.93664e3) / 1000000;
            case SpecificEnergyUnits.GigawattHoursPerPound:
                return (this.value / 7.93664e3) / 1000000000;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case SpecificEnergyUnits.JoulesPerKilogram:
                return value;
            case SpecificEnergyUnits.MegaJoulesPerTonne:
                return value * 1e3;
            case SpecificEnergyUnits.CaloriesPerGram:
                return value * 4.184e3;
            case SpecificEnergyUnits.WattHoursPerKilogram:
                return value * 3.6e3;
            case SpecificEnergyUnits.WattDaysPerKilogram:
                return value * (24 * 3.6e3);
            case SpecificEnergyUnits.WattDaysPerTonne:
                return value * ((24 * 3.6e3) / 1e3);
            case SpecificEnergyUnits.WattDaysPerShortTon:
                return value * ((24 * 3.6e3) / 9.0718474e2);
            case SpecificEnergyUnits.WattHoursPerPound:
                return value * 7.93664e3;
            case SpecificEnergyUnits.BtuPerPound:
                return value * 2326.000075362;
            case SpecificEnergyUnits.KilojoulesPerKilogram:
                return (value) * 1000;
            case SpecificEnergyUnits.MegajoulesPerKilogram:
                return (value) * 1000000;
            case SpecificEnergyUnits.KilocaloriesPerGram:
                return (value * 4.184e3) * 1000;
            case SpecificEnergyUnits.KilowattHoursPerKilogram:
                return (value * 3.6e3) * 1000;
            case SpecificEnergyUnits.MegawattHoursPerKilogram:
                return (value * 3.6e3) * 1000000;
            case SpecificEnergyUnits.GigawattHoursPerKilogram:
                return (value * 3.6e3) * 1000000000;
            case SpecificEnergyUnits.KilowattDaysPerKilogram:
                return (value * (24 * 3.6e3)) * 1000;
            case SpecificEnergyUnits.MegawattDaysPerKilogram:
                return (value * (24 * 3.6e3)) * 1000000;
            case SpecificEnergyUnits.GigawattDaysPerKilogram:
                return (value * (24 * 3.6e3)) * 1000000000;
            case SpecificEnergyUnits.TerawattDaysPerKilogram:
                return (value * (24 * 3.6e3)) * 1000000000000;
            case SpecificEnergyUnits.KilowattDaysPerTonne:
                return (value * ((24 * 3.6e3) / 1e3)) * 1000;
            case SpecificEnergyUnits.MegawattDaysPerTonne:
                return (value * ((24 * 3.6e3) / 1e3)) * 1000000;
            case SpecificEnergyUnits.GigawattDaysPerTonne:
                return (value * ((24 * 3.6e3) / 1e3)) * 1000000000;
            case SpecificEnergyUnits.TerawattDaysPerTonne:
                return (value * ((24 * 3.6e3) / 1e3)) * 1000000000000;
            case SpecificEnergyUnits.KilowattDaysPerShortTon:
                return (value * ((24 * 3.6e3) / 9.0718474e2)) * 1000;
            case SpecificEnergyUnits.MegawattDaysPerShortTon:
                return (value * ((24 * 3.6e3) / 9.0718474e2)) * 1000000;
            case SpecificEnergyUnits.GigawattDaysPerShortTon:
                return (value * ((24 * 3.6e3) / 9.0718474e2)) * 1000000000;
            case SpecificEnergyUnits.TerawattDaysPerShortTon:
                return (value * ((24 * 3.6e3) / 9.0718474e2)) * 1000000000000;
            case SpecificEnergyUnits.KilowattHoursPerPound:
                return (value * 7.93664e3) * 1000;
            case SpecificEnergyUnits.MegawattHoursPerPound:
                return (value * 7.93664e3) * 1000000;
            case SpecificEnergyUnits.GigawattHoursPerPound:
                return (value * 7.93664e3) * 1000000000;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the SpecificEnergy to string.
     * Note! the default format for SpecificEnergy is JoulesPerKilogram.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the SpecificEnergy.
     * @returns The string format of the SpecificEnergy.
     */
    toString(unit = SpecificEnergyUnits.JoulesPerKilogram) {
        switch (unit) {
            case SpecificEnergyUnits.JoulesPerKilogram:
                return this.JoulesPerKilogram + ` J/kg`;
            case SpecificEnergyUnits.MegaJoulesPerTonne:
                return this.MegaJoulesPerTonne + ` MJ/t`;
            case SpecificEnergyUnits.CaloriesPerGram:
                return this.CaloriesPerGram + ` cal/g`;
            case SpecificEnergyUnits.WattHoursPerKilogram:
                return this.WattHoursPerKilogram + ` Wh/kg`;
            case SpecificEnergyUnits.WattDaysPerKilogram:
                return this.WattDaysPerKilogram + ` Wd/kg`;
            case SpecificEnergyUnits.WattDaysPerTonne:
                return this.WattDaysPerTonne + ` Wd/t`;
            case SpecificEnergyUnits.WattDaysPerShortTon:
                return this.WattDaysPerShortTon + ` Wd/ST`;
            case SpecificEnergyUnits.WattHoursPerPound:
                return this.WattHoursPerPound + ` Wh/lbs`;
            case SpecificEnergyUnits.BtuPerPound:
                return this.BtuPerPound + ` btu/lb`;
            case SpecificEnergyUnits.KilojoulesPerKilogram:
                return this.KilojoulesPerKilogram + ` kJ/kg`;
            case SpecificEnergyUnits.MegajoulesPerKilogram:
                return this.MegajoulesPerKilogram + ` MJ/kg`;
            case SpecificEnergyUnits.KilocaloriesPerGram:
                return this.KilocaloriesPerGram + ` kcal/g`;
            case SpecificEnergyUnits.KilowattHoursPerKilogram:
                return this.KilowattHoursPerKilogram + ` kWh/kg`;
            case SpecificEnergyUnits.MegawattHoursPerKilogram:
                return this.MegawattHoursPerKilogram + ` MWh/kg`;
            case SpecificEnergyUnits.GigawattHoursPerKilogram:
                return this.GigawattHoursPerKilogram + ` GWh/kg`;
            case SpecificEnergyUnits.KilowattDaysPerKilogram:
                return this.KilowattDaysPerKilogram + ` kWd/kg`;
            case SpecificEnergyUnits.MegawattDaysPerKilogram:
                return this.MegawattDaysPerKilogram + ` MWd/kg`;
            case SpecificEnergyUnits.GigawattDaysPerKilogram:
                return this.GigawattDaysPerKilogram + ` GWd/kg`;
            case SpecificEnergyUnits.TerawattDaysPerKilogram:
                return this.TerawattDaysPerKilogram + ` TWd/kg`;
            case SpecificEnergyUnits.KilowattDaysPerTonne:
                return this.KilowattDaysPerTonne + ` kWd/t`;
            case SpecificEnergyUnits.MegawattDaysPerTonne:
                return this.MegawattDaysPerTonne + ` MWd/t`;
            case SpecificEnergyUnits.GigawattDaysPerTonne:
                return this.GigawattDaysPerTonne + ` GWd/t`;
            case SpecificEnergyUnits.TerawattDaysPerTonne:
                return this.TerawattDaysPerTonne + ` TWd/t`;
            case SpecificEnergyUnits.KilowattDaysPerShortTon:
                return this.KilowattDaysPerShortTon + ` kWd/ST`;
            case SpecificEnergyUnits.MegawattDaysPerShortTon:
                return this.MegawattDaysPerShortTon + ` MWd/ST`;
            case SpecificEnergyUnits.GigawattDaysPerShortTon:
                return this.GigawattDaysPerShortTon + ` GWd/ST`;
            case SpecificEnergyUnits.TerawattDaysPerShortTon:
                return this.TerawattDaysPerShortTon + ` TWd/ST`;
            case SpecificEnergyUnits.KilowattHoursPerPound:
                return this.KilowattHoursPerPound + ` kWh/lbs`;
            case SpecificEnergyUnits.MegawattHoursPerPound:
                return this.MegawattHoursPerPound + ` MWh/lbs`;
            case SpecificEnergyUnits.GigawattHoursPerPound:
                return this.GigawattHoursPerPound + ` GWh/lbs`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get SpecificEnergy unit abbreviation.
     * Note! the default abbreviation for SpecificEnergy is JoulesPerKilogram.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the SpecificEnergy.
     * @returns The abbreviation string of SpecificEnergy.
     */
    getUnitAbbreviation(unitAbbreviation = SpecificEnergyUnits.JoulesPerKilogram) {
        switch (unitAbbreviation) {
            case SpecificEnergyUnits.JoulesPerKilogram:
                return `J/kg`;
            case SpecificEnergyUnits.MegaJoulesPerTonne:
                return `MJ/t`;
            case SpecificEnergyUnits.CaloriesPerGram:
                return `cal/g`;
            case SpecificEnergyUnits.WattHoursPerKilogram:
                return `Wh/kg`;
            case SpecificEnergyUnits.WattDaysPerKilogram:
                return `Wd/kg`;
            case SpecificEnergyUnits.WattDaysPerTonne:
                return `Wd/t`;
            case SpecificEnergyUnits.WattDaysPerShortTon:
                return `Wd/ST`;
            case SpecificEnergyUnits.WattHoursPerPound:
                return `Wh/lbs`;
            case SpecificEnergyUnits.BtuPerPound:
                return `btu/lb`;
            case SpecificEnergyUnits.KilojoulesPerKilogram:
                return `kJ/kg`;
            case SpecificEnergyUnits.MegajoulesPerKilogram:
                return `MJ/kg`;
            case SpecificEnergyUnits.KilocaloriesPerGram:
                return `kcal/g`;
            case SpecificEnergyUnits.KilowattHoursPerKilogram:
                return `kWh/kg`;
            case SpecificEnergyUnits.MegawattHoursPerKilogram:
                return `MWh/kg`;
            case SpecificEnergyUnits.GigawattHoursPerKilogram:
                return `GWh/kg`;
            case SpecificEnergyUnits.KilowattDaysPerKilogram:
                return `kWd/kg`;
            case SpecificEnergyUnits.MegawattDaysPerKilogram:
                return `MWd/kg`;
            case SpecificEnergyUnits.GigawattDaysPerKilogram:
                return `GWd/kg`;
            case SpecificEnergyUnits.TerawattDaysPerKilogram:
                return `TWd/kg`;
            case SpecificEnergyUnits.KilowattDaysPerTonne:
                return `kWd/t`;
            case SpecificEnergyUnits.MegawattDaysPerTonne:
                return `MWd/t`;
            case SpecificEnergyUnits.GigawattDaysPerTonne:
                return `GWd/t`;
            case SpecificEnergyUnits.TerawattDaysPerTonne:
                return `TWd/t`;
            case SpecificEnergyUnits.KilowattDaysPerShortTon:
                return `kWd/ST`;
            case SpecificEnergyUnits.MegawattDaysPerShortTon:
                return `MWd/ST`;
            case SpecificEnergyUnits.GigawattDaysPerShortTon:
                return `GWd/ST`;
            case SpecificEnergyUnits.TerawattDaysPerShortTon:
                return `TWd/ST`;
            case SpecificEnergyUnits.KilowattHoursPerPound:
                return `kWh/lbs`;
            case SpecificEnergyUnits.MegawattHoursPerPound:
                return `MWh/lbs`;
            case SpecificEnergyUnits.GigawattHoursPerPound:
                return `GWh/lbs`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given SpecificEnergy are equals to the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns True if the given SpecificEnergy are equal to the current SpecificEnergy.
     */
    equals(specificEnergy) {
        return this.value === specificEnergy.BaseValue;
    }
    /**
     * Compare the given SpecificEnergy against the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns 0 if they are equal, -1 if the current SpecificEnergy is less then other, 1 if the current SpecificEnergy is greater then other.
     */
    compareTo(specificEnergy) {
        if (this.value > specificEnergy.BaseValue)
            return 1;
        if (this.value < specificEnergy.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given SpecificEnergy with the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns A new SpecificEnergy instance with the results.
     */
    add(specificEnergy) {
        return new SpecificEnergy(this.value + specificEnergy.BaseValue);
    }
    /**
     * Subtract the given SpecificEnergy with the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns A new SpecificEnergy instance with the results.
     */
    subtract(specificEnergy) {
        return new SpecificEnergy(this.value - specificEnergy.BaseValue);
    }
    /**
     * Multiply the given SpecificEnergy with the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns A new SpecificEnergy instance with the results.
     */
    multiply(specificEnergy) {
        return new SpecificEnergy(this.value * specificEnergy.BaseValue);
    }
    /**
     * Divide the given SpecificEnergy with the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns A new SpecificEnergy instance with the results.
     */
    divide(specificEnergy) {
        return new SpecificEnergy(this.value / specificEnergy.BaseValue);
    }
    /**
     * Modulo the given SpecificEnergy with the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns A new SpecificEnergy instance with the results.
     */
    modulo(specificEnergy) {
        return new SpecificEnergy(this.value % specificEnergy.BaseValue);
    }
    /**
     * Pow the given SpecificEnergy with the current SpecificEnergy.
     * @param specificEnergy The other SpecificEnergy.
     * @returns A new SpecificEnergy instance with the results.
     */
    pow(specificEnergy) {
        return new SpecificEnergy(this.value ** specificEnergy.BaseValue);
    }
}
exports.SpecificEnergy = SpecificEnergy;
