// unitsnet-js/dist/temperaturegradient.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemperatureGradient = exports.TemperatureGradientUnits = void 0;
/** TemperatureGradientUnits enumeration */
var TemperatureGradientUnits;
(function (TemperatureGradientUnits) {
    /** */
    TemperatureGradientUnits[TemperatureGradientUnits["KelvinsPerMeter"] = 0] = "KelvinsPerMeter";
    /** */
    TemperatureGradientUnits[TemperatureGradientUnits["DegreesCelciusPerMeter"] = 1] = "DegreesCelciusPerMeter";
    /** */
    TemperatureGradientUnits[TemperatureGradientUnits["DegreesFahrenheitPerFoot"] = 2] = "DegreesFahrenheitPerFoot";
    /** */
    TemperatureGradientUnits[TemperatureGradientUnits["DegreesCelciusPerKilometer"] = 3] = "DegreesCelciusPerKilometer";
})(TemperatureGradientUnits = exports.TemperatureGradientUnits || (exports.TemperatureGradientUnits = {}));
/** The rate of change of temperature with displacement in a given direction (as with increase of height) */
class TemperatureGradient {
    /**
     * Create a new TemperatureGradient.
     * @param value The value.
     * @param fromUnit The ‘TemperatureGradient’ unit to create from.
     * The default unit is KelvinsPerMeter
     */
    constructor(value, fromUnit = TemperatureGradientUnits.KelvinsPerMeter) {
        this.kelvinspermeterLazy = null;
        this.degreescelciuspermeterLazy = null;
        this.degreesfahrenheitperfootLazy = null;
        this.degreescelciusperkilometerLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of TemperatureGradient is KelvinsPerMeter.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get KelvinsPerMeter() {
        if (this.kelvinspermeterLazy !== null) {
            return this.kelvinspermeterLazy;
        }
        return this.kelvinspermeterLazy = this.convertFromBase(TemperatureGradientUnits.KelvinsPerMeter);
    }
    /** */
    get DegreesCelciusPerMeter() {
        if (this.degreescelciuspermeterLazy !== null) {
            return this.degreescelciuspermeterLazy;
        }
        return this.degreescelciuspermeterLazy = this.convertFromBase(TemperatureGradientUnits.DegreesCelciusPerMeter);
    }
    /** */
    get DegreesFahrenheitPerFoot() {
        if (this.degreesfahrenheitperfootLazy !== null) {
            return this.degreesfahrenheitperfootLazy;
        }
        return this.degreesfahrenheitperfootLazy = this.convertFromBase(TemperatureGradientUnits.DegreesFahrenheitPerFoot);
    }
    /** */
    get DegreesCelciusPerKilometer() {
        if (this.degreescelciusperkilometerLazy !== null) {
            return this.degreescelciusperkilometerLazy;
        }
        return this.degreescelciusperkilometerLazy = this.convertFromBase(TemperatureGradientUnits.DegreesCelciusPerKilometer);
    }
    /**
     * Create a new TemperatureGradient instance from a KelvinsPerMeter
     *
     * @param value The unit as KelvinsPerMeter to create a new TemperatureGradient from.
     * @returns The new TemperatureGradient instance.
     */
    static FromKelvinsPerMeter(value) {
        return new TemperatureGradient(value, TemperatureGradientUnits.KelvinsPerMeter);
    }
    /**
     * Create a new TemperatureGradient instance from a DegreesCelciusPerMeter
     *
     * @param value The unit as DegreesCelciusPerMeter to create a new TemperatureGradient from.
     * @returns The new TemperatureGradient instance.
     */
    static FromDegreesCelciusPerMeter(value) {
        return new TemperatureGradient(value, TemperatureGradientUnits.DegreesCelciusPerMeter);
    }
    /**
     * Create a new TemperatureGradient instance from a DegreesFahrenheitPerFoot
     *
     * @param value The unit as DegreesFahrenheitPerFoot to create a new TemperatureGradient from.
     * @returns The new TemperatureGradient instance.
     */
    static FromDegreesFahrenheitPerFoot(value) {
        return new TemperatureGradient(value, TemperatureGradientUnits.DegreesFahrenheitPerFoot);
    }
    /**
     * Create a new TemperatureGradient instance from a DegreesCelciusPerKilometer
     *
     * @param value The unit as DegreesCelciusPerKilometer to create a new TemperatureGradient from.
     * @returns The new TemperatureGradient instance.
     */
    static FromDegreesCelciusPerKilometer(value) {
        return new TemperatureGradient(value, TemperatureGradientUnits.DegreesCelciusPerKilometer);
    }
    /**
     * Convert TemperatureGradient to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case TemperatureGradientUnits.KelvinsPerMeter: return this.KelvinsPerMeter;
            case TemperatureGradientUnits.DegreesCelciusPerMeter: return this.DegreesCelciusPerMeter;
            case TemperatureGradientUnits.DegreesFahrenheitPerFoot: return this.DegreesFahrenheitPerFoot;
            case TemperatureGradientUnits.DegreesCelciusPerKilometer: return this.DegreesCelciusPerKilometer;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case TemperatureGradientUnits.KelvinsPerMeter:
                return this.value;
            case TemperatureGradientUnits.DegreesCelciusPerMeter:
                return this.value;
            case TemperatureGradientUnits.DegreesFahrenheitPerFoot:
                return (this.value * 0.3048) * 9 / 5;
            case TemperatureGradientUnits.DegreesCelciusPerKilometer:
                return this.value * 1e3;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case TemperatureGradientUnits.KelvinsPerMeter:
                return value;
            case TemperatureGradientUnits.DegreesCelciusPerMeter:
                return value;
            case TemperatureGradientUnits.DegreesFahrenheitPerFoot:
                return (value / 0.3048) * 5 / 9;
            case TemperatureGradientUnits.DegreesCelciusPerKilometer:
                return value / 1e3;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the TemperatureGradient to string.
     * Note! the default format for TemperatureGradient is KelvinsPerMeter.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the TemperatureGradient.
     * @returns The string format of the TemperatureGradient.
     */
    toString(unit = TemperatureGradientUnits.KelvinsPerMeter) {
        switch (unit) {
            case TemperatureGradientUnits.KelvinsPerMeter:
                return this.KelvinsPerMeter + ` ∆°K/m`;
            case TemperatureGradientUnits.DegreesCelciusPerMeter:
                return this.DegreesCelciusPerMeter + ` ∆°C/m`;
            case TemperatureGradientUnits.DegreesFahrenheitPerFoot:
                return this.DegreesFahrenheitPerFoot + ` ∆°F/ft`;
            case TemperatureGradientUnits.DegreesCelciusPerKilometer:
                return this.DegreesCelciusPerKilometer + ` ∆°C/km`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get TemperatureGradient unit abbreviation.
     * Note! the default abbreviation for TemperatureGradient is KelvinsPerMeter.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the TemperatureGradient.
     * @returns The abbreviation string of TemperatureGradient.
     */
    getUnitAbbreviation(unitAbbreviation = TemperatureGradientUnits.KelvinsPerMeter) {
        switch (unitAbbreviation) {
            case TemperatureGradientUnits.KelvinsPerMeter:
                return `∆°K/m`;
            case TemperatureGradientUnits.DegreesCelciusPerMeter:
                return `∆°C/m`;
            case TemperatureGradientUnits.DegreesFahrenheitPerFoot:
                return `∆°F/ft`;
            case TemperatureGradientUnits.DegreesCelciusPerKilometer:
                return `∆°C/km`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given TemperatureGradient are equals to the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns True if the given TemperatureGradient are equal to the current TemperatureGradient.
     */
    equals(temperatureGradient) {
        return this.value === temperatureGradient.BaseValue;
    }
    /**
     * Compare the given TemperatureGradient against the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns 0 if they are equal, -1 if the current TemperatureGradient is less then other, 1 if the current TemperatureGradient is greater then other.
     */
    compareTo(temperatureGradient) {
        if (this.value > temperatureGradient.BaseValue)
            return 1;
        if (this.value < temperatureGradient.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given TemperatureGradient with the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns A new TemperatureGradient instance with the results.
     */
    add(temperatureGradient) {
        return new TemperatureGradient(this.value + temperatureGradient.BaseValue);
    }
    /**
     * Subtract the given TemperatureGradient with the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns A new TemperatureGradient instance with the results.
     */
    subtract(temperatureGradient) {
        return new TemperatureGradient(this.value - temperatureGradient.BaseValue);
    }
    /**
     * Multiply the given TemperatureGradient with the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns A new TemperatureGradient instance with the results.
     */
    multiply(temperatureGradient) {
        return new TemperatureGradient(this.value * temperatureGradient.BaseValue);
    }
    /**
     * Divide the given TemperatureGradient with the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns A new TemperatureGradient instance with the results.
     */
    divide(temperatureGradient) {
        return new TemperatureGradient(this.value / temperatureGradient.BaseValue);
    }
    /**
     * Modulo the given TemperatureGradient with the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns A new TemperatureGradient instance with the results.
     */
    modulo(temperatureGradient) {
        return new TemperatureGradient(this.value % temperatureGradient.BaseValue);
    }
    /**
     * Pow the given TemperatureGradient with the current TemperatureGradient.
     * @param temperatureGradient The other TemperatureGradient.
     * @returns A new TemperatureGradient instance with the results.
     */
    pow(temperatureGradient) {
        return new TemperatureGradient(this.value ** temperatureGradient.BaseValue);
    }
}
exports.TemperatureGradient = TemperatureGradient;
