// unitsnet-js/dist/temperaturedelta.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemperatureDelta = exports.TemperatureDeltaUnits = void 0;
/** TemperatureDeltaUnits enumeration */
var TemperatureDeltaUnits;
(function (TemperatureDeltaUnits) {
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["Kelvins"] = 0] = "Kelvins";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesCelsius"] = 1] = "DegreesCelsius";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesDelisle"] = 2] = "DegreesDelisle";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesFahrenheit"] = 3] = "DegreesFahrenheit";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesNewton"] = 4] = "DegreesNewton";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesRankine"] = 5] = "DegreesRankine";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesReaumur"] = 6] = "DegreesReaumur";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["DegreesRoemer"] = 7] = "DegreesRoemer";
    /** */
    TemperatureDeltaUnits[TemperatureDeltaUnits["MillidegreesCelsius"] = 8] = "MillidegreesCelsius";
})(TemperatureDeltaUnits = exports.TemperatureDeltaUnits || (exports.TemperatureDeltaUnits = {}));
/** Difference between two temperatures. The conversions are different than for Temperature. */
class TemperatureDelta {
    /**
     * Create a new TemperatureDelta.
     * @param value The value.
     * @param fromUnit The ‘TemperatureDelta’ unit to create from.
     * The default unit is Kelvins
     */
    constructor(value, fromUnit = TemperatureDeltaUnits.Kelvins) {
        this.kelvinsLazy = null;
        this.degreescelsiusLazy = null;
        this.degreesdelisleLazy = null;
        this.degreesfahrenheitLazy = null;
        this.degreesnewtonLazy = null;
        this.degreesrankineLazy = null;
        this.degreesreaumurLazy = null;
        this.degreesroemerLazy = null;
        this.millidegreescelsiusLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of TemperatureDelta is Kelvins.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get Kelvins() {
        if (this.kelvinsLazy !== null) {
            return this.kelvinsLazy;
        }
        return this.kelvinsLazy = this.convertFromBase(TemperatureDeltaUnits.Kelvins);
    }
    /** */
    get DegreesCelsius() {
        if (this.degreescelsiusLazy !== null) {
            return this.degreescelsiusLazy;
        }
        return this.degreescelsiusLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesCelsius);
    }
    /** */
    get DegreesDelisle() {
        if (this.degreesdelisleLazy !== null) {
            return this.degreesdelisleLazy;
        }
        return this.degreesdelisleLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesDelisle);
    }
    /** */
    get DegreesFahrenheit() {
        if (this.degreesfahrenheitLazy !== null) {
            return this.degreesfahrenheitLazy;
        }
        return this.degreesfahrenheitLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesFahrenheit);
    }
    /** */
    get DegreesNewton() {
        if (this.degreesnewtonLazy !== null) {
            return this.degreesnewtonLazy;
        }
        return this.degreesnewtonLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesNewton);
    }
    /** */
    get DegreesRankine() {
        if (this.degreesrankineLazy !== null) {
            return this.degreesrankineLazy;
        }
        return this.degreesrankineLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesRankine);
    }
    /** */
    get DegreesReaumur() {
        if (this.degreesreaumurLazy !== null) {
            return this.degreesreaumurLazy;
        }
        return this.degreesreaumurLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesReaumur);
    }
    /** */
    get DegreesRoemer() {
        if (this.degreesroemerLazy !== null) {
            return this.degreesroemerLazy;
        }
        return this.degreesroemerLazy = this.convertFromBase(TemperatureDeltaUnits.DegreesRoemer);
    }
    /** */
    get MillidegreesCelsius() {
        if (this.millidegreescelsiusLazy !== null) {
            return this.millidegreescelsiusLazy;
        }
        return this.millidegreescelsiusLazy = this.convertFromBase(TemperatureDeltaUnits.MillidegreesCelsius);
    }
    /**
     * Create a new TemperatureDelta instance from a Kelvins
     *
     * @param value The unit as Kelvins to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromKelvins(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.Kelvins);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesCelsius
     *
     * @param value The unit as DegreesCelsius to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesCelsius(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesCelsius);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesDelisle
     *
     * @param value The unit as DegreesDelisle to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesDelisle(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesDelisle);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesFahrenheit
     *
     * @param value The unit as DegreesFahrenheit to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesFahrenheit(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesFahrenheit);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesNewton
     *
     * @param value The unit as DegreesNewton to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesNewton(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesNewton);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesRankine
     *
     * @param value The unit as DegreesRankine to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesRankine(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesRankine);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesReaumur
     *
     * @param value The unit as DegreesReaumur to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesReaumur(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesReaumur);
    }
    /**
     * Create a new TemperatureDelta instance from a DegreesRoemer
     *
     * @param value The unit as DegreesRoemer to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromDegreesRoemer(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.DegreesRoemer);
    }
    /**
     * Create a new TemperatureDelta instance from a MillidegreesCelsius
     *
     * @param value The unit as MillidegreesCelsius to create a new TemperatureDelta from.
     * @returns The new TemperatureDelta instance.
     */
    static FromMillidegreesCelsius(value) {
        return new TemperatureDelta(value, TemperatureDeltaUnits.MillidegreesCelsius);
    }
    /**
     * Convert TemperatureDelta to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case TemperatureDeltaUnits.Kelvins: return this.Kelvins;
            case TemperatureDeltaUnits.DegreesCelsius: return this.DegreesCelsius;
            case TemperatureDeltaUnits.DegreesDelisle: return this.DegreesDelisle;
            case TemperatureDeltaUnits.DegreesFahrenheit: return this.DegreesFahrenheit;
            case TemperatureDeltaUnits.DegreesNewton: return this.DegreesNewton;
            case TemperatureDeltaUnits.DegreesRankine: return this.DegreesRankine;
            case TemperatureDeltaUnits.DegreesReaumur: return this.DegreesReaumur;
            case TemperatureDeltaUnits.DegreesRoemer: return this.DegreesRoemer;
            case TemperatureDeltaUnits.MillidegreesCelsius: return this.MillidegreesCelsius;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case TemperatureDeltaUnits.Kelvins:
                return this.value;
            case TemperatureDeltaUnits.DegreesCelsius:
                return this.value;
            case TemperatureDeltaUnits.DegreesDelisle:
                return this.value * -3 / 2;
            case TemperatureDeltaUnits.DegreesFahrenheit:
                return this.value * 9 / 5;
            case TemperatureDeltaUnits.DegreesNewton:
                return this.value * 33 / 100;
            case TemperatureDeltaUnits.DegreesRankine:
                return this.value * 9 / 5;
            case TemperatureDeltaUnits.DegreesReaumur:
                return this.value * 4 / 5;
            case TemperatureDeltaUnits.DegreesRoemer:
                return this.value * 21 / 40;
            case TemperatureDeltaUnits.MillidegreesCelsius:
                return (this.value) / 0.001;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case TemperatureDeltaUnits.Kelvins:
                return value;
            case TemperatureDeltaUnits.DegreesCelsius:
                return value;
            case TemperatureDeltaUnits.DegreesDelisle:
                return value * -2 / 3;
            case TemperatureDeltaUnits.DegreesFahrenheit:
                return value * 5 / 9;
            case TemperatureDeltaUnits.DegreesNewton:
                return value * 100 / 33;
            case TemperatureDeltaUnits.DegreesRankine:
                return value * 5 / 9;
            case TemperatureDeltaUnits.DegreesReaumur:
                return value * 5 / 4;
            case TemperatureDeltaUnits.DegreesRoemer:
                return value * 40 / 21;
            case TemperatureDeltaUnits.MillidegreesCelsius:
                return (value) * 0.001;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the TemperatureDelta to string.
     * Note! the default format for TemperatureDelta is Kelvins.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the TemperatureDelta.
     * @returns The string format of the TemperatureDelta.
     */
    toString(unit = TemperatureDeltaUnits.Kelvins) {
        switch (unit) {
            case TemperatureDeltaUnits.Kelvins:
                return this.Kelvins + ` ∆K`;
            case TemperatureDeltaUnits.DegreesCelsius:
                return this.DegreesCelsius + ` ∆°C`;
            case TemperatureDeltaUnits.DegreesDelisle:
                return this.DegreesDelisle + ` ∆°De`;
            case TemperatureDeltaUnits.DegreesFahrenheit:
                return this.DegreesFahrenheit + ` ∆°F`;
            case TemperatureDeltaUnits.DegreesNewton:
                return this.DegreesNewton + ` ∆°N`;
            case TemperatureDeltaUnits.DegreesRankine:
                return this.DegreesRankine + ` ∆°R`;
            case TemperatureDeltaUnits.DegreesReaumur:
                return this.DegreesReaumur + ` ∆°Ré`;
            case TemperatureDeltaUnits.DegreesRoemer:
                return this.DegreesRoemer + ` ∆°Rø`;
            case TemperatureDeltaUnits.MillidegreesCelsius:
                return this.MillidegreesCelsius + ` m∆°C`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get TemperatureDelta unit abbreviation.
     * Note! the default abbreviation for TemperatureDelta is Kelvins.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the TemperatureDelta.
     * @returns The abbreviation string of TemperatureDelta.
     */
    getUnitAbbreviation(unitAbbreviation = TemperatureDeltaUnits.Kelvins) {
        switch (unitAbbreviation) {
            case TemperatureDeltaUnits.Kelvins:
                return `∆K`;
            case TemperatureDeltaUnits.DegreesCelsius:
                return `∆°C`;
            case TemperatureDeltaUnits.DegreesDelisle:
                return `∆°De`;
            case TemperatureDeltaUnits.DegreesFahrenheit:
                return `∆°F`;
            case TemperatureDeltaUnits.DegreesNewton:
                return `∆°N`;
            case TemperatureDeltaUnits.DegreesRankine:
                return `∆°R`;
            case TemperatureDeltaUnits.DegreesReaumur:
                return `∆°Ré`;
            case TemperatureDeltaUnits.DegreesRoemer:
                return `∆°Rø`;
            case TemperatureDeltaUnits.MillidegreesCelsius:
                return `m∆°C`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given TemperatureDelta are equals to the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns True if the given TemperatureDelta are equal to the current TemperatureDelta.
     */
    equals(temperatureDelta) {
        return this.value === temperatureDelta.BaseValue;
    }
    /**
     * Compare the given TemperatureDelta against the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns 0 if they are equal, -1 if the current TemperatureDelta is less then other, 1 if the current TemperatureDelta is greater then other.
     */
    compareTo(temperatureDelta) {
        if (this.value > temperatureDelta.BaseValue)
            return 1;
        if (this.value < temperatureDelta.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given TemperatureDelta with the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns A new TemperatureDelta instance with the results.
     */
    add(temperatureDelta) {
        return new TemperatureDelta(this.value + temperatureDelta.BaseValue);
    }
    /**
     * Subtract the given TemperatureDelta with the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns A new TemperatureDelta instance with the results.
     */
    subtract(temperatureDelta) {
        return new TemperatureDelta(this.value - temperatureDelta.BaseValue);
    }
    /**
     * Multiply the given TemperatureDelta with the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns A new TemperatureDelta instance with the results.
     */
    multiply(temperatureDelta) {
        return new TemperatureDelta(this.value * temperatureDelta.BaseValue);
    }
    /**
     * Divide the given TemperatureDelta with the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns A new TemperatureDelta instance with the results.
     */
    divide(temperatureDelta) {
        return new TemperatureDelta(this.value / temperatureDelta.BaseValue);
    }
    /**
     * Modulo the given TemperatureDelta with the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns A new TemperatureDelta instance with the results.
     */
    modulo(temperatureDelta) {
        return new TemperatureDelta(this.value % temperatureDelta.BaseValue);
    }
    /**
     * Pow the given TemperatureDelta with the current TemperatureDelta.
     * @param temperatureDelta The other TemperatureDelta.
     * @returns A new TemperatureDelta instance with the results.
     */
    pow(temperatureDelta) {
        return new TemperatureDelta(this.value ** temperatureDelta.BaseValue);
    }
}
exports.TemperatureDelta = TemperatureDelta;
