// unitsnet-js/dist/volumeflowperarea.g.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeFlowPerArea = exports.VolumeFlowPerAreaUnits = void 0;
/** VolumeFlowPerAreaUnits enumeration */
var VolumeFlowPerAreaUnits;
(function (VolumeFlowPerAreaUnits) {
    /** */
    VolumeFlowPerAreaUnits[VolumeFlowPerAreaUnits["CubicMetersPerSecondPerSquareMeter"] = 0] = "CubicMetersPerSecondPerSquareMeter";
    /** */
    VolumeFlowPerAreaUnits[VolumeFlowPerAreaUnits["CubicFeetPerMinutePerSquareFoot"] = 1] = "CubicFeetPerMinutePerSquareFoot";
})(VolumeFlowPerAreaUnits = exports.VolumeFlowPerAreaUnits || (exports.VolumeFlowPerAreaUnits = {}));
/** The volumetric flow rate per area is the volume of fluid which passes through a given unit surface area per unit time. */
class VolumeFlowPerArea {
    /**
     * Create a new VolumeFlowPerArea.
     * @param value The value.
     * @param fromUnit The ‘VolumeFlowPerArea’ unit to create from.
     * The default unit is CubicMetersPerSecondPerSquareMeter
     */
    constructor(value, fromUnit = VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter) {
        this.cubicmeterspersecondpersquaremeterLazy = null;
        this.cubicfeetperminutepersquarefootLazy = null;
        if (isNaN(value))
            throw new TypeError('invalid unit value ‘' + value + '’');
        this.value = this.convertToBase(value, fromUnit);
    }
    /**
     * The base value of VolumeFlowPerArea is CubicMetersPerSecondPerSquareMeter.
     * This accessor used when needs a value for calculations and it's better to use directly the base value
     */
    get BaseValue() {
        return this.value;
    }
    /** */
    get CubicMetersPerSecondPerSquareMeter() {
        if (this.cubicmeterspersecondpersquaremeterLazy !== null) {
            return this.cubicmeterspersecondpersquaremeterLazy;
        }
        return this.cubicmeterspersecondpersquaremeterLazy = this.convertFromBase(VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter);
    }
    /** */
    get CubicFeetPerMinutePerSquareFoot() {
        if (this.cubicfeetperminutepersquarefootLazy !== null) {
            return this.cubicfeetperminutepersquarefootLazy;
        }
        return this.cubicfeetperminutepersquarefootLazy = this.convertFromBase(VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot);
    }
    /**
     * Create a new VolumeFlowPerArea instance from a CubicMetersPerSecondPerSquareMeter
     *
     * @param value The unit as CubicMetersPerSecondPerSquareMeter to create a new VolumeFlowPerArea from.
     * @returns The new VolumeFlowPerArea instance.
     */
    static FromCubicMetersPerSecondPerSquareMeter(value) {
        return new VolumeFlowPerArea(value, VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter);
    }
    /**
     * Create a new VolumeFlowPerArea instance from a CubicFeetPerMinutePerSquareFoot
     *
     * @param value The unit as CubicFeetPerMinutePerSquareFoot to create a new VolumeFlowPerArea from.
     * @returns The new VolumeFlowPerArea instance.
     */
    static FromCubicFeetPerMinutePerSquareFoot(value) {
        return new VolumeFlowPerArea(value, VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot);
    }
    /**
     * Convert VolumeFlowPerArea to a specific unit value.
     * @param toUnit The specific unit to convert to
     * @returns The value of the specific unit provided.
     */
    convert(toUnit) {
        switch (toUnit) {
            case VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter: return this.CubicMetersPerSecondPerSquareMeter;
            case VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot: return this.CubicFeetPerMinutePerSquareFoot;
            default:
                break;
        }
        return NaN;
    }
    convertFromBase(toUnit) {
        switch (toUnit) {
            case VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter:
                return this.value;
            case VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot:
                return this.value * 196.850394;
            default:
                break;
        }
        return NaN;
    }
    convertToBase(value, fromUnit) {
        switch (fromUnit) {
            case VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter:
                return value;
            case VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot:
                return value / 196.850394;
            default:
                break;
        }
        return NaN;
    }
    /**
     * Format the VolumeFlowPerArea to string.
     * Note! the default format for VolumeFlowPerArea is CubicMetersPerSecondPerSquareMeter.
     * To specify the unit format set the 'unit' parameter.
     * @param unit The unit to format the VolumeFlowPerArea.
     * @returns The string format of the VolumeFlowPerArea.
     */
    toString(unit = VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter) {
        switch (unit) {
            case VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter:
                return this.CubicMetersPerSecondPerSquareMeter + ` m³/(s·m²)`;
            case VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot:
                return this.CubicFeetPerMinutePerSquareFoot + ` CFM/ft²`;
            default:
                break;
        }
        return this.value.toString();
    }
    /**
     * Get VolumeFlowPerArea unit abbreviation.
     * Note! the default abbreviation for VolumeFlowPerArea is CubicMetersPerSecondPerSquareMeter.
     * To specify the unit abbreviation set the 'unitAbbreviation' parameter.
     * @param unitAbbreviation The unit abbreviation of the VolumeFlowPerArea.
     * @returns The abbreviation string of VolumeFlowPerArea.
     */
    getUnitAbbreviation(unitAbbreviation = VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter) {
        switch (unitAbbreviation) {
            case VolumeFlowPerAreaUnits.CubicMetersPerSecondPerSquareMeter:
                return `m³/(s·m²)`;
            case VolumeFlowPerAreaUnits.CubicFeetPerMinutePerSquareFoot:
                return `CFM/ft²`;
            default:
                break;
        }
        return '';
    }
    /**
     * Check if the given VolumeFlowPerArea are equals to the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns True if the given VolumeFlowPerArea are equal to the current VolumeFlowPerArea.
     */
    equals(volumeFlowPerArea) {
        return this.value === volumeFlowPerArea.BaseValue;
    }
    /**
     * Compare the given VolumeFlowPerArea against the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns 0 if they are equal, -1 if the current VolumeFlowPerArea is less then other, 1 if the current VolumeFlowPerArea is greater then other.
     */
    compareTo(volumeFlowPerArea) {
        if (this.value > volumeFlowPerArea.BaseValue)
            return 1;
        if (this.value < volumeFlowPerArea.BaseValue)
            return -1;
        return 0;
    }
    /**
     * Add the given VolumeFlowPerArea with the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns A new VolumeFlowPerArea instance with the results.
     */
    add(volumeFlowPerArea) {
        return new VolumeFlowPerArea(this.value + volumeFlowPerArea.BaseValue);
    }
    /**
     * Subtract the given VolumeFlowPerArea with the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns A new VolumeFlowPerArea instance with the results.
     */
    subtract(volumeFlowPerArea) {
        return new VolumeFlowPerArea(this.value - volumeFlowPerArea.BaseValue);
    }
    /**
     * Multiply the given VolumeFlowPerArea with the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns A new VolumeFlowPerArea instance with the results.
     */
    multiply(volumeFlowPerArea) {
        return new VolumeFlowPerArea(this.value * volumeFlowPerArea.BaseValue);
    }
    /**
     * Divide the given VolumeFlowPerArea with the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns A new VolumeFlowPerArea instance with the results.
     */
    divide(volumeFlowPerArea) {
        return new VolumeFlowPerArea(this.value / volumeFlowPerArea.BaseValue);
    }
    /**
     * Modulo the given VolumeFlowPerArea with the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns A new VolumeFlowPerArea instance with the results.
     */
    modulo(volumeFlowPerArea) {
        return new VolumeFlowPerArea(this.value % volumeFlowPerArea.BaseValue);
    }
    /**
     * Pow the given VolumeFlowPerArea with the current VolumeFlowPerArea.
     * @param volumeFlowPerArea The other VolumeFlowPerArea.
     * @returns A new VolumeFlowPerArea instance with the results.
     */
    pow(volumeFlowPerArea) {
        return new VolumeFlowPerArea(this.value ** volumeFlowPerArea.BaseValue);
    }
}
exports.VolumeFlowPerArea = VolumeFlowPerArea;
